{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #!/usr/bin/env python\n",
    "# # -*- encoding: utf-8 -*-\n",
    "# '''\n",
    "# @Manuscript  : Integrated Proteogenomic Characterization of Cholangiocarcinoma\n",
    "# @File     : load_directory_structure.py\n",
    "# @Time     : 2022/03/11 16:32:19\n",
    "# @Author   : RanPeng \n",
    "# @Version  : python 3.8.6 \n",
    "# @Contact  : 2502388440@hotmail.com \n",
    "# @License  : (C)Copyright 2021-2022, DingLab-CHINA-SHANGHAI \n",
    "# @Description : Exploratory analysis of clinical data \n",
    "# '''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from load_directory_structure import get_directory_structure\n",
    "from utils import paste0\n",
    "import matplotlib as mpl\n",
    "mpl.rcParams['pdf.fonttype'] = 42\n",
    "%config InlineBackend.figure_format = 'svg'\n",
    "import matplotlib.pyplot as plt\n",
    "plt.rcParams['font.sans-serif'] = ['Arial'] # 用来正常显示中文标签\n",
    "plt.rcParams['axes.unicode_minus'] = False # 用来正常显示负号\n",
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The outputDir already exists: /Users/ranpeng/Desktop/Integrated_multiomics_analysis_CCA/figure_subpanels/ \n",
      "The outTableDir already exists: /Users/ranpeng/Desktop/Integrated_multiomics_analysis_CCA/output_tables/ \n",
      "The suppTableDir already exists: /Users/ranpeng/Desktop/Integrated_multiomics_analysis_CCA/supplementary_tables/ \n",
      "The resourcesDir already exists: /Users/ranpeng/Desktop/Integrated_multiomics_analysis_CCA/resources/ \n",
      "The dataDir already exists: /Users/ranpeng/Desktop/Integrated_multiomics_analysis_CCA/data/ \n"
     ]
    }
   ],
   "source": [
    "# change the contents of variable baseDir to the root analysis folder \n",
    "baseDir = \"/Users/ranpeng/Desktop/Integrated_multiomics_analysis_CCA/\"\n",
    "outputDir, outTableDir, suppTableDir, resourcesDir, dataDir = get_directory_structure(baseDir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## load clinical metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load clinical metadata (Supplementary Table 1)\n",
    "metadata  = pd.read_excel(paste0(suppTableDir,\"Supplementary_Table1.xlsx\"), sheet_name = '1. Clinical_information', index_col=0)\n",
    "metadata\n",
    "# metadata.to_excel(paste0(outTableDir,\"1_Clinical_information.xlsx\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Classify continuous and discrete variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The numerical serial features in the dataset are: \n",
      "['Age', 'Overall survival(month)', 'Overall survival(day)', 'Tumor (T) WES ID', 'Tumor (T) RNA-seq ID', 'Tumor (T) proteomic ID', 'Adjacent tissue (NAT) proteomic ID', 'Tumor (T) phosphoproteomic ID', 'Adjacent tissue (NAT) phosphoproteomic ID', 'Tumor size (cm)', 'Preoperative AFP（ng/mL）', 'Preoperative CA19-9（U/ml）', 'Preoperative CEA（ng/ml）', 'ALB, albumin, (g/L)', 'TB, total bilirubin (µmol/L)', 'ALT, aminoleucine transferase (U/L)', 'γ-GT, γ-glutamyltransferase (U/L)', '\\nAST/GOT,Aspartate aminotransferase\\n（U/L）', 'TBA, total bile acid (µmol/L)', '\\nPLT, platelet counts\\n(10^9/L)', 'PT, prothrombin time (s) ', 'Absolute tumor cellularity', 'Histology tumor cellularity'].\n",
      "\n",
      "The category variables features in dataset are: \n",
      "['Gender', 'Survival(1,dead;0,alive)', 'Histology', 'iCCA classification', 'HBV infection status', 'HBsAg', 'HBcAb', 'Cholelithiasis', 'Jaundice', 'Type II diabetes', 'Hypertension', 'Presence of fluke infection', 'Perineural invasion (PNI)', 'Lymphovascular invasion (LVI)', 'eccStage', 'iccStage', 'Grade', 'Proteome subtype', 'Phosphoproteome cluster', 'Immune cluster'].\n"
     ]
    }
   ],
   "source": [
    "from EDA import split_data_dtypes, split_numerical_serial_fea\n",
    "# excluded_columns = ['Date of surgery', 'Anatomical classification', 'Adjacent tissue (NAT) WES ID', 'Adjacent tissue (NAT) RNA-seq ID', 'T', 'N', 'M']\n",
    "excluded_columns = ['Date of surgery', 'Anatomical classification', 'Adjacent tissue (NAT) WES ID', 'Adjacent tissue (NAT) RNA-seq ID', 'T', 'N', 'M']\n",
    "metadata_excluded = metadata.drop(excluded_columns, axis = 1)\n",
    "metadata_excluded.head()\n",
    "numerical_columns, category_columns = split_numerical_serial_fea(metadata_excluded, metadata_excluded.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_excluded_numerical = metadata_excluded[numerical_columns]\n",
    "metadata_excluded_category = metadata_excluded[category_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Age</th>\n",
       "      <th>Overall survival(month)</th>\n",
       "      <th>Overall survival(day)</th>\n",
       "      <th>Tumor (T) WES ID</th>\n",
       "      <th>Tumor (T) RNA-seq ID</th>\n",
       "      <th>Tumor (T) proteomic ID</th>\n",
       "      <th>Adjacent tissue (NAT) proteomic ID</th>\n",
       "      <th>Tumor (T) phosphoproteomic ID</th>\n",
       "      <th>Adjacent tissue (NAT) phosphoproteomic ID</th>\n",
       "      <th>Tumor size (cm)</th>\n",
       "      <th>...</th>\n",
       "      <th>ALB, albumin, (g/L)</th>\n",
       "      <th>TB, total bilirubin (µmol/L)</th>\n",
       "      <th>ALT, aminoleucine transferase (U/L)</th>\n",
       "      <th>γ-GT, γ-glutamyltransferase (U/L)</th>\n",
       "      <th>\\nAST/GOT,Aspartate aminotransferase\\n（U/L）</th>\n",
       "      <th>TBA, total bile acid (µmol/L)</th>\n",
       "      <th>\\nPLT, platelet counts\\n(10^9/L)</th>\n",
       "      <th>PT, prothrombin time (s)</th>\n",
       "      <th>Absolute tumor cellularity</th>\n",
       "      <th>Histology tumor cellularity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient No.</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CCA#1</th>\n",
       "      <td>53</td>\n",
       "      <td>44.666667</td>\n",
       "      <td>1340.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60477</td>\n",
       "      <td>60478.0</td>\n",
       "      <td>CCA_402</td>\n",
       "      <td>CCA_180</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>30</td>\n",
       "      <td>61.2</td>\n",
       "      <td>339</td>\n",
       "      <td>407</td>\n",
       "      <td>487</td>\n",
       "      <td>59.8</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>15.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#2</th>\n",
       "      <td>74</td>\n",
       "      <td>40.533333</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R21026827_F20_35_253</td>\n",
       "      <td>60479</td>\n",
       "      <td>60480.0</td>\n",
       "      <td>CCA_81458</td>\n",
       "      <td>CCA_224</td>\n",
       "      <td>3.5</td>\n",
       "      <td>...</td>\n",
       "      <td>41</td>\n",
       "      <td>82.9</td>\n",
       "      <td>53</td>\n",
       "      <td>176</td>\n",
       "      <td>62</td>\n",
       "      <td>16.8</td>\n",
       "      <td>315</td>\n",
       "      <td>11.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#3</th>\n",
       "      <td>59</td>\n",
       "      <td>44.600000</td>\n",
       "      <td>1338.0</td>\n",
       "      <td>CCAT132</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60481</td>\n",
       "      <td>60484.0</td>\n",
       "      <td>CCA_228</td>\n",
       "      <td>CCA_229</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>35</td>\n",
       "      <td>153.5</td>\n",
       "      <td>54</td>\n",
       "      <td>230</td>\n",
       "      <td>54</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>267</td>\n",
       "      <td>11.9</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#4</th>\n",
       "      <td>71</td>\n",
       "      <td>14.566667</td>\n",
       "      <td>437.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R21026825_F20_33_251</td>\n",
       "      <td>68986</td>\n",
       "      <td>68987.0</td>\n",
       "      <td>CCA_181</td>\n",
       "      <td>CCA_182</td>\n",
       "      <td>1.8</td>\n",
       "      <td>...</td>\n",
       "      <td>35</td>\n",
       "      <td>242</td>\n",
       "      <td>122</td>\n",
       "      <td>61</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>256</td>\n",
       "      <td>10.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#5</th>\n",
       "      <td>60</td>\n",
       "      <td>29.066667</td>\n",
       "      <td>872.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68988</td>\n",
       "      <td>68989.0</td>\n",
       "      <td>CCA_184</td>\n",
       "      <td>CCA_185</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>27</td>\n",
       "      <td>54.7</td>\n",
       "      <td>466</td>\n",
       "      <td>349</td>\n",
       "      <td>258</td>\n",
       "      <td>10.5</td>\n",
       "      <td>354</td>\n",
       "      <td>15.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#213</th>\n",
       "      <td>75</td>\n",
       "      <td>17.766667</td>\n",
       "      <td>533.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71303</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>128.8</td>\n",
       "      <td>46</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#214</th>\n",
       "      <td>47</td>\n",
       "      <td>17.433333</td>\n",
       "      <td>523.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCA_27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>46</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>53</td>\n",
       "      <td>84</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>179</td>\n",
       "      <td>12.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#215</th>\n",
       "      <td>62</td>\n",
       "      <td>10.033333</td>\n",
       "      <td>301.0</td>\n",
       "      <td>CCAT219</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71314</td>\n",
       "      <td>71315.0</td>\n",
       "      <td>CCA_396</td>\n",
       "      <td>CCA_397</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>282.2</td>\n",
       "      <td>52</td>\n",
       "      <td>135</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>336.9</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#216</th>\n",
       "      <td>69</td>\n",
       "      <td>1.066667</td>\n",
       "      <td>32.0</td>\n",
       "      <td>CCAT220</td>\n",
       "      <td>R21026749_F18_19_177</td>\n",
       "      <td>71316</td>\n",
       "      <td>71317.0</td>\n",
       "      <td>CCA_398</td>\n",
       "      <td>CCA_399</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>22.8</td>\n",
       "      <td>37</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.72</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#217</th>\n",
       "      <td>52</td>\n",
       "      <td>9.066667</td>\n",
       "      <td>272.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R21026747_F18_17_175</td>\n",
       "      <td>71318</td>\n",
       "      <td>71319.0</td>\n",
       "      <td>CCA_30</td>\n",
       "      <td>CCA_400</td>\n",
       "      <td>9</td>\n",
       "      <td>...</td>\n",
       "      <td>46</td>\n",
       "      <td>17.7</td>\n",
       "      <td>16</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>182</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>217 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Age  Overall survival(month)  Overall survival(day)  \\\n",
       "Patient No.                                                        \n",
       "CCA#1         53                44.666667                 1340.0   \n",
       "CCA#2         74                40.533333                 1216.0   \n",
       "CCA#3         59                44.600000                 1338.0   \n",
       "CCA#4         71                14.566667                  437.0   \n",
       "CCA#5         60                29.066667                  872.0   \n",
       "...          ...                      ...                    ...   \n",
       "CCA#213       75                17.766667                  533.0   \n",
       "CCA#214       47                17.433333                  523.0   \n",
       "CCA#215       62                10.033333                  301.0   \n",
       "CCA#216       69                 1.066667                   32.0   \n",
       "CCA#217       52                 9.066667                  272.0   \n",
       "\n",
       "            Tumor (T) WES ID  Tumor (T) RNA-seq ID  Tumor (T) proteomic ID  \\\n",
       "Patient No.                                                                  \n",
       "CCA#1                    NaN                   NaN                   60477   \n",
       "CCA#2                    NaN  R21026827_F20_35_253                   60479   \n",
       "CCA#3                CCAT132                   NaN                   60481   \n",
       "CCA#4                    NaN  R21026825_F20_33_251                   68986   \n",
       "CCA#5                    NaN                   NaN                   68988   \n",
       "...                      ...                   ...                     ...   \n",
       "CCA#213                  NaN                   NaN                   71303   \n",
       "CCA#214                  NaN                   NaN                   71311   \n",
       "CCA#215              CCAT219                   NaN                   71314   \n",
       "CCA#216              CCAT220  R21026749_F18_19_177                   71316   \n",
       "CCA#217                  NaN  R21026747_F18_17_175                   71318   \n",
       "\n",
       "             Adjacent tissue (NAT) proteomic ID Tumor (T) phosphoproteomic ID  \\\n",
       "Patient No.                                                                     \n",
       "CCA#1                                   60478.0                       CCA_402   \n",
       "CCA#2                                   60480.0                     CCA_81458   \n",
       "CCA#3                                   60484.0                       CCA_228   \n",
       "CCA#4                                   68987.0                       CCA_181   \n",
       "CCA#5                                   68989.0                       CCA_184   \n",
       "...                                         ...                           ...   \n",
       "CCA#213                                     NaN                           NaN   \n",
       "CCA#214                                     NaN                        CCA_27   \n",
       "CCA#215                                 71315.0                       CCA_396   \n",
       "CCA#216                                 71317.0                       CCA_398   \n",
       "CCA#217                                 71319.0                        CCA_30   \n",
       "\n",
       "            Adjacent tissue (NAT) phosphoproteomic ID Tumor size (cm)  ...  \\\n",
       "Patient No.                                                            ...   \n",
       "CCA#1                                         CCA_180               4  ...   \n",
       "CCA#2                                         CCA_224             3.5  ...   \n",
       "CCA#3                                         CCA_229               3  ...   \n",
       "CCA#4                                         CCA_182             1.8  ...   \n",
       "CCA#5                                         CCA_185               2  ...   \n",
       "...                                               ...             ...  ...   \n",
       "CCA#213                                           NaN               3  ...   \n",
       "CCA#214                                           NaN               5  ...   \n",
       "CCA#215                                       CCA_397               2  ...   \n",
       "CCA#216                                       CCA_399               4  ...   \n",
       "CCA#217                                       CCA_400               9  ...   \n",
       "\n",
       "            ALB, albumin, (g/L) TB, total bilirubin (µmol/L)  \\\n",
       "Patient No.                                                    \n",
       "CCA#1                        30                         61.2   \n",
       "CCA#2                        41                         82.9   \n",
       "CCA#3                        35                        153.5   \n",
       "CCA#4                        35                          242   \n",
       "CCA#5                        27                         54.7   \n",
       "...                         ...                          ...   \n",
       "CCA#213                 Unknown                        128.8   \n",
       "CCA#214                      46                      Unknown   \n",
       "CCA#215                 Unknown                        282.2   \n",
       "CCA#216                 Unknown                         22.8   \n",
       "CCA#217                      46                         17.7   \n",
       "\n",
       "            ALT, aminoleucine transferase (U/L)  \\\n",
       "Patient No.                                       \n",
       "CCA#1                                       339   \n",
       "CCA#2                                        53   \n",
       "CCA#3                                        54   \n",
       "CCA#4                                       122   \n",
       "CCA#5                                       466   \n",
       "...                                         ...   \n",
       "CCA#213                                      46   \n",
       "CCA#214                                      53   \n",
       "CCA#215                                      52   \n",
       "CCA#216                                      37   \n",
       "CCA#217                                      16   \n",
       "\n",
       "            γ-GT, γ-glutamyltransferase (U/L)  \\\n",
       "Patient No.                                     \n",
       "CCA#1                                     407   \n",
       "CCA#2                                     176   \n",
       "CCA#3                                     230   \n",
       "CCA#4                                      61   \n",
       "CCA#5                                     349   \n",
       "...                                       ...   \n",
       "CCA#213                               Unknown   \n",
       "CCA#214                                    84   \n",
       "CCA#215                                   135   \n",
       "CCA#216                               Unknown   \n",
       "CCA#217                               Unknown   \n",
       "\n",
       "            \\nAST/GOT,Aspartate aminotransferase\\n（U/L）  \\\n",
       "Patient No.                                               \n",
       "CCA#1                                               487   \n",
       "CCA#2                                                62   \n",
       "CCA#3                                                54   \n",
       "CCA#4                                           Unknown   \n",
       "CCA#5                                               258   \n",
       "...                                                 ...   \n",
       "CCA#213                                         Unknown   \n",
       "CCA#214                                         Unknown   \n",
       "CCA#215                                         Unknown   \n",
       "CCA#216                                         Unknown   \n",
       "CCA#217                                         Unknown   \n",
       "\n",
       "            TBA, total bile acid (µmol/L) \\nPLT, platelet counts\\n(10^9/L)  \\\n",
       "Patient No.                                                                  \n",
       "CCA#1                                59.8                          Unknown   \n",
       "CCA#2                                16.8                              315   \n",
       "CCA#3                             Unknown                              267   \n",
       "CCA#4                             Unknown                              256   \n",
       "CCA#5                                10.5                              354   \n",
       "...                                   ...                              ...   \n",
       "CCA#213                           Unknown                          Unknown   \n",
       "CCA#214                           Unknown                              179   \n",
       "CCA#215                             336.9                          Unknown   \n",
       "CCA#216                           Unknown                          Unknown   \n",
       "CCA#217                           Unknown                              182   \n",
       "\n",
       "            PT, prothrombin time (s)  Absolute tumor cellularity  \\\n",
       "Patient No.                                                        \n",
       "CCA#1                            15.2                        NaN   \n",
       "CCA#2                            11.3                        NaN   \n",
       "CCA#3                            11.9                       0.71   \n",
       "CCA#4                            10.6                        NaN   \n",
       "CCA#5                            15.1                        NaN   \n",
       "...                               ...                        ...   \n",
       "CCA#213                           NaN                        NaN   \n",
       "CCA#214                          12.9                        NaN   \n",
       "CCA#215                           NaN                       0.70   \n",
       "CCA#216                           NaN                       0.72   \n",
       "CCA#217                           NaN                        NaN   \n",
       "\n",
       "            Histology tumor cellularity  \n",
       "Patient No.                              \n",
       "CCA#1                               NaN  \n",
       "CCA#2                               NaN  \n",
       "CCA#3                              0.69  \n",
       "CCA#4                               NaN  \n",
       "CCA#5                               NaN  \n",
       "...                                 ...  \n",
       "CCA#213                             NaN  \n",
       "CCA#214                             NaN  \n",
       "CCA#215                            0.67  \n",
       "CCA#216                            0.70  \n",
       "CCA#217                             NaN  \n",
       "\n",
       "[217 rows x 23 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_excluded_numerical.to_excel(paste0(outTableDir,\"metadata_excluded_numerical.xlsx\"))\n",
    "metadata_excluded_numerical"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gender</th>\n",
       "      <th>Survival(1,dead;0,alive)</th>\n",
       "      <th>Histology</th>\n",
       "      <th>iCCA classification</th>\n",
       "      <th>HBV infection status</th>\n",
       "      <th>HBsAg</th>\n",
       "      <th>HBcAb</th>\n",
       "      <th>Cholelithiasis</th>\n",
       "      <th>Jaundice</th>\n",
       "      <th>Type II diabetes</th>\n",
       "      <th>Hypertension</th>\n",
       "      <th>Presence of fluke infection</th>\n",
       "      <th>Perineural invasion (PNI)</th>\n",
       "      <th>Lymphovascular invasion (LVI)</th>\n",
       "      <th>eccStage</th>\n",
       "      <th>iccStage</th>\n",
       "      <th>Grade</th>\n",
       "      <th>Proteome subtype</th>\n",
       "      <th>Phosphoproteome cluster</th>\n",
       "      <th>Immune cluster</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient No.</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CCA#1</th>\n",
       "      <td>Male</td>\n",
       "      <td>1.0</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>S-III</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#2</th>\n",
       "      <td>Female</td>\n",
       "      <td>1.0</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>IIIB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-III</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Cluster_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#3</th>\n",
       "      <td>Male</td>\n",
       "      <td>1.0</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>S-III</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Cluster_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#4</th>\n",
       "      <td>Female</td>\n",
       "      <td>1.0</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-III</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#5</th>\n",
       "      <td>Male</td>\n",
       "      <td>1.0</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>II</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>S-II</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#213</th>\n",
       "      <td>Male</td>\n",
       "      <td>1.0</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>I</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#214</th>\n",
       "      <td>Male</td>\n",
       "      <td>1.0</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>S-III</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#215</th>\n",
       "      <td>Female</td>\n",
       "      <td>1.0</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-III</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Cluster_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#216</th>\n",
       "      <td>Male</td>\n",
       "      <td>1.0</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>IIIB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-III</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Cluster_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#217</th>\n",
       "      <td>Male</td>\n",
       "      <td>1.0</td>\n",
       "      <td>iCCA</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IB</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-I</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>217 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gender  Survival(1,dead;0,alive) Histology iCCA classification  \\\n",
       "Patient No.                                                                   \n",
       "CCA#1          Male                       1.0      eCCA                 NaN   \n",
       "CCA#2        Female                       1.0      eCCA                 NaN   \n",
       "CCA#3          Male                       1.0      eCCA                 NaN   \n",
       "CCA#4        Female                       1.0      eCCA                 NaN   \n",
       "CCA#5          Male                       1.0      eCCA                 NaN   \n",
       "...             ...                       ...       ...                 ...   \n",
       "CCA#213        Male                       1.0      eCCA                 NaN   \n",
       "CCA#214        Male                       1.0      eCCA                 NaN   \n",
       "CCA#215      Female                       1.0      eCCA                 NaN   \n",
       "CCA#216        Male                       1.0      eCCA                 NaN   \n",
       "CCA#217        Male                       1.0      iCCA             Unknown   \n",
       "\n",
       "            HBV infection status     HBsAg     HBcAb Cholelithiasis Jaundice  \\\n",
       "Patient No.                                                                    \n",
       "CCA#1                   Positive  Negative  Positive             No       No   \n",
       "CCA#2                   Positive  Negative  Positive             No       No   \n",
       "CCA#3                   Positive  Negative  Positive             No      Yes   \n",
       "CCA#4                   Positive  Negative  Positive            Yes      Yes   \n",
       "CCA#5                   Negative  Negative   Unknown             No      Yes   \n",
       "...                          ...       ...       ...            ...      ...   \n",
       "CCA#213                 Negative  Negative  Negative             No       No   \n",
       "CCA#214                 Positive  Positive  Positive             No       No   \n",
       "CCA#215                  Unknown   Unknown   Unknown             No       No   \n",
       "CCA#216                 Negative  Negative  Negative             No       No   \n",
       "CCA#217                 Positive  Positive  Positive             No       No   \n",
       "\n",
       "            Type II diabetes Hypertension Presence of fluke infection  \\\n",
       "Patient No.                                                             \n",
       "CCA#1                     No          Yes                          No   \n",
       "CCA#2                     No           No                          No   \n",
       "CCA#3                     No           No                          No   \n",
       "CCA#4                     No           No                          No   \n",
       "CCA#5                     No           No                          No   \n",
       "...                      ...          ...                         ...   \n",
       "CCA#213                   No          Yes                          No   \n",
       "CCA#214                   No           No                          No   \n",
       "CCA#215                   No           No                          No   \n",
       "CCA#216                  Yes           No                          No   \n",
       "CCA#217                   No           No                          No   \n",
       "\n",
       "            Perineural invasion (PNI) Lymphovascular invasion (LVI) eccStage  \\\n",
       "Patient No.                                                                    \n",
       "CCA#1                             Yes                           Yes     IIIA   \n",
       "CCA#2                              No                           Yes     IIIB   \n",
       "CCA#3                             Yes                            No     IIIC   \n",
       "CCA#4                             Yes                            No     IIIC   \n",
       "CCA#5                             Yes                       Unknown       II   \n",
       "...                               ...                           ...      ...   \n",
       "CCA#213                           Yes                            No        I   \n",
       "CCA#214                           Yes                            No     IIIC   \n",
       "CCA#215                           Yes                            No     IIIA   \n",
       "CCA#216                           Yes                            No     IIIB   \n",
       "CCA#217                            No                           Yes      NaN   \n",
       "\n",
       "            iccStage Grade Proteome subtype  Phosphoproteome cluster  \\\n",
       "Patient No.                                                            \n",
       "CCA#1            NaN  G2_3            S-III                      1.0   \n",
       "CCA#2            NaN    G2            S-III                      2.0   \n",
       "CCA#3            NaN  G2_3            S-III                      1.0   \n",
       "CCA#4            NaN    G2            S-III                      2.0   \n",
       "CCA#5            NaN  G2_3             S-II                      2.0   \n",
       "...              ...   ...              ...                      ...   \n",
       "CCA#213          NaN    G2            S-III                      NaN   \n",
       "CCA#214          NaN  G2_3            S-III                      2.0   \n",
       "CCA#215          NaN    G2            S-III                      2.0   \n",
       "CCA#216          NaN    G2            S-III                      2.0   \n",
       "CCA#217           IB    G2              S-I                      2.0   \n",
       "\n",
       "            Immune cluster  \n",
       "Patient No.                 \n",
       "CCA#1                  NaN  \n",
       "CCA#2            Cluster_6  \n",
       "CCA#3            Cluster_6  \n",
       "CCA#4                  NaN  \n",
       "CCA#5                  NaN  \n",
       "...                    ...  \n",
       "CCA#213                NaN  \n",
       "CCA#214                NaN  \n",
       "CCA#215          Cluster_6  \n",
       "CCA#216          Cluster_1  \n",
       "CCA#217                NaN  \n",
       "\n",
       "[217 rows x 20 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_excluded_category.to_excel(paste0(outTableDir,\"metadata_excluded_category.xlsx\"))\n",
    "metadata_excluded_category"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean the clinical data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1、clean the category clincial data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The eccStage unique is: ['IIIA' 'IIIB' 'IIIC' 'II' 'I' 'IVB' nan])\n",
      "The iccStage unique is: [nan 'IIIB' 'II' 'IA' 'IB' 'IIIA' 'IV'])\n",
      "The Grade unique is: ['G2_3' 'G2' 'GX' 'G1_2' 'G1' 'G3'])\n"
     ]
    }
   ],
   "source": [
    "print(f\"The eccStage unique is: {metadata_excluded_category['eccStage'].unique()})\")\n",
    "print(f\"The iccStage unique is: {metadata_excluded_category['iccStage'].unique()})\")\n",
    "print(f\"The Grade unique is: {metadata_excluded_category['Grade'].unique()})\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "category_dic = {\"Male\":1, \"Female\":0, \n",
    "                \"iCCA\":1, \"eCCA\":0,\n",
    "                \"Positive\":1, \"Negative\":0, \"Unknown\":np.nan,\n",
    "                \"Yes\":1, \"No\":0,\n",
    "                \"Yes\":1, \"No\":0,\n",
    "                \"large iBD\":1, \"small iBD\":0,\n",
    "                \"I\":1, \"IA\":1, \"IB\":1,\"II\":2, \"IIIA\":3, \"IIIB\":3, \"IIIC\":3, \"IV\":4, \"IVB\":4,\n",
    "                \"G1\":1, \"G1_2\":2, \"G2\":2,\"G2_3\":3, \"G3\":3, \"GX\":np.nan,\n",
    "                \"S-I\":1, \"S-II\":2, \"S-III\":3,\n",
    "                \"Cluster_1\":1, \"Cluster_2\":2, \"Cluster_3\":3,\"Cluster_4\":4, \"Cluster_5\":5, \"Cluster_6\":6}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_excluded_category_washed = metadata_excluded_category.replace(category_dic)\n",
    "metadata_excluded_category_washed.to_excel(paste0(outTableDir,\"metadata_excluded_category_washed.xlsx\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2、clean the numerical clincial data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Age</th>\n",
       "      <th>Overall survival(month)</th>\n",
       "      <th>Overall survival(day)</th>\n",
       "      <th>Tumor size (cm)</th>\n",
       "      <th>Preoperative AFP（ng/mL）</th>\n",
       "      <th>Preoperative CA19-9（U/ml）</th>\n",
       "      <th>Preoperative CEA（ng/ml）</th>\n",
       "      <th>ALB, albumin, (g/L)</th>\n",
       "      <th>TB, total bilirubin (µmol/L)</th>\n",
       "      <th>ALT, aminoleucine transferase (U/L)</th>\n",
       "      <th>γ-GT, γ-glutamyltransferase (U/L)</th>\n",
       "      <th>\\nAST/GOT,Aspartate aminotransferase\\n（U/L）</th>\n",
       "      <th>TBA, total bile acid (µmol/L)</th>\n",
       "      <th>\\nPLT, platelet counts\\n(10^9/L)</th>\n",
       "      <th>PT, prothrombin time (s)</th>\n",
       "      <th>Absolute tumor cellularity</th>\n",
       "      <th>Histology tumor cellularity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient No.</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CCA#1</th>\n",
       "      <td>53</td>\n",
       "      <td>44.666667</td>\n",
       "      <td>1340.0</td>\n",
       "      <td>4</td>\n",
       "      <td>5.6</td>\n",
       "      <td>32.3</td>\n",
       "      <td>2.6</td>\n",
       "      <td>30</td>\n",
       "      <td>61.2</td>\n",
       "      <td>339</td>\n",
       "      <td>407</td>\n",
       "      <td>487</td>\n",
       "      <td>59.8</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>15.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#2</th>\n",
       "      <td>74</td>\n",
       "      <td>40.533333</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>3.5</td>\n",
       "      <td>1.6</td>\n",
       "      <td>10.7</td>\n",
       "      <td>2.42</td>\n",
       "      <td>41</td>\n",
       "      <td>82.9</td>\n",
       "      <td>53</td>\n",
       "      <td>176</td>\n",
       "      <td>62</td>\n",
       "      <td>16.8</td>\n",
       "      <td>315</td>\n",
       "      <td>11.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#3</th>\n",
       "      <td>59</td>\n",
       "      <td>44.600000</td>\n",
       "      <td>1338.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.8</td>\n",
       "      <td>25.7</td>\n",
       "      <td>3.38</td>\n",
       "      <td>35</td>\n",
       "      <td>153.5</td>\n",
       "      <td>54</td>\n",
       "      <td>230</td>\n",
       "      <td>54</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>267</td>\n",
       "      <td>11.9</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#4</th>\n",
       "      <td>71</td>\n",
       "      <td>14.566667</td>\n",
       "      <td>437.0</td>\n",
       "      <td>1.8</td>\n",
       "      <td>2.5</td>\n",
       "      <td>297.4</td>\n",
       "      <td>1.17</td>\n",
       "      <td>35</td>\n",
       "      <td>242</td>\n",
       "      <td>122</td>\n",
       "      <td>61</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>256</td>\n",
       "      <td>10.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#5</th>\n",
       "      <td>60</td>\n",
       "      <td>29.066667</td>\n",
       "      <td>872.0</td>\n",
       "      <td>2</td>\n",
       "      <td>4.5</td>\n",
       "      <td>311.2</td>\n",
       "      <td>4.3</td>\n",
       "      <td>27</td>\n",
       "      <td>54.7</td>\n",
       "      <td>466</td>\n",
       "      <td>349</td>\n",
       "      <td>258</td>\n",
       "      <td>10.5</td>\n",
       "      <td>354</td>\n",
       "      <td>15.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#213</th>\n",
       "      <td>75</td>\n",
       "      <td>17.766667</td>\n",
       "      <td>533.0</td>\n",
       "      <td>3</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>128.8</td>\n",
       "      <td>46</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#214</th>\n",
       "      <td>47</td>\n",
       "      <td>17.433333</td>\n",
       "      <td>523.0</td>\n",
       "      <td>5</td>\n",
       "      <td>5.7</td>\n",
       "      <td>151.5</td>\n",
       "      <td>5.3</td>\n",
       "      <td>46</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>53</td>\n",
       "      <td>84</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>179</td>\n",
       "      <td>12.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#215</th>\n",
       "      <td>62</td>\n",
       "      <td>10.033333</td>\n",
       "      <td>301.0</td>\n",
       "      <td>2</td>\n",
       "      <td>9.8</td>\n",
       "      <td>1140</td>\n",
       "      <td>5.8</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>282.2</td>\n",
       "      <td>52</td>\n",
       "      <td>135</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>336.9</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#216</th>\n",
       "      <td>69</td>\n",
       "      <td>1.066667</td>\n",
       "      <td>32.0</td>\n",
       "      <td>4</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>22.8</td>\n",
       "      <td>37</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.72</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#217</th>\n",
       "      <td>52</td>\n",
       "      <td>9.066667</td>\n",
       "      <td>272.0</td>\n",
       "      <td>9</td>\n",
       "      <td>5.6</td>\n",
       "      <td>40.4</td>\n",
       "      <td>0.5</td>\n",
       "      <td>46</td>\n",
       "      <td>17.7</td>\n",
       "      <td>16</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>182</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>217 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Age  Overall survival(month)  Overall survival(day)  \\\n",
       "Patient No.                                                        \n",
       "CCA#1         53                44.666667                 1340.0   \n",
       "CCA#2         74                40.533333                 1216.0   \n",
       "CCA#3         59                44.600000                 1338.0   \n",
       "CCA#4         71                14.566667                  437.0   \n",
       "CCA#5         60                29.066667                  872.0   \n",
       "...          ...                      ...                    ...   \n",
       "CCA#213       75                17.766667                  533.0   \n",
       "CCA#214       47                17.433333                  523.0   \n",
       "CCA#215       62                10.033333                  301.0   \n",
       "CCA#216       69                 1.066667                   32.0   \n",
       "CCA#217       52                 9.066667                  272.0   \n",
       "\n",
       "            Tumor size (cm) Preoperative AFP（ng/mL） Preoperative CA19-9（U/ml）  \\\n",
       "Patient No.                                                                     \n",
       "CCA#1                     4                     5.6                      32.3   \n",
       "CCA#2                   3.5                     1.6                      10.7   \n",
       "CCA#3                     3                     2.8                      25.7   \n",
       "CCA#4                   1.8                     2.5                     297.4   \n",
       "CCA#5                     2                     4.5                     311.2   \n",
       "...                     ...                     ...                       ...   \n",
       "CCA#213                   3                 Unknown                   Unknown   \n",
       "CCA#214                   5                     5.7                     151.5   \n",
       "CCA#215                   2                     9.8                      1140   \n",
       "CCA#216                   4                 Unknown                   Unknown   \n",
       "CCA#217                   9                     5.6                      40.4   \n",
       "\n",
       "            Preoperative CEA（ng/ml） ALB, albumin, (g/L)  \\\n",
       "Patient No.                                               \n",
       "CCA#1                           2.6                  30   \n",
       "CCA#2                          2.42                  41   \n",
       "CCA#3                          3.38                  35   \n",
       "CCA#4                          1.17                  35   \n",
       "CCA#5                           4.3                  27   \n",
       "...                             ...                 ...   \n",
       "CCA#213                     Unknown             Unknown   \n",
       "CCA#214                         5.3                  46   \n",
       "CCA#215                         5.8             Unknown   \n",
       "CCA#216                     Unknown             Unknown   \n",
       "CCA#217                         0.5                  46   \n",
       "\n",
       "            TB, total bilirubin (µmol/L) ALT, aminoleucine transferase (U/L)  \\\n",
       "Patient No.                                                                    \n",
       "CCA#1                               61.2                                 339   \n",
       "CCA#2                               82.9                                  53   \n",
       "CCA#3                              153.5                                  54   \n",
       "CCA#4                                242                                 122   \n",
       "CCA#5                               54.7                                 466   \n",
       "...                                  ...                                 ...   \n",
       "CCA#213                            128.8                                  46   \n",
       "CCA#214                          Unknown                                  53   \n",
       "CCA#215                            282.2                                  52   \n",
       "CCA#216                             22.8                                  37   \n",
       "CCA#217                             17.7                                  16   \n",
       "\n",
       "            γ-GT, γ-glutamyltransferase (U/L)  \\\n",
       "Patient No.                                     \n",
       "CCA#1                                     407   \n",
       "CCA#2                                     176   \n",
       "CCA#3                                     230   \n",
       "CCA#4                                      61   \n",
       "CCA#5                                     349   \n",
       "...                                       ...   \n",
       "CCA#213                               Unknown   \n",
       "CCA#214                                    84   \n",
       "CCA#215                                   135   \n",
       "CCA#216                               Unknown   \n",
       "CCA#217                               Unknown   \n",
       "\n",
       "            \\nAST/GOT,Aspartate aminotransferase\\n（U/L）  \\\n",
       "Patient No.                                               \n",
       "CCA#1                                               487   \n",
       "CCA#2                                                62   \n",
       "CCA#3                                                54   \n",
       "CCA#4                                           Unknown   \n",
       "CCA#5                                               258   \n",
       "...                                                 ...   \n",
       "CCA#213                                         Unknown   \n",
       "CCA#214                                         Unknown   \n",
       "CCA#215                                         Unknown   \n",
       "CCA#216                                         Unknown   \n",
       "CCA#217                                         Unknown   \n",
       "\n",
       "            TBA, total bile acid (µmol/L) \\nPLT, platelet counts\\n(10^9/L)  \\\n",
       "Patient No.                                                                  \n",
       "CCA#1                                59.8                          Unknown   \n",
       "CCA#2                                16.8                              315   \n",
       "CCA#3                             Unknown                              267   \n",
       "CCA#4                             Unknown                              256   \n",
       "CCA#5                                10.5                              354   \n",
       "...                                   ...                              ...   \n",
       "CCA#213                           Unknown                          Unknown   \n",
       "CCA#214                           Unknown                              179   \n",
       "CCA#215                             336.9                          Unknown   \n",
       "CCA#216                           Unknown                          Unknown   \n",
       "CCA#217                           Unknown                              182   \n",
       "\n",
       "             PT, prothrombin time (s)   Absolute tumor cellularity  \\\n",
       "Patient No.                                                          \n",
       "CCA#1                             15.2                         NaN   \n",
       "CCA#2                             11.3                         NaN   \n",
       "CCA#3                             11.9                        0.71   \n",
       "CCA#4                             10.6                         NaN   \n",
       "CCA#5                             15.1                         NaN   \n",
       "...                                ...                         ...   \n",
       "CCA#213                            NaN                         NaN   \n",
       "CCA#214                           12.9                         NaN   \n",
       "CCA#215                            NaN                        0.70   \n",
       "CCA#216                            NaN                        0.72   \n",
       "CCA#217                            NaN                         NaN   \n",
       "\n",
       "             Histology tumor cellularity  \n",
       "Patient No.                               \n",
       "CCA#1                                NaN  \n",
       "CCA#2                                NaN  \n",
       "CCA#3                               0.69  \n",
       "CCA#4                                NaN  \n",
       "CCA#5                                NaN  \n",
       "...                                  ...  \n",
       "CCA#213                              NaN  \n",
       "CCA#214                              NaN  \n",
       "CCA#215                             0.67  \n",
       "CCA#216                             0.70  \n",
       "CCA#217                              NaN  \n",
       "\n",
       "[217 rows x 17 columns]"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "multiomics_profiling_index = [\n",
    "    'Tumor (T) WES ID', 'Tumor (T) RNA-seq ID', \n",
    "    'Tumor (T) proteomic ID', 'Adjacent tissue (NAT) proteomic ID',\n",
    "    'Tumor (T) phosphoproteomic ID', 'Adjacent tissue (NAT) phosphoproteomic ID']\n",
    "\n",
    "\n",
    "multiomics_profiling_index = metadata_excluded_numerical[multiomics_profiling_index]\n",
    "numerical_columns_select = [i for i in metadata_excluded_numerical.columns if i not in multiomics_profiling_index]\n",
    "numerical_clinical_feature = metadata_excluded_numerical[numerical_columns_select]\n",
    "# multiomics_profiling_clinical = multiomics_profiling_index.append(multiomics_profiling_clinical)\n",
    "numerical_clinical_feature\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/ranpeng/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py:5238: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  return super().replace(\n"
     ]
    }
   ],
   "source": [
    "numerical_clinical_feature.replace('>', '', regex=True, inplace=True)\n",
    "numerical_clinical_feature.replace('<', '', regex=True, inplace=True)\n",
    "numerical_clinical_feature.replace('Unknown', np.nan, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Age                                              int64\n",
       "Overall survival(month)                        float64\n",
       "Overall survival(day)                          float64\n",
       "Tumor size (cm)                                float64\n",
       "Preoperative AFP（ng/mL）                         object\n",
       "Preoperative CA19-9（U/ml）                       object\n",
       "Preoperative CEA（ng/ml）                        float64\n",
       "ALB, albumin, (g/L)                            float64\n",
       "TB, total bilirubin (µmol/L)                   float64\n",
       "ALT, aminoleucine transferase (U/L)            float64\n",
       "γ-GT, γ-glutamyltransferase (U/L)              float64\n",
       "\\nAST/GOT,Aspartate aminotransferase\\n（U/L）    float64\n",
       "TBA, total bile acid (µmol/L)                   object\n",
       "\\nPLT, platelet counts\\n(10^9/L)               float64\n",
       "PT, prothrombin time (s)                       float64\n",
       "Absolute tumor cellularity                     float64\n",
       "Histology tumor cellularity                    float64\n",
       "dtype: object"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "numerical_clinical_feature.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-119-771d5bd55b56>:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  numerical_clinical_feature['Preoperative AFP（ng/mL）'] = numerical_clinical_feature['Preoperative AFP（ng/mL）'].astype('float64')\n",
      "<ipython-input-119-771d5bd55b56>:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  numerical_clinical_feature['Preoperative CA19-9（U/ml）'] = numerical_clinical_feature['Preoperative CA19-9（U/ml）'].astype('float64')\n",
      "<ipython-input-119-771d5bd55b56>:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  numerical_clinical_feature['TBA, total bile acid (µmol/L)'] = numerical_clinical_feature['TBA, total bile acid (µmol/L)'].astype('float64')\n"
     ]
    }
   ],
   "source": [
    "numerical_clinical_feature['Preoperative AFP（ng/mL）'] = numerical_clinical_feature['Preoperative AFP（ng/mL）'].astype('float64')\n",
    "numerical_clinical_feature['Preoperative CA19-9（U/ml）'] = numerical_clinical_feature['Preoperative CA19-9（U/ml）'].astype('float64')\n",
    "numerical_clinical_feature['TBA, total bile acid (µmol/L)'] = numerical_clinical_feature['TBA, total bile acid (µmol/L)'].astype('float64')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "numerical_clinical_feature.to_excel(paste0(outTableDir,\"numerical_clinical_feature_washed.xlsx\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor (T) WES ID</th>\n",
       "      <th>Tumor (T) RNA-seq ID</th>\n",
       "      <th>Tumor (T) proteomic ID</th>\n",
       "      <th>Adjacent tissue (NAT) proteomic ID</th>\n",
       "      <th>Tumor (T) phosphoproteomic ID</th>\n",
       "      <th>Adjacent tissue (NAT) phosphoproteomic ID</th>\n",
       "      <th>Age</th>\n",
       "      <th>Overall survival(month)</th>\n",
       "      <th>Overall survival(day)</th>\n",
       "      <th>Tumor size (cm)</th>\n",
       "      <th>...</th>\n",
       "      <th>ALB, albumin, (g/L)</th>\n",
       "      <th>TB, total bilirubin (µmol/L)</th>\n",
       "      <th>ALT, aminoleucine transferase (U/L)</th>\n",
       "      <th>γ-GT, γ-glutamyltransferase (U/L)</th>\n",
       "      <th>\\nAST/GOT,Aspartate aminotransferase\\n（U/L）</th>\n",
       "      <th>TBA, total bile acid (µmol/L)</th>\n",
       "      <th>\\nPLT, platelet counts\\n(10^9/L)</th>\n",
       "      <th>PT, prothrombin time (s)</th>\n",
       "      <th>Absolute tumor cellularity</th>\n",
       "      <th>Histology tumor cellularity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient No.</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CCA#1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60477</td>\n",
       "      <td>60478.0</td>\n",
       "      <td>CCA_402</td>\n",
       "      <td>CCA_180</td>\n",
       "      <td>53</td>\n",
       "      <td>44.666667</td>\n",
       "      <td>1340.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>30.0</td>\n",
       "      <td>61.2</td>\n",
       "      <td>339.0</td>\n",
       "      <td>407.0</td>\n",
       "      <td>487.0</td>\n",
       "      <td>59.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>R21026827_F20_35_253</td>\n",
       "      <td>60479</td>\n",
       "      <td>60480.0</td>\n",
       "      <td>CCA_81458</td>\n",
       "      <td>CCA_224</td>\n",
       "      <td>74</td>\n",
       "      <td>40.533333</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>3.5</td>\n",
       "      <td>...</td>\n",
       "      <td>41.0</td>\n",
       "      <td>82.9</td>\n",
       "      <td>53.0</td>\n",
       "      <td>176.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>16.8</td>\n",
       "      <td>315.0</td>\n",
       "      <td>11.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#3</th>\n",
       "      <td>CCAT132</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60481</td>\n",
       "      <td>60484.0</td>\n",
       "      <td>CCA_228</td>\n",
       "      <td>CCA_229</td>\n",
       "      <td>59</td>\n",
       "      <td>44.600000</td>\n",
       "      <td>1338.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>35.0</td>\n",
       "      <td>153.5</td>\n",
       "      <td>54.0</td>\n",
       "      <td>230.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>267.0</td>\n",
       "      <td>11.9</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>R21026825_F20_33_251</td>\n",
       "      <td>68986</td>\n",
       "      <td>68987.0</td>\n",
       "      <td>CCA_181</td>\n",
       "      <td>CCA_182</td>\n",
       "      <td>71</td>\n",
       "      <td>14.566667</td>\n",
       "      <td>437.0</td>\n",
       "      <td>1.8</td>\n",
       "      <td>...</td>\n",
       "      <td>35.0</td>\n",
       "      <td>242.0</td>\n",
       "      <td>122.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>256.0</td>\n",
       "      <td>10.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68988</td>\n",
       "      <td>68989.0</td>\n",
       "      <td>CCA_184</td>\n",
       "      <td>CCA_185</td>\n",
       "      <td>60</td>\n",
       "      <td>29.066667</td>\n",
       "      <td>872.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>27.0</td>\n",
       "      <td>54.7</td>\n",
       "      <td>466.0</td>\n",
       "      <td>349.0</td>\n",
       "      <td>258.0</td>\n",
       "      <td>10.5</td>\n",
       "      <td>354.0</td>\n",
       "      <td>15.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#213</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71303</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75</td>\n",
       "      <td>17.766667</td>\n",
       "      <td>533.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>128.8</td>\n",
       "      <td>46.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#214</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCA_27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47</td>\n",
       "      <td>17.433333</td>\n",
       "      <td>523.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>...</td>\n",
       "      <td>46.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>53.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>179.0</td>\n",
       "      <td>12.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#215</th>\n",
       "      <td>CCAT219</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71314</td>\n",
       "      <td>71315.0</td>\n",
       "      <td>CCA_396</td>\n",
       "      <td>CCA_397</td>\n",
       "      <td>62</td>\n",
       "      <td>10.033333</td>\n",
       "      <td>301.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>282.2</td>\n",
       "      <td>52.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>336.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#216</th>\n",
       "      <td>CCAT220</td>\n",
       "      <td>R21026749_F18_19_177</td>\n",
       "      <td>71316</td>\n",
       "      <td>71317.0</td>\n",
       "      <td>CCA_398</td>\n",
       "      <td>CCA_399</td>\n",
       "      <td>69</td>\n",
       "      <td>1.066667</td>\n",
       "      <td>32.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.8</td>\n",
       "      <td>37.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.72</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#217</th>\n",
       "      <td>NaN</td>\n",
       "      <td>R21026747_F18_17_175</td>\n",
       "      <td>71318</td>\n",
       "      <td>71319.0</td>\n",
       "      <td>CCA_30</td>\n",
       "      <td>CCA_400</td>\n",
       "      <td>52</td>\n",
       "      <td>9.066667</td>\n",
       "      <td>272.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>...</td>\n",
       "      <td>46.0</td>\n",
       "      <td>17.7</td>\n",
       "      <td>16.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>182.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>217 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Tumor (T) WES ID  Tumor (T) RNA-seq ID  Tumor (T) proteomic ID  \\\n",
       "Patient No.                                                                  \n",
       "CCA#1                    NaN                   NaN                   60477   \n",
       "CCA#2                    NaN  R21026827_F20_35_253                   60479   \n",
       "CCA#3                CCAT132                   NaN                   60481   \n",
       "CCA#4                    NaN  R21026825_F20_33_251                   68986   \n",
       "CCA#5                    NaN                   NaN                   68988   \n",
       "...                      ...                   ...                     ...   \n",
       "CCA#213                  NaN                   NaN                   71303   \n",
       "CCA#214                  NaN                   NaN                   71311   \n",
       "CCA#215              CCAT219                   NaN                   71314   \n",
       "CCA#216              CCAT220  R21026749_F18_19_177                   71316   \n",
       "CCA#217                  NaN  R21026747_F18_17_175                   71318   \n",
       "\n",
       "             Adjacent tissue (NAT) proteomic ID Tumor (T) phosphoproteomic ID  \\\n",
       "Patient No.                                                                     \n",
       "CCA#1                                   60478.0                       CCA_402   \n",
       "CCA#2                                   60480.0                     CCA_81458   \n",
       "CCA#3                                   60484.0                       CCA_228   \n",
       "CCA#4                                   68987.0                       CCA_181   \n",
       "CCA#5                                   68989.0                       CCA_184   \n",
       "...                                         ...                           ...   \n",
       "CCA#213                                     NaN                           NaN   \n",
       "CCA#214                                     NaN                        CCA_27   \n",
       "CCA#215                                 71315.0                       CCA_396   \n",
       "CCA#216                                 71317.0                       CCA_398   \n",
       "CCA#217                                 71319.0                        CCA_30   \n",
       "\n",
       "            Adjacent tissue (NAT) phosphoproteomic ID  Age  \\\n",
       "Patient No.                                                  \n",
       "CCA#1                                         CCA_180   53   \n",
       "CCA#2                                         CCA_224   74   \n",
       "CCA#3                                         CCA_229   59   \n",
       "CCA#4                                         CCA_182   71   \n",
       "CCA#5                                         CCA_185   60   \n",
       "...                                               ...  ...   \n",
       "CCA#213                                           NaN   75   \n",
       "CCA#214                                           NaN   47   \n",
       "CCA#215                                       CCA_397   62   \n",
       "CCA#216                                       CCA_399   69   \n",
       "CCA#217                                       CCA_400   52   \n",
       "\n",
       "             Overall survival(month)  Overall survival(day)  Tumor size (cm)  \\\n",
       "Patient No.                                                                    \n",
       "CCA#1                      44.666667                 1340.0              4.0   \n",
       "CCA#2                      40.533333                 1216.0              3.5   \n",
       "CCA#3                      44.600000                 1338.0              3.0   \n",
       "CCA#4                      14.566667                  437.0              1.8   \n",
       "CCA#5                      29.066667                  872.0              2.0   \n",
       "...                              ...                    ...              ...   \n",
       "CCA#213                    17.766667                  533.0              3.0   \n",
       "CCA#214                    17.433333                  523.0              5.0   \n",
       "CCA#215                    10.033333                  301.0              2.0   \n",
       "CCA#216                     1.066667                   32.0              4.0   \n",
       "CCA#217                     9.066667                  272.0              9.0   \n",
       "\n",
       "             ...  ALB, albumin, (g/L)  TB, total bilirubin (µmol/L)  \\\n",
       "Patient No.  ...                                                      \n",
       "CCA#1        ...                 30.0                          61.2   \n",
       "CCA#2        ...                 41.0                          82.9   \n",
       "CCA#3        ...                 35.0                         153.5   \n",
       "CCA#4        ...                 35.0                         242.0   \n",
       "CCA#5        ...                 27.0                          54.7   \n",
       "...          ...                  ...                           ...   \n",
       "CCA#213      ...                  NaN                         128.8   \n",
       "CCA#214      ...                 46.0                           NaN   \n",
       "CCA#215      ...                  NaN                         282.2   \n",
       "CCA#216      ...                  NaN                          22.8   \n",
       "CCA#217      ...                 46.0                          17.7   \n",
       "\n",
       "             ALT, aminoleucine transferase (U/L)  \\\n",
       "Patient No.                                        \n",
       "CCA#1                                      339.0   \n",
       "CCA#2                                       53.0   \n",
       "CCA#3                                       54.0   \n",
       "CCA#4                                      122.0   \n",
       "CCA#5                                      466.0   \n",
       "...                                          ...   \n",
       "CCA#213                                     46.0   \n",
       "CCA#214                                     53.0   \n",
       "CCA#215                                     52.0   \n",
       "CCA#216                                     37.0   \n",
       "CCA#217                                     16.0   \n",
       "\n",
       "             γ-GT, γ-glutamyltransferase (U/L)  \\\n",
       "Patient No.                                      \n",
       "CCA#1                                    407.0   \n",
       "CCA#2                                    176.0   \n",
       "CCA#3                                    230.0   \n",
       "CCA#4                                     61.0   \n",
       "CCA#5                                    349.0   \n",
       "...                                        ...   \n",
       "CCA#213                                    NaN   \n",
       "CCA#214                                   84.0   \n",
       "CCA#215                                  135.0   \n",
       "CCA#216                                    NaN   \n",
       "CCA#217                                    NaN   \n",
       "\n",
       "             \\nAST/GOT,Aspartate aminotransferase\\n（U/L）  \\\n",
       "Patient No.                                                \n",
       "CCA#1                                              487.0   \n",
       "CCA#2                                               62.0   \n",
       "CCA#3                                               54.0   \n",
       "CCA#4                                                NaN   \n",
       "CCA#5                                              258.0   \n",
       "...                                                  ...   \n",
       "CCA#213                                              NaN   \n",
       "CCA#214                                              NaN   \n",
       "CCA#215                                              NaN   \n",
       "CCA#216                                              NaN   \n",
       "CCA#217                                              NaN   \n",
       "\n",
       "             TBA, total bile acid (µmol/L)  \\nPLT, platelet counts\\n(10^9/L)  \\\n",
       "Patient No.                                                                    \n",
       "CCA#1                                 59.8                               NaN   \n",
       "CCA#2                                 16.8                             315.0   \n",
       "CCA#3                                  NaN                             267.0   \n",
       "CCA#4                                  NaN                             256.0   \n",
       "CCA#5                                 10.5                             354.0   \n",
       "...                                    ...                               ...   \n",
       "CCA#213                                NaN                               NaN   \n",
       "CCA#214                                NaN                             179.0   \n",
       "CCA#215                              336.9                               NaN   \n",
       "CCA#216                                NaN                               NaN   \n",
       "CCA#217                                NaN                             182.0   \n",
       "\n",
       "             PT, prothrombin time (s)   Absolute tumor cellularity  \\\n",
       "Patient No.                                                          \n",
       "CCA#1                             15.2                         NaN   \n",
       "CCA#2                             11.3                         NaN   \n",
       "CCA#3                             11.9                        0.71   \n",
       "CCA#4                             10.6                         NaN   \n",
       "CCA#5                             15.1                         NaN   \n",
       "...                                ...                         ...   \n",
       "CCA#213                            NaN                         NaN   \n",
       "CCA#214                           12.9                         NaN   \n",
       "CCA#215                            NaN                        0.70   \n",
       "CCA#216                            NaN                        0.72   \n",
       "CCA#217                            NaN                         NaN   \n",
       "\n",
       "             Histology tumor cellularity  \n",
       "Patient No.                               \n",
       "CCA#1                                NaN  \n",
       "CCA#2                                NaN  \n",
       "CCA#3                               0.69  \n",
       "CCA#4                                NaN  \n",
       "CCA#5                                NaN  \n",
       "...                                  ...  \n",
       "CCA#213                              NaN  \n",
       "CCA#214                              NaN  \n",
       "CCA#215                             0.67  \n",
       "CCA#216                             0.70  \n",
       "CCA#217                              NaN  \n",
       "\n",
       "[217 rows x 23 columns]"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "numerical_clinical_feature_washed = pd.concat([multiomics_profiling_index, numerical_clinical_feature], axis=1)\n",
    "numerical_clinical_feature_washed.to_excel(paste0(outTableDir,\"numerical_clinical_feature_washed1.xlsx\"))\n",
    "numerical_clinical_feature_washed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3、merge the clean numerical and category clincial data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor (T) WES ID</th>\n",
       "      <th>Tumor (T) RNA-seq ID</th>\n",
       "      <th>Tumor (T) proteomic ID</th>\n",
       "      <th>Adjacent tissue (NAT) proteomic ID</th>\n",
       "      <th>Tumor (T) phosphoproteomic ID</th>\n",
       "      <th>Adjacent tissue (NAT) phosphoproteomic ID</th>\n",
       "      <th>Age</th>\n",
       "      <th>Overall survival(month)</th>\n",
       "      <th>Overall survival(day)</th>\n",
       "      <th>Tumor size (cm)</th>\n",
       "      <th>...</th>\n",
       "      <th>Hypertension</th>\n",
       "      <th>Presence of fluke infection</th>\n",
       "      <th>Perineural invasion (PNI)</th>\n",
       "      <th>Lymphovascular invasion (LVI)</th>\n",
       "      <th>eccStage</th>\n",
       "      <th>iccStage</th>\n",
       "      <th>Grade</th>\n",
       "      <th>Proteome subtype</th>\n",
       "      <th>Phosphoproteome cluster</th>\n",
       "      <th>Immune cluster</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient No.</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CCA#1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60477</td>\n",
       "      <td>60478.0</td>\n",
       "      <td>CCA_402</td>\n",
       "      <td>CCA_180</td>\n",
       "      <td>53</td>\n",
       "      <td>44.666667</td>\n",
       "      <td>1340.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>R21026827_F20_35_253</td>\n",
       "      <td>60479</td>\n",
       "      <td>60480.0</td>\n",
       "      <td>CCA_81458</td>\n",
       "      <td>CCA_224</td>\n",
       "      <td>74</td>\n",
       "      <td>40.533333</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>3.5</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#3</th>\n",
       "      <td>CCAT132</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60481</td>\n",
       "      <td>60484.0</td>\n",
       "      <td>CCA_228</td>\n",
       "      <td>CCA_229</td>\n",
       "      <td>59</td>\n",
       "      <td>44.600000</td>\n",
       "      <td>1338.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>R21026825_F20_33_251</td>\n",
       "      <td>68986</td>\n",
       "      <td>68987.0</td>\n",
       "      <td>CCA_181</td>\n",
       "      <td>CCA_182</td>\n",
       "      <td>71</td>\n",
       "      <td>14.566667</td>\n",
       "      <td>437.0</td>\n",
       "      <td>1.8</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68988</td>\n",
       "      <td>68989.0</td>\n",
       "      <td>CCA_184</td>\n",
       "      <td>CCA_185</td>\n",
       "      <td>60</td>\n",
       "      <td>29.066667</td>\n",
       "      <td>872.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#213</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71303</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75</td>\n",
       "      <td>17.766667</td>\n",
       "      <td>533.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#214</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCA_27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47</td>\n",
       "      <td>17.433333</td>\n",
       "      <td>523.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#215</th>\n",
       "      <td>CCAT219</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71314</td>\n",
       "      <td>71315.0</td>\n",
       "      <td>CCA_396</td>\n",
       "      <td>CCA_397</td>\n",
       "      <td>62</td>\n",
       "      <td>10.033333</td>\n",
       "      <td>301.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#216</th>\n",
       "      <td>CCAT220</td>\n",
       "      <td>R21026749_F18_19_177</td>\n",
       "      <td>71316</td>\n",
       "      <td>71317.0</td>\n",
       "      <td>CCA_398</td>\n",
       "      <td>CCA_399</td>\n",
       "      <td>69</td>\n",
       "      <td>1.066667</td>\n",
       "      <td>32.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#217</th>\n",
       "      <td>NaN</td>\n",
       "      <td>R21026747_F18_17_175</td>\n",
       "      <td>71318</td>\n",
       "      <td>71319.0</td>\n",
       "      <td>CCA_30</td>\n",
       "      <td>CCA_400</td>\n",
       "      <td>52</td>\n",
       "      <td>9.066667</td>\n",
       "      <td>272.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>217 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Tumor (T) WES ID  Tumor (T) RNA-seq ID  Tumor (T) proteomic ID  \\\n",
       "Patient No.                                                                  \n",
       "CCA#1                    NaN                   NaN                   60477   \n",
       "CCA#2                    NaN  R21026827_F20_35_253                   60479   \n",
       "CCA#3                CCAT132                   NaN                   60481   \n",
       "CCA#4                    NaN  R21026825_F20_33_251                   68986   \n",
       "CCA#5                    NaN                   NaN                   68988   \n",
       "...                      ...                   ...                     ...   \n",
       "CCA#213                  NaN                   NaN                   71303   \n",
       "CCA#214                  NaN                   NaN                   71311   \n",
       "CCA#215              CCAT219                   NaN                   71314   \n",
       "CCA#216              CCAT220  R21026749_F18_19_177                   71316   \n",
       "CCA#217                  NaN  R21026747_F18_17_175                   71318   \n",
       "\n",
       "             Adjacent tissue (NAT) proteomic ID Tumor (T) phosphoproteomic ID  \\\n",
       "Patient No.                                                                     \n",
       "CCA#1                                   60478.0                       CCA_402   \n",
       "CCA#2                                   60480.0                     CCA_81458   \n",
       "CCA#3                                   60484.0                       CCA_228   \n",
       "CCA#4                                   68987.0                       CCA_181   \n",
       "CCA#5                                   68989.0                       CCA_184   \n",
       "...                                         ...                           ...   \n",
       "CCA#213                                     NaN                           NaN   \n",
       "CCA#214                                     NaN                        CCA_27   \n",
       "CCA#215                                 71315.0                       CCA_396   \n",
       "CCA#216                                 71317.0                       CCA_398   \n",
       "CCA#217                                 71319.0                        CCA_30   \n",
       "\n",
       "            Adjacent tissue (NAT) phosphoproteomic ID  Age  \\\n",
       "Patient No.                                                  \n",
       "CCA#1                                         CCA_180   53   \n",
       "CCA#2                                         CCA_224   74   \n",
       "CCA#3                                         CCA_229   59   \n",
       "CCA#4                                         CCA_182   71   \n",
       "CCA#5                                         CCA_185   60   \n",
       "...                                               ...  ...   \n",
       "CCA#213                                           NaN   75   \n",
       "CCA#214                                           NaN   47   \n",
       "CCA#215                                       CCA_397   62   \n",
       "CCA#216                                       CCA_399   69   \n",
       "CCA#217                                       CCA_400   52   \n",
       "\n",
       "             Overall survival(month)  Overall survival(day)  Tumor size (cm)  \\\n",
       "Patient No.                                                                    \n",
       "CCA#1                      44.666667                 1340.0              4.0   \n",
       "CCA#2                      40.533333                 1216.0              3.5   \n",
       "CCA#3                      44.600000                 1338.0              3.0   \n",
       "CCA#4                      14.566667                  437.0              1.8   \n",
       "CCA#5                      29.066667                  872.0              2.0   \n",
       "...                              ...                    ...              ...   \n",
       "CCA#213                    17.766667                  533.0              3.0   \n",
       "CCA#214                    17.433333                  523.0              5.0   \n",
       "CCA#215                    10.033333                  301.0              2.0   \n",
       "CCA#216                     1.066667                   32.0              4.0   \n",
       "CCA#217                     9.066667                  272.0              9.0   \n",
       "\n",
       "             ...  Hypertension  Presence of fluke infection  \\\n",
       "Patient No.  ...                                              \n",
       "CCA#1        ...             1                            0   \n",
       "CCA#2        ...             0                            0   \n",
       "CCA#3        ...             0                            0   \n",
       "CCA#4        ...             0                            0   \n",
       "CCA#5        ...             0                            0   \n",
       "...          ...           ...                          ...   \n",
       "CCA#213      ...             1                            0   \n",
       "CCA#214      ...             0                            0   \n",
       "CCA#215      ...             0                            0   \n",
       "CCA#216      ...             0                            0   \n",
       "CCA#217      ...             0                            0   \n",
       "\n",
       "             Perineural invasion (PNI)  Lymphovascular invasion (LVI)  \\\n",
       "Patient No.                                                             \n",
       "CCA#1                              1.0                            1.0   \n",
       "CCA#2                              0.0                            1.0   \n",
       "CCA#3                              1.0                            0.0   \n",
       "CCA#4                              1.0                            0.0   \n",
       "CCA#5                              1.0                            NaN   \n",
       "...                                ...                            ...   \n",
       "CCA#213                            1.0                            0.0   \n",
       "CCA#214                            1.0                            0.0   \n",
       "CCA#215                            1.0                            0.0   \n",
       "CCA#216                            1.0                            0.0   \n",
       "CCA#217                            0.0                            1.0   \n",
       "\n",
       "             eccStage  iccStage  Grade  Proteome subtype  \\\n",
       "Patient No.                                                \n",
       "CCA#1             3.0       NaN    3.0                 3   \n",
       "CCA#2             3.0       NaN    2.0                 3   \n",
       "CCA#3             3.0       NaN    3.0                 3   \n",
       "CCA#4             3.0       NaN    2.0                 3   \n",
       "CCA#5             2.0       NaN    3.0                 2   \n",
       "...               ...       ...    ...               ...   \n",
       "CCA#213           1.0       NaN    2.0                 3   \n",
       "CCA#214           3.0       NaN    3.0                 3   \n",
       "CCA#215           3.0       NaN    2.0                 3   \n",
       "CCA#216           3.0       NaN    2.0                 3   \n",
       "CCA#217           NaN       1.0    2.0                 1   \n",
       "\n",
       "             Phosphoproteome cluster  Immune cluster  \n",
       "Patient No.                                           \n",
       "CCA#1                            1.0             NaN  \n",
       "CCA#2                            2.0             6.0  \n",
       "CCA#3                            1.0             6.0  \n",
       "CCA#4                            2.0             NaN  \n",
       "CCA#5                            2.0             NaN  \n",
       "...                              ...             ...  \n",
       "CCA#213                          NaN             NaN  \n",
       "CCA#214                          2.0             NaN  \n",
       "CCA#215                          2.0             6.0  \n",
       "CCA#216                          2.0             1.0  \n",
       "CCA#217                          2.0             NaN  \n",
       "\n",
       "[217 rows x 43 columns]"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_information_washed = pd.concat([numerical_clinical_feature_washed, metadata_excluded_category_washed], axis=1)\n",
    "clinical_information_washed.to_excel(paste0(outTableDir,\"clinical_information_washed.xlsx\"))\n",
    "clinical_information_washed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## gene mutation oncoplot anatation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Proteome_Subtype</th>\n",
       "      <th>Cholelithiasis_1</th>\n",
       "      <th>Perineural_invasion</th>\n",
       "      <th>Presence_of_fluke_infection</th>\n",
       "      <th>HBV</th>\n",
       "      <th>total_perMB</th>\n",
       "      <th>Grade</th>\n",
       "      <th>Age</th>\n",
       "      <th>Gender</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CCAT104</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S_II</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Ever</td>\n",
       "      <td>2.14</td>\n",
       "      <td>G2</td>\n",
       "      <td>81</td>\n",
       "      <td>Female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCAT58</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S_I</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Ever</td>\n",
       "      <td>22.62</td>\n",
       "      <td>G2</td>\n",
       "      <td>49</td>\n",
       "      <td>Female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCAT46</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S_I</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Ever</td>\n",
       "      <td>9.34</td>\n",
       "      <td>G2</td>\n",
       "      <td>59</td>\n",
       "      <td>Female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCAT22</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S_I</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Ever</td>\n",
       "      <td>0.80</td>\n",
       "      <td>G2</td>\n",
       "      <td>66</td>\n",
       "      <td>Female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCAT123</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S_II</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Ever</td>\n",
       "      <td>5.28</td>\n",
       "      <td>G2</td>\n",
       "      <td>47</td>\n",
       "      <td>Female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCAT196</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S_III</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Ever</td>\n",
       "      <td>0.72</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>76</td>\n",
       "      <td>Male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCAT200</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S_III</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Ever</td>\n",
       "      <td>0.06</td>\n",
       "      <td>G2</td>\n",
       "      <td>53</td>\n",
       "      <td>Male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCAT207</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S_II</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Ever</td>\n",
       "      <td>1.18</td>\n",
       "      <td>G2</td>\n",
       "      <td>54</td>\n",
       "      <td>Male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCAT213</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S_III</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Never</td>\n",
       "      <td>0.64</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>58</td>\n",
       "      <td>Male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCAT99</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S_III</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Never</td>\n",
       "      <td>0.30</td>\n",
       "      <td>G2</td>\n",
       "      <td>52</td>\n",
       "      <td>Female</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>138 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Pathology Proteome_Subtype Cholelithiasis_1 Perineural_invasion  \\\n",
       "gene_symbol                                                                   \n",
       "CCAT104          iCCA             S_II               No                  No   \n",
       "CCAT58           iCCA              S_I               No                  No   \n",
       "CCAT46           iCCA              S_I               No                  No   \n",
       "CCAT22           iCCA              S_I               No                 Yes   \n",
       "CCAT123          iCCA             S_II               No                  No   \n",
       "...               ...              ...              ...                 ...   \n",
       "CCAT196          eCCA            S_III              Yes                 Yes   \n",
       "CCAT200          eCCA            S_III              Yes                 Yes   \n",
       "CCAT207          eCCA             S_II               No                 Yes   \n",
       "CCAT213          eCCA            S_III               No                 Yes   \n",
       "CCAT99           eCCA            S_III               No                 Yes   \n",
       "\n",
       "            Presence_of_fluke_infection    HBV  total_perMB Grade  Age  Gender  \n",
       "gene_symbol                                                                     \n",
       "CCAT104                              No   Ever         2.14    G2   81  Female  \n",
       "CCAT58                               No   Ever        22.62    G2   49  Female  \n",
       "CCAT46                               No   Ever         9.34    G2   59  Female  \n",
       "CCAT22                               No   Ever         0.80    G2   66  Female  \n",
       "CCAT123                              No   Ever         5.28    G2   47  Female  \n",
       "...                                 ...    ...          ...   ...  ...     ...  \n",
       "CCAT196                              No   Ever         0.72  G2_3   76    Male  \n",
       "CCAT200                              No   Ever         0.06    G2   53    Male  \n",
       "CCAT207                              No   Ever         1.18    G2   54    Male  \n",
       "CCAT213                              No  Never         0.64  G2_3   58    Male  \n",
       "CCAT99                               No  Never         0.30    G2   52  Female  \n",
       "\n",
       "[138 rows x 10 columns]"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "figure1b_clionical_ana = pd.read_csv(\"/Users/ranpeng/Desktop/CCA/Data/Fig1/Figure-1b/figure1_anatation.csv\",index_col = 0)\n",
    "figure1b_clionical_ana"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age</th>\n",
       "      <th>Date of surgery</th>\n",
       "      <th>Overall survival(month)</th>\n",
       "      <th>Overall survival(day)</th>\n",
       "      <th>Survival(1,dead;0,alive)</th>\n",
       "      <th>Anatomical classification</th>\n",
       "      <th>Histology</th>\n",
       "      <th>iCCA classification</th>\n",
       "      <th>Tumor (T) WES ID</th>\n",
       "      <th>...</th>\n",
       "      <th>TNM Stage</th>\n",
       "      <th>eccStage</th>\n",
       "      <th>iccStage</th>\n",
       "      <th>Grade</th>\n",
       "      <th>Proteome subtype</th>\n",
       "      <th>Phosphoproteome cluster</th>\n",
       "      <th>Immune cluster</th>\n",
       "      <th>Absolute tumor cellularity</th>\n",
       "      <th>Histology tumor cellularity</th>\n",
       "      <th>CCP_Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient No.</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CCA#1</th>\n",
       "      <td>Male</td>\n",
       "      <td>53</td>\n",
       "      <td>2012-01-30</td>\n",
       "      <td>44.666667</td>\n",
       "      <td>1340.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>III</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>S-III</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#2</th>\n",
       "      <td>Female</td>\n",
       "      <td>74</td>\n",
       "      <td>2012-02-01</td>\n",
       "      <td>40.533333</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>III</td>\n",
       "      <td>IIIB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-III</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Cluster_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cluster_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#3</th>\n",
       "      <td>Male</td>\n",
       "      <td>59</td>\n",
       "      <td>2012-02-01</td>\n",
       "      <td>44.600000</td>\n",
       "      <td>1338.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCAT132</td>\n",
       "      <td>...</td>\n",
       "      <td>III</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>S-III</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Cluster_6</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.69</td>\n",
       "      <td>Cluster_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#4</th>\n",
       "      <td>Female</td>\n",
       "      <td>71</td>\n",
       "      <td>2012-02-03</td>\n",
       "      <td>14.566667</td>\n",
       "      <td>437.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>III</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-III</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cluster_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#5</th>\n",
       "      <td>Male</td>\n",
       "      <td>60</td>\n",
       "      <td>2012-02-10</td>\n",
       "      <td>29.066667</td>\n",
       "      <td>872.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>II</td>\n",
       "      <td>II</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>S-II</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cluster_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#213</th>\n",
       "      <td>Male</td>\n",
       "      <td>75</td>\n",
       "      <td>2018-07-17</td>\n",
       "      <td>17.766667</td>\n",
       "      <td>533.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>I</td>\n",
       "      <td>I</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#214</th>\n",
       "      <td>Male</td>\n",
       "      <td>47</td>\n",
       "      <td>2018-07-23</td>\n",
       "      <td>17.433333</td>\n",
       "      <td>523.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>III</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>S-III</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#215</th>\n",
       "      <td>Female</td>\n",
       "      <td>62</td>\n",
       "      <td>2018-08-28</td>\n",
       "      <td>10.033333</td>\n",
       "      <td>301.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCAT219</td>\n",
       "      <td>...</td>\n",
       "      <td>III</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-III</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Cluster_6</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.67</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#216</th>\n",
       "      <td>Male</td>\n",
       "      <td>69</td>\n",
       "      <td>2018-10-30</td>\n",
       "      <td>1.066667</td>\n",
       "      <td>32.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCAT220</td>\n",
       "      <td>...</td>\n",
       "      <td>III</td>\n",
       "      <td>IIIB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-III</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Cluster_1</td>\n",
       "      <td>0.72</td>\n",
       "      <td>0.70</td>\n",
       "      <td>Cluster_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCA#217</th>\n",
       "      <td>Male</td>\n",
       "      <td>52</td>\n",
       "      <td>2018-12-03</td>\n",
       "      <td>9.066667</td>\n",
       "      <td>272.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>intrahepatic</td>\n",
       "      <td>iCCA</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>I</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IB</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-I</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cluster_1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>217 rows × 52 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gender  Age Date of surgery  Overall survival(month)  \\\n",
       "Patient No.                                                         \n",
       "CCA#1          Male   53      2012-01-30                44.666667   \n",
       "CCA#2        Female   74      2012-02-01                40.533333   \n",
       "CCA#3          Male   59      2012-02-01                44.600000   \n",
       "CCA#4        Female   71      2012-02-03                14.566667   \n",
       "CCA#5          Male   60      2012-02-10                29.066667   \n",
       "...             ...  ...             ...                      ...   \n",
       "CCA#213        Male   75      2018-07-17                17.766667   \n",
       "CCA#214        Male   47      2018-07-23                17.433333   \n",
       "CCA#215      Female   62      2018-08-28                10.033333   \n",
       "CCA#216        Male   69      2018-10-30                 1.066667   \n",
       "CCA#217        Male   52      2018-12-03                 9.066667   \n",
       "\n",
       "             Overall survival(day)  Survival(1,dead;0,alive)  \\\n",
       "Patient No.                                                    \n",
       "CCA#1                       1340.0                       1.0   \n",
       "CCA#2                       1216.0                       1.0   \n",
       "CCA#3                       1338.0                       1.0   \n",
       "CCA#4                        437.0                       1.0   \n",
       "CCA#5                        872.0                       1.0   \n",
       "...                            ...                       ...   \n",
       "CCA#213                      533.0                       1.0   \n",
       "CCA#214                      523.0                       1.0   \n",
       "CCA#215                      301.0                       1.0   \n",
       "CCA#216                       32.0                       1.0   \n",
       "CCA#217                      272.0                       1.0   \n",
       "\n",
       "            Anatomical classification Histology iCCA classification  \\\n",
       "Patient No.                                                           \n",
       "CCA#1                 hilar/perihilar      eCCA                 NaN   \n",
       "CCA#2                 hilar/perihilar      eCCA                 NaN   \n",
       "CCA#3                 hilar/perihilar      eCCA                 NaN   \n",
       "CCA#4                 hilar/perihilar      eCCA                 NaN   \n",
       "CCA#5                 hilar/perihilar      eCCA                 NaN   \n",
       "...                               ...       ...                 ...   \n",
       "CCA#213               hilar/perihilar      eCCA                 NaN   \n",
       "CCA#214               hilar/perihilar      eCCA                 NaN   \n",
       "CCA#215               hilar/perihilar      eCCA                 NaN   \n",
       "CCA#216               hilar/perihilar      eCCA                 NaN   \n",
       "CCA#217                  intrahepatic      iCCA             Unknown   \n",
       "\n",
       "            Tumor (T) WES ID  ... TNM Stage eccStage iccStage  Grade  \\\n",
       "Patient No.                   ...                                      \n",
       "CCA#1                    NaN  ...       III     IIIA      NaN   G2_3   \n",
       "CCA#2                    NaN  ...       III     IIIB      NaN     G2   \n",
       "CCA#3                CCAT132  ...       III     IIIC      NaN   G2_3   \n",
       "CCA#4                    NaN  ...       III     IIIC      NaN     G2   \n",
       "CCA#5                    NaN  ...        II       II      NaN   G2_3   \n",
       "...                      ...  ...       ...      ...      ...    ...   \n",
       "CCA#213                  NaN  ...         I        I      NaN     G2   \n",
       "CCA#214                  NaN  ...       III     IIIC      NaN   G2_3   \n",
       "CCA#215              CCAT219  ...       III     IIIA      NaN     G2   \n",
       "CCA#216              CCAT220  ...       III     IIIB      NaN     G2   \n",
       "CCA#217                  NaN  ...         I      NaN       IB     G2   \n",
       "\n",
       "             Proteome subtype Phosphoproteome cluster Immune cluster  \\\n",
       "Patient No.                                                            \n",
       "CCA#1                   S-III                     1.0            NaN   \n",
       "CCA#2                   S-III                     2.0      Cluster_6   \n",
       "CCA#3                   S-III                     1.0      Cluster_6   \n",
       "CCA#4                   S-III                     2.0            NaN   \n",
       "CCA#5                    S-II                     2.0            NaN   \n",
       "...                       ...                     ...            ...   \n",
       "CCA#213                 S-III                     NaN            NaN   \n",
       "CCA#214                 S-III                     2.0            NaN   \n",
       "CCA#215                 S-III                     2.0      Cluster_6   \n",
       "CCA#216                 S-III                     2.0      Cluster_1   \n",
       "CCA#217                   S-I                     2.0            NaN   \n",
       "\n",
       "            Absolute tumor cellularity Histology tumor cellularity  CCP_Class  \n",
       "Patient No.                                                                    \n",
       "CCA#1                              NaN                         NaN  Cluster_5  \n",
       "CCA#2                              NaN                         NaN  Cluster_6  \n",
       "CCA#3                             0.71                        0.69  Cluster_6  \n",
       "CCA#4                              NaN                         NaN  Cluster_6  \n",
       "CCA#5                              NaN                         NaN  Cluster_3  \n",
       "...                                ...                         ...        ...  \n",
       "CCA#213                            NaN                         NaN  Cluster_5  \n",
       "CCA#214                            NaN                         NaN  Cluster_5  \n",
       "CCA#215                           0.70                        0.67  Cluster_5  \n",
       "CCA#216                           0.72                        0.70  Cluster_6  \n",
       "CCA#217                            NaN                         NaN  Cluster_1  \n",
       "\n",
       "[217 rows x 52 columns]"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_metadata = pd.read_excel(\"/Users/ranpeng/Desktop/Integrated_multiomics_analysis_CCA/output_tables/1_Clinical_information.xlsx\", index_col = 0)\n",
    "clinical_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient No.</th>\n",
       "      <th>Tumor (T) proteomic ID</th>\n",
       "      <th>Survival(1,dead;0,alive)</th>\n",
       "      <th>Overall survival(month)</th>\n",
       "      <th>Overall survival(day)</th>\n",
       "      <th>Age</th>\n",
       "      <th>Tumor size (cm)</th>\n",
       "      <th>Preoperative AFP（ng/mL）</th>\n",
       "      <th>Preoperative CA19-9（U/ml）</th>\n",
       "      <th>Preoperative CEA（ng/ml）</th>\n",
       "      <th>...</th>\n",
       "      <th>Type II diabetes</th>\n",
       "      <th>Hypertension</th>\n",
       "      <th>Presence of fluke infection</th>\n",
       "      <th>Perineural invasion (PNI)</th>\n",
       "      <th>Lymphovascular invasion (LVI)</th>\n",
       "      <th>TNM Stage</th>\n",
       "      <th>Grade</th>\n",
       "      <th>Proteome subtype</th>\n",
       "      <th>Phosphoproteome cluster</th>\n",
       "      <th>Immune cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CCA#1</td>\n",
       "      <td>60477</td>\n",
       "      <td>1.0</td>\n",
       "      <td>44.666667</td>\n",
       "      <td>1340.0</td>\n",
       "      <td>53</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.6</td>\n",
       "      <td>32.3</td>\n",
       "      <td>2.60</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCA#2</td>\n",
       "      <td>60479</td>\n",
       "      <td>1.0</td>\n",
       "      <td>40.533333</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>74</td>\n",
       "      <td>3.5</td>\n",
       "      <td>1.6</td>\n",
       "      <td>10.7</td>\n",
       "      <td>2.42</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCA#3</td>\n",
       "      <td>60481</td>\n",
       "      <td>1.0</td>\n",
       "      <td>44.600000</td>\n",
       "      <td>1338.0</td>\n",
       "      <td>59</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.8</td>\n",
       "      <td>25.7</td>\n",
       "      <td>3.38</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCA#4</td>\n",
       "      <td>68986</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.566667</td>\n",
       "      <td>437.0</td>\n",
       "      <td>71</td>\n",
       "      <td>1.8</td>\n",
       "      <td>2.5</td>\n",
       "      <td>297.4</td>\n",
       "      <td>1.17</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CCA#5</td>\n",
       "      <td>68988</td>\n",
       "      <td>1.0</td>\n",
       "      <td>29.066667</td>\n",
       "      <td>872.0</td>\n",
       "      <td>60</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.5</td>\n",
       "      <td>311.2</td>\n",
       "      <td>4.30</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>CCA#213</td>\n",
       "      <td>71303</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17.766667</td>\n",
       "      <td>533.0</td>\n",
       "      <td>75</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>CCA#214</td>\n",
       "      <td>71311</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17.433333</td>\n",
       "      <td>523.0</td>\n",
       "      <td>47</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.7</td>\n",
       "      <td>151.5</td>\n",
       "      <td>5.30</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>CCA#215</td>\n",
       "      <td>71314</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10.033333</td>\n",
       "      <td>301.0</td>\n",
       "      <td>62</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.8</td>\n",
       "      <td>1140.0</td>\n",
       "      <td>5.80</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>CCA#216</td>\n",
       "      <td>71316</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.066667</td>\n",
       "      <td>32.0</td>\n",
       "      <td>69</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>CCA#217</td>\n",
       "      <td>71318</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.066667</td>\n",
       "      <td>272.0</td>\n",
       "      <td>52</td>\n",
       "      <td>9.0</td>\n",
       "      <td>5.6</td>\n",
       "      <td>40.4</td>\n",
       "      <td>0.50</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>217 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Patient No.  Tumor (T) proteomic ID  Survival(1,dead;0,alive)  \\\n",
       "0         CCA#1                   60477                       1.0   \n",
       "1         CCA#2                   60479                       1.0   \n",
       "2         CCA#3                   60481                       1.0   \n",
       "3         CCA#4                   68986                       1.0   \n",
       "4         CCA#5                   68988                       1.0   \n",
       "..          ...                     ...                       ...   \n",
       "212     CCA#213                   71303                       1.0   \n",
       "213     CCA#214                   71311                       1.0   \n",
       "214     CCA#215                   71314                       1.0   \n",
       "215     CCA#216                   71316                       1.0   \n",
       "216     CCA#217                   71318                       1.0   \n",
       "\n",
       "     Overall survival(month)  Overall survival(day)  Age  Tumor size (cm)  \\\n",
       "0                  44.666667                 1340.0   53              4.0   \n",
       "1                  40.533333                 1216.0   74              3.5   \n",
       "2                  44.600000                 1338.0   59              3.0   \n",
       "3                  14.566667                  437.0   71              1.8   \n",
       "4                  29.066667                  872.0   60              2.0   \n",
       "..                       ...                    ...  ...              ...   \n",
       "212                17.766667                  533.0   75              3.0   \n",
       "213                17.433333                  523.0   47              5.0   \n",
       "214                10.033333                  301.0   62              2.0   \n",
       "215                 1.066667                   32.0   69              4.0   \n",
       "216                 9.066667                  272.0   52              9.0   \n",
       "\n",
       "     Preoperative AFP（ng/mL）  Preoperative CA19-9（U/ml）  \\\n",
       "0                        5.6                       32.3   \n",
       "1                        1.6                       10.7   \n",
       "2                        2.8                       25.7   \n",
       "3                        2.5                      297.4   \n",
       "4                        4.5                      311.2   \n",
       "..                       ...                        ...   \n",
       "212                      NaN                        NaN   \n",
       "213                      5.7                      151.5   \n",
       "214                      9.8                     1140.0   \n",
       "215                      NaN                        NaN   \n",
       "216                      5.6                       40.4   \n",
       "\n",
       "     Preoperative CEA（ng/ml）  ...  Type II diabetes  Hypertension  \\\n",
       "0                       2.60  ...                 0             1   \n",
       "1                       2.42  ...                 0             0   \n",
       "2                       3.38  ...                 0             0   \n",
       "3                       1.17  ...                 0             0   \n",
       "4                       4.30  ...                 0             0   \n",
       "..                       ...  ...               ...           ...   \n",
       "212                      NaN  ...                 0             1   \n",
       "213                     5.30  ...                 0             0   \n",
       "214                     5.80  ...                 0             0   \n",
       "215                      NaN  ...                 1             0   \n",
       "216                     0.50  ...                 0             0   \n",
       "\n",
       "     Presence of fluke infection  Perineural invasion (PNI)  \\\n",
       "0                              0                        1.0   \n",
       "1                              0                        0.0   \n",
       "2                              0                        1.0   \n",
       "3                              0                        1.0   \n",
       "4                              0                        1.0   \n",
       "..                           ...                        ...   \n",
       "212                            0                        1.0   \n",
       "213                            0                        1.0   \n",
       "214                            0                        1.0   \n",
       "215                            0                        1.0   \n",
       "216                            0                        0.0   \n",
       "\n",
       "     Lymphovascular invasion (LVI)  TNM Stage  Grade  Proteome subtype  \\\n",
       "0                              1.0          3    3.0                 3   \n",
       "1                              1.0          3    2.0                 3   \n",
       "2                              0.0          3    3.0                 3   \n",
       "3                              0.0          3    2.0                 3   \n",
       "4                              NaN          2    3.0                 2   \n",
       "..                             ...        ...    ...               ...   \n",
       "212                            0.0          1    2.0                 3   \n",
       "213                            0.0          3    3.0                 3   \n",
       "214                            0.0          3    2.0                 3   \n",
       "215                            0.0          3    2.0                 3   \n",
       "216                            1.0          1    2.0                 1   \n",
       "\n",
       "     Phosphoproteome cluster  Immune cluster  \n",
       "0                        1.0             NaN  \n",
       "1                        2.0             6.0  \n",
       "2                        1.0             6.0  \n",
       "3                        2.0             NaN  \n",
       "4                        2.0             NaN  \n",
       "..                       ...             ...  \n",
       "212                      NaN             NaN  \n",
       "213                      2.0             NaN  \n",
       "214                      2.0             6.0  \n",
       "215                      2.0             1.0  \n",
       "216                      2.0             NaN  \n",
       "\n",
       "[217 rows x 38 columns]"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_information_cox_model = pd.read_excel(\"/Users/ranpeng/Desktop/Integrated_multiomics_analysis_CCA/data/Fig1/clinical_information_cox_model.xlsx\")#, index_col = 0)\n",
    "clinical_information_cox_model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>firT</th>\n",
       "      <th>CCP_Class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>60477</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>68990</td>\n",
       "      <td>Cluster_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>60415</td>\n",
       "      <td>Cluster_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>60426</td>\n",
       "      <td>Cluster_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>60472</td>\n",
       "      <td>Cluster_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>60459</td>\n",
       "      <td>Cluster_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>60442</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>69053</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>60428</td>\n",
       "      <td>Cluster_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>60412</td>\n",
       "      <td>Cluster_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>217 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      firT  CCP_Class\n",
       "0    60477  Cluster_5\n",
       "1    68990  Cluster_3\n",
       "2    60415  Cluster_2\n",
       "3    60426  Cluster_2\n",
       "4    60472  Cluster_2\n",
       "..     ...        ...\n",
       "212  60459  Cluster_1\n",
       "213  60442  Cluster_5\n",
       "214  69053  Cluster_5\n",
       "215  60428  Cluster_2\n",
       "216  60412  Cluster_4\n",
       "\n",
       "[217 rows x 2 columns]"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "immune_cluster = pd.read_clipboard()\n",
    "immune_cluster  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical_information_cox_model1 = pd.merge(clinical_information_cox_model, immune_cluster, left_on = 'Tumor (T) proteomic ID', right_on = 'firT', how = 'left')\n",
    "clinical_information_cox_model1.to_excel(\"/Users/ranpeng/Desktop/Integrated_multiomics_analysis_CCA/data/Fig1/clinical_information_cox_model1.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient No.</th>\n",
       "      <th>Tumor (T) proteomic ID</th>\n",
       "      <th>Survival(1,dead;0,alive)</th>\n",
       "      <th>Overall survival(month)</th>\n",
       "      <th>Overall survival(day)</th>\n",
       "      <th>Age</th>\n",
       "      <th>Tumor size (cm)</th>\n",
       "      <th>Preoperative AFP（ng/mL）</th>\n",
       "      <th>Preoperative CA19-9（U/ml）</th>\n",
       "      <th>Preoperative CEA（ng/ml）</th>\n",
       "      <th>...</th>\n",
       "      <th>Presence of fluke infection</th>\n",
       "      <th>Perineural invasion (PNI)</th>\n",
       "      <th>Lymphovascular invasion (LVI)</th>\n",
       "      <th>TNM Stage</th>\n",
       "      <th>Grade</th>\n",
       "      <th>Proteome subtype</th>\n",
       "      <th>Phosphoproteome cluster</th>\n",
       "      <th>Immune cluster</th>\n",
       "      <th>firT</th>\n",
       "      <th>CCP_Class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CCA#1</td>\n",
       "      <td>60477</td>\n",
       "      <td>1.0</td>\n",
       "      <td>44.666667</td>\n",
       "      <td>1340.0</td>\n",
       "      <td>53</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.6</td>\n",
       "      <td>32.3</td>\n",
       "      <td>2.60</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60477</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCA#2</td>\n",
       "      <td>60479</td>\n",
       "      <td>1.0</td>\n",
       "      <td>40.533333</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>74</td>\n",
       "      <td>3.5</td>\n",
       "      <td>1.6</td>\n",
       "      <td>10.7</td>\n",
       "      <td>2.42</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>60479</td>\n",
       "      <td>Cluster_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCA#3</td>\n",
       "      <td>60481</td>\n",
       "      <td>1.0</td>\n",
       "      <td>44.600000</td>\n",
       "      <td>1338.0</td>\n",
       "      <td>59</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.8</td>\n",
       "      <td>25.7</td>\n",
       "      <td>3.38</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>60481</td>\n",
       "      <td>Cluster_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCA#4</td>\n",
       "      <td>68986</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.566667</td>\n",
       "      <td>437.0</td>\n",
       "      <td>71</td>\n",
       "      <td>1.8</td>\n",
       "      <td>2.5</td>\n",
       "      <td>297.4</td>\n",
       "      <td>1.17</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68986</td>\n",
       "      <td>Cluster_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CCA#5</td>\n",
       "      <td>68988</td>\n",
       "      <td>1.0</td>\n",
       "      <td>29.066667</td>\n",
       "      <td>872.0</td>\n",
       "      <td>60</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.5</td>\n",
       "      <td>311.2</td>\n",
       "      <td>4.30</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68988</td>\n",
       "      <td>Cluster_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>CCA#213</td>\n",
       "      <td>71303</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17.766667</td>\n",
       "      <td>533.0</td>\n",
       "      <td>75</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71303</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>CCA#214</td>\n",
       "      <td>71311</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17.433333</td>\n",
       "      <td>523.0</td>\n",
       "      <td>47</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.7</td>\n",
       "      <td>151.5</td>\n",
       "      <td>5.30</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71311</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>CCA#215</td>\n",
       "      <td>71314</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10.033333</td>\n",
       "      <td>301.0</td>\n",
       "      <td>62</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.8</td>\n",
       "      <td>1140.0</td>\n",
       "      <td>5.80</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>71314</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>CCA#216</td>\n",
       "      <td>71316</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.066667</td>\n",
       "      <td>32.0</td>\n",
       "      <td>69</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>71316</td>\n",
       "      <td>Cluster_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>CCA#217</td>\n",
       "      <td>71318</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.066667</td>\n",
       "      <td>272.0</td>\n",
       "      <td>52</td>\n",
       "      <td>9.0</td>\n",
       "      <td>5.6</td>\n",
       "      <td>40.4</td>\n",
       "      <td>0.50</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>71318</td>\n",
       "      <td>Cluster_1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>217 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Patient No.  Tumor (T) proteomic ID  Survival(1,dead;0,alive)  \\\n",
       "0         CCA#1                   60477                       1.0   \n",
       "1         CCA#2                   60479                       1.0   \n",
       "2         CCA#3                   60481                       1.0   \n",
       "3         CCA#4                   68986                       1.0   \n",
       "4         CCA#5                   68988                       1.0   \n",
       "..          ...                     ...                       ...   \n",
       "212     CCA#213                   71303                       1.0   \n",
       "213     CCA#214                   71311                       1.0   \n",
       "214     CCA#215                   71314                       1.0   \n",
       "215     CCA#216                   71316                       1.0   \n",
       "216     CCA#217                   71318                       1.0   \n",
       "\n",
       "     Overall survival(month)  Overall survival(day)  Age  Tumor size (cm)  \\\n",
       "0                  44.666667                 1340.0   53              4.0   \n",
       "1                  40.533333                 1216.0   74              3.5   \n",
       "2                  44.600000                 1338.0   59              3.0   \n",
       "3                  14.566667                  437.0   71              1.8   \n",
       "4                  29.066667                  872.0   60              2.0   \n",
       "..                       ...                    ...  ...              ...   \n",
       "212                17.766667                  533.0   75              3.0   \n",
       "213                17.433333                  523.0   47              5.0   \n",
       "214                10.033333                  301.0   62              2.0   \n",
       "215                 1.066667                   32.0   69              4.0   \n",
       "216                 9.066667                  272.0   52              9.0   \n",
       "\n",
       "     Preoperative AFP（ng/mL）  Preoperative CA19-9（U/ml）  \\\n",
       "0                        5.6                       32.3   \n",
       "1                        1.6                       10.7   \n",
       "2                        2.8                       25.7   \n",
       "3                        2.5                      297.4   \n",
       "4                        4.5                      311.2   \n",
       "..                       ...                        ...   \n",
       "212                      NaN                        NaN   \n",
       "213                      5.7                      151.5   \n",
       "214                      9.8                     1140.0   \n",
       "215                      NaN                        NaN   \n",
       "216                      5.6                       40.4   \n",
       "\n",
       "     Preoperative CEA（ng/ml）  ...  Presence of fluke infection  \\\n",
       "0                       2.60  ...                            0   \n",
       "1                       2.42  ...                            0   \n",
       "2                       3.38  ...                            0   \n",
       "3                       1.17  ...                            0   \n",
       "4                       4.30  ...                            0   \n",
       "..                       ...  ...                          ...   \n",
       "212                      NaN  ...                            0   \n",
       "213                     5.30  ...                            0   \n",
       "214                     5.80  ...                            0   \n",
       "215                      NaN  ...                            0   \n",
       "216                     0.50  ...                            0   \n",
       "\n",
       "     Perineural invasion (PNI)  Lymphovascular invasion (LVI)  TNM Stage  \\\n",
       "0                          1.0                            1.0          3   \n",
       "1                          0.0                            1.0          3   \n",
       "2                          1.0                            0.0          3   \n",
       "3                          1.0                            0.0          3   \n",
       "4                          1.0                            NaN          2   \n",
       "..                         ...                            ...        ...   \n",
       "212                        1.0                            0.0          1   \n",
       "213                        1.0                            0.0          3   \n",
       "214                        1.0                            0.0          3   \n",
       "215                        1.0                            0.0          3   \n",
       "216                        0.0                            1.0          1   \n",
       "\n",
       "     Grade  Proteome subtype  Phosphoproteome cluster  Immune cluster   firT  \\\n",
       "0      3.0                 3                      1.0             NaN  60477   \n",
       "1      2.0                 3                      2.0             6.0  60479   \n",
       "2      3.0                 3                      1.0             6.0  60481   \n",
       "3      2.0                 3                      2.0             NaN  68986   \n",
       "4      3.0                 2                      2.0             NaN  68988   \n",
       "..     ...               ...                      ...             ...    ...   \n",
       "212    2.0                 3                      NaN             NaN  71303   \n",
       "213    3.0                 3                      2.0             NaN  71311   \n",
       "214    2.0                 3                      2.0             6.0  71314   \n",
       "215    2.0                 3                      2.0             1.0  71316   \n",
       "216    2.0                 1                      2.0             NaN  71318   \n",
       "\n",
       "     CCP_Class  \n",
       "0    Cluster_5  \n",
       "1    Cluster_6  \n",
       "2    Cluster_6  \n",
       "3    Cluster_6  \n",
       "4    Cluster_3  \n",
       "..         ...  \n",
       "212  Cluster_5  \n",
       "213  Cluster_5  \n",
       "214  Cluster_5  \n",
       "215  Cluster_6  \n",
       "216  Cluster_1  \n",
       "\n",
       "[217 rows x 40 columns]"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_information_cox_model1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gender', 'Age', 'Date of surgery', 'Overall survival(month)',\n",
       "       'Overall survival(day)', 'Survival(1,dead;0,alive)',\n",
       "       'Anatomical classification', 'Histology', 'iCCA classification',\n",
       "       'Tumor (T) WES ID', 'Adjacent tissue (NAT) WES ID',\n",
       "       'Tumor (T) RNA-seq ID', 'Adjacent tissue (NAT) RNA-seq ID',\n",
       "       'Tumor (T) proteomic ID', 'Adjacent tissue (NAT) proteomic ID',\n",
       "       'Tumor (T) phosphoproteomic ID',\n",
       "       'Adjacent tissue (NAT) phosphoproteomic ID', 'Tumor size (cm)',\n",
       "       'HBV infection status', 'HBsAg', 'HBcAb', 'Preoperative AFP（ng/mL）',\n",
       "       'Preoperative CA19-9（U/ml）', 'Preoperative CEA（ng/ml）',\n",
       "       'ALB, albumin, (g/L)', 'TB, total bilirubin (µmol/L)',\n",
       "       'ALT, aminoleucine transferase (U/L)',\n",
       "       'γ-GT, γ-glutamyltransferase (U/L)',\n",
       "       '\\nAST/GOT,Aspartate aminotransferase\\n（U/L）',\n",
       "       'TBA, total bile acid (µmol/L)', '\\nPLT, platelet counts\\n(10^9/L)',\n",
       "       'PT, prothrombin time (s) ', 'Cholelithiasis', 'Jaundice',\n",
       "       'Type II diabetes', 'Hypertension', 'Presence of fluke infection',\n",
       "       'Perineural invasion (PNI)', 'Lymphovascular invasion (LVI)', 'T', 'N',\n",
       "       'M', 'TNM Stage', 'eccStage', 'iccStage', 'Grade', 'Proteome subtype',\n",
       "       'Phosphoproteome cluster', 'Immune cluster',\n",
       "       'Absolute tumor cellularity', 'Histology tumor cellularity',\n",
       "       'CCP_Class'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_metadata.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Histology',\n",
       " 'Proteome subtype',\n",
       " 'Lymphovascular invasion (LVI)',\n",
       " 'Perineural invasion (PNI)',\n",
       " 'ALT, aminoleucine transferase (U/L)',\n",
       " 'TB, total bilirubin (µmol/L)',\n",
       " 'ALB, albumin, (g/L)',\n",
       " 'Preoperative CEA（ng/ml）',\n",
       " 'Preoperative CA19-9（U/ml）',\n",
       " 'Grade',\n",
       " 'TNM Stage',\n",
       " 'Age',\n",
       " 'Gender',\n",
       " 'gene_symbol']"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fig1b_anatation_list = ['gene_symbol','Gender','Age','TNM Stage','Grade','Preoperative CA19-9（U/ml）','Preoperative CEA（ng/ml）','ALB, albumin, (g/L)','TB, total bilirubin (µmol/L)',\n",
    "                        'ALT, aminoleucine transferase (U/L)','Perineural invasion (PNI)','Lymphovascular invasion (LVI)','Proteome subtype','Histology']\n",
    "\n",
    "fig1b_anatation_list.reverse()\n",
    "fig1b_anatation_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>Pathology</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CCAT104</td>\n",
       "      <td>iCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCAT58</td>\n",
       "      <td>iCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCAT46</td>\n",
       "      <td>iCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCAT22</td>\n",
       "      <td>iCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CCAT123</td>\n",
       "      <td>iCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>CCAT196</td>\n",
       "      <td>eCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>CCAT200</td>\n",
       "      <td>eCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>CCAT207</td>\n",
       "      <td>eCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>CCAT213</td>\n",
       "      <td>eCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>CCAT99</td>\n",
       "      <td>eCCA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>138 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol Pathology\n",
       "0       CCAT104      iCCA\n",
       "1        CCAT58      iCCA\n",
       "2        CCAT46      iCCA\n",
       "3        CCAT22      iCCA\n",
       "4       CCAT123      iCCA\n",
       "..          ...       ...\n",
       "133     CCAT196      eCCA\n",
       "134     CCAT200      eCCA\n",
       "135     CCAT207      eCCA\n",
       "136     CCAT213      eCCA\n",
       "137      CCAT99      eCCA\n",
       "\n",
       "[138 rows x 2 columns]"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fig1b_anatation = pd.read_clipboard()\n",
    "fig1b_anatation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_T</th>\n",
       "      <th>PT</th>\n",
       "      <th>HBsAg</th>\n",
       "      <th>HBcAb</th>\n",
       "      <th>HBV</th>\n",
       "      <th>GGT</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Tbil</th>\n",
       "      <th>CA199</th>\n",
       "      <th>Perineural_invasion</th>\n",
       "      <th>Lymphovascular_invasion</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CCAT10</td>\n",
       "      <td>Normal</td>\n",
       "      <td>negative</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCAT11</td>\n",
       "      <td>Normal</td>\n",
       "      <td>positive</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCAT12</td>\n",
       "      <td>Normal</td>\n",
       "      <td>negative</td>\n",
       "      <td>negative</td>\n",
       "      <td>Never</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCAT14</td>\n",
       "      <td>Normal</td>\n",
       "      <td>positive</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CCAT15</td>\n",
       "      <td>Normal</td>\n",
       "      <td>negative</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>CCAT212</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>negative</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>CCAT213</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>negative</td>\n",
       "      <td>positive</td>\n",
       "      <td>Never</td>\n",
       "      <td>NA_1</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>CCAT215</td>\n",
       "      <td>Normal</td>\n",
       "      <td>negative</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>CCAT219</td>\n",
       "      <td>NA_1</td>\n",
       "      <td>NA_1</td>\n",
       "      <td>NA_1</td>\n",
       "      <td>NA_1</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>CCAT220</td>\n",
       "      <td>NA_1</td>\n",
       "      <td>negative</td>\n",
       "      <td>negative</td>\n",
       "      <td>Never</td>\n",
       "      <td>NA_1</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>NA_1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_T        PT     HBsAg     HBcAb    HBV       GGT       ALT  \\\n",
       "0     CCAT10    Normal  negative  positive   Ever    Normal    Normal   \n",
       "1     CCAT11    Normal  positive  positive   Ever    Normal    Normal   \n",
       "2     CCAT12    Normal  negative  negative  Never  Elevated  Elevated   \n",
       "3     CCAT14    Normal  positive  positive   Ever    Normal  Elevated   \n",
       "4     CCAT15    Normal  negative  positive   Ever  Elevated    Normal   \n",
       "..       ...       ...       ...       ...    ...       ...       ...   \n",
       "134  CCAT212  Elevated  negative  positive   Ever  Elevated  Elevated   \n",
       "135  CCAT213  Elevated  negative  positive  Never      NA_1  Elevated   \n",
       "136  CCAT215    Normal  negative  positive   Ever  Elevated  Elevated   \n",
       "137  CCAT219      NA_1      NA_1      NA_1   NA_1  Elevated  Elevated   \n",
       "138  CCAT220      NA_1  negative  negative  Never      NA_1    Normal   \n",
       "\n",
       "         Tbil     CA199 Perineural_invasion Lymphovascular_invasion  \n",
       "0      Normal    Normal                  No                      No  \n",
       "1      Normal    Normal                  No                     Yes  \n",
       "2      Normal  Elevated                  No                      No  \n",
       "3      Normal    Normal                  No                      No  \n",
       "4      Normal  Elevated                 Yes                     Yes  \n",
       "..        ...       ...                 ...                     ...  \n",
       "134  Elevated  Elevated                 Yes                      No  \n",
       "135  Elevated  Elevated                 Yes                     Yes  \n",
       "136  Elevated  Elevated                 Yes                      No  \n",
       "137  Elevated  Elevated                 Yes                      No  \n",
       "138  Elevated      NA_1                 Yes                      No  \n",
       "\n",
       "[139 rows x 11 columns]"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_select = pd.read_clipboard()\n",
    "clinical_select"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>Pathology</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age</th>\n",
       "      <th>Date of surgery</th>\n",
       "      <th>Overall survival(month)</th>\n",
       "      <th>Overall survival(day)</th>\n",
       "      <th>Survival(1,dead;0,alive)</th>\n",
       "      <th>Anatomical classification</th>\n",
       "      <th>Histology</th>\n",
       "      <th>...</th>\n",
       "      <th>TNM Stage</th>\n",
       "      <th>eccStage</th>\n",
       "      <th>iccStage</th>\n",
       "      <th>Grade</th>\n",
       "      <th>Proteome subtype</th>\n",
       "      <th>Phosphoproteome cluster</th>\n",
       "      <th>Immune cluster</th>\n",
       "      <th>Absolute tumor cellularity</th>\n",
       "      <th>Histology tumor cellularity</th>\n",
       "      <th>CCP_Class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CCAT104</td>\n",
       "      <td>iCCA</td>\n",
       "      <td>Female</td>\n",
       "      <td>81</td>\n",
       "      <td>2015-09-23</td>\n",
       "      <td>20.066667</td>\n",
       "      <td>602.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>intrahepatic</td>\n",
       "      <td>iCCA</td>\n",
       "      <td>...</td>\n",
       "      <td>I</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IA</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-II</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.84</td>\n",
       "      <td>0.84</td>\n",
       "      <td>Cluster_4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCAT58</td>\n",
       "      <td>iCCA</td>\n",
       "      <td>Female</td>\n",
       "      <td>49</td>\n",
       "      <td>2015-03-19</td>\n",
       "      <td>48.433333</td>\n",
       "      <td>1453.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intrahepatic</td>\n",
       "      <td>iCCA</td>\n",
       "      <td>...</td>\n",
       "      <td>II</td>\n",
       "      <td>NaN</td>\n",
       "      <td>II</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-I</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.80</td>\n",
       "      <td>Cluster_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCAT46</td>\n",
       "      <td>iCCA</td>\n",
       "      <td>Female</td>\n",
       "      <td>59</td>\n",
       "      <td>2015-01-29</td>\n",
       "      <td>45.500000</td>\n",
       "      <td>1365.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intrahepatic</td>\n",
       "      <td>iCCA</td>\n",
       "      <td>...</td>\n",
       "      <td>I</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IA</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-I</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.88</td>\n",
       "      <td>Cluster_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCAT22</td>\n",
       "      <td>iCCA</td>\n",
       "      <td>Female</td>\n",
       "      <td>66</td>\n",
       "      <td>2014-11-03</td>\n",
       "      <td>56.466667</td>\n",
       "      <td>1694.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intrahepatic</td>\n",
       "      <td>iCCA</td>\n",
       "      <td>...</td>\n",
       "      <td>I</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IB</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-I</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Cluster_4</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.78</td>\n",
       "      <td>Cluster_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CCAT123</td>\n",
       "      <td>iCCA</td>\n",
       "      <td>Female</td>\n",
       "      <td>47</td>\n",
       "      <td>2015-12-21</td>\n",
       "      <td>22.133333</td>\n",
       "      <td>664.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>intrahepatic</td>\n",
       "      <td>iCCA</td>\n",
       "      <td>...</td>\n",
       "      <td>III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IIIB</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-II</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.80</td>\n",
       "      <td>Cluster_4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>CCAT196</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>Male</td>\n",
       "      <td>76</td>\n",
       "      <td>2017-05-10</td>\n",
       "      <td>22.500000</td>\n",
       "      <td>675.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>...</td>\n",
       "      <td>I</td>\n",
       "      <td>I</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>S-III</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Cluster_5</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.69</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>CCAT200</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>Male</td>\n",
       "      <td>53</td>\n",
       "      <td>2017-07-13</td>\n",
       "      <td>22.933333</td>\n",
       "      <td>688.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>...</td>\n",
       "      <td>II</td>\n",
       "      <td>II</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-III</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Cluster_3</td>\n",
       "      <td>0.73</td>\n",
       "      <td>0.72</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>CCAT207</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>Male</td>\n",
       "      <td>54</td>\n",
       "      <td>2017-12-25</td>\n",
       "      <td>14.366667</td>\n",
       "      <td>431.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>...</td>\n",
       "      <td>III</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-II</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.78</td>\n",
       "      <td>0.77</td>\n",
       "      <td>Cluster_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>CCAT213</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>Male</td>\n",
       "      <td>58</td>\n",
       "      <td>2018-03-19</td>\n",
       "      <td>9.600000</td>\n",
       "      <td>288.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>...</td>\n",
       "      <td>III</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>S-III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cluster_3</td>\n",
       "      <td>0.73</td>\n",
       "      <td>0.72</td>\n",
       "      <td>Cluster_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>CCAT99</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>Female</td>\n",
       "      <td>52</td>\n",
       "      <td>2015-06-18</td>\n",
       "      <td>38.700000</td>\n",
       "      <td>1161.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>hilar/perihilar</td>\n",
       "      <td>eCCA</td>\n",
       "      <td>...</td>\n",
       "      <td>III</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G2</td>\n",
       "      <td>S-III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cluster_2</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.68</td>\n",
       "      <td>Cluster_6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>138 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol Pathology  Gender  Age Date of surgery  \\\n",
       "0       CCAT104      iCCA  Female   81      2015-09-23   \n",
       "1        CCAT58      iCCA  Female   49      2015-03-19   \n",
       "2        CCAT46      iCCA  Female   59      2015-01-29   \n",
       "3        CCAT22      iCCA  Female   66      2014-11-03   \n",
       "4       CCAT123      iCCA  Female   47      2015-12-21   \n",
       "..          ...       ...     ...  ...             ...   \n",
       "133     CCAT196      eCCA    Male   76      2017-05-10   \n",
       "134     CCAT200      eCCA    Male   53      2017-07-13   \n",
       "135     CCAT207      eCCA    Male   54      2017-12-25   \n",
       "136     CCAT213      eCCA    Male   58      2018-03-19   \n",
       "137      CCAT99      eCCA  Female   52      2015-06-18   \n",
       "\n",
       "     Overall survival(month)  Overall survival(day)  Survival(1,dead;0,alive)  \\\n",
       "0                  20.066667                  602.0                       1.0   \n",
       "1                  48.433333                 1453.0                       0.0   \n",
       "2                  45.500000                 1365.0                       0.0   \n",
       "3                  56.466667                 1694.0                       0.0   \n",
       "4                  22.133333                  664.0                       1.0   \n",
       "..                       ...                    ...                       ...   \n",
       "133                22.500000                  675.0                       1.0   \n",
       "134                22.933333                  688.0                       1.0   \n",
       "135                14.366667                  431.0                       1.0   \n",
       "136                 9.600000                  288.0                       1.0   \n",
       "137                38.700000                 1161.0                       0.0   \n",
       "\n",
       "    Anatomical classification Histology  ... TNM Stage eccStage iccStage  \\\n",
       "0                intrahepatic      iCCA  ...         I      NaN       IA   \n",
       "1                intrahepatic      iCCA  ...        II      NaN       II   \n",
       "2                intrahepatic      iCCA  ...         I      NaN       IA   \n",
       "3                intrahepatic      iCCA  ...         I      NaN       IB   \n",
       "4                intrahepatic      iCCA  ...       III      NaN     IIIB   \n",
       "..                        ...       ...  ...       ...      ...      ...   \n",
       "133           hilar/perihilar      eCCA  ...         I        I      NaN   \n",
       "134           hilar/perihilar      eCCA  ...        II       II      NaN   \n",
       "135           hilar/perihilar      eCCA  ...       III     IIIC      NaN   \n",
       "136           hilar/perihilar      eCCA  ...       III     IIIC      NaN   \n",
       "137           hilar/perihilar      eCCA  ...       III     IIIC      NaN   \n",
       "\n",
       "    Grade Proteome subtype  Phosphoproteome cluster  Immune cluster  \\\n",
       "0      G2             S-II                      2.0             NaN   \n",
       "1      G2              S-I                      2.0             NaN   \n",
       "2      G2              S-I                      3.0             NaN   \n",
       "3      G2              S-I                      1.0       Cluster_4   \n",
       "4      G2             S-II                      3.0             NaN   \n",
       "..    ...              ...                      ...             ...   \n",
       "133  G2_3            S-III                      1.0       Cluster_5   \n",
       "134    G2            S-III                      1.0       Cluster_3   \n",
       "135    G2             S-II                      2.0             NaN   \n",
       "136  G2_3            S-III                      NaN       Cluster_3   \n",
       "137    G2            S-III                      NaN       Cluster_2   \n",
       "\n",
       "    Absolute tumor cellularity Histology tumor cellularity  CCP_Class  \n",
       "0                         0.84                        0.84  Cluster_4  \n",
       "1                         0.82                        0.80  Cluster_1  \n",
       "2                         0.88                        0.88  Cluster_1  \n",
       "3                         0.79                        0.78  Cluster_3  \n",
       "4                         0.82                        0.80  Cluster_4  \n",
       "..                         ...                         ...        ...  \n",
       "133                       0.71                        0.69  Cluster_5  \n",
       "134                       0.73                        0.72  Cluster_5  \n",
       "135                       0.78                        0.77  Cluster_3  \n",
       "136                       0.73                        0.72  Cluster_5  \n",
       "137                       0.71                        0.68  Cluster_6  \n",
       "\n",
       "[138 rows x 54 columns]"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fig1b_anatation = pd.merge(fig1b_anatation, clinical_metadata, left_on = 'gene_symbol', right_on = 'Tumor (T) WES ID', how = 'left')\n",
    "fig1b_anatation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Histology</th>\n",
       "      <th>Proteome subtype</th>\n",
       "      <th>Lymphovascular invasion (LVI)</th>\n",
       "      <th>Perineural invasion (PNI)</th>\n",
       "      <th>ALT, aminoleucine transferase (U/L)</th>\n",
       "      <th>TB, total bilirubin (µmol/L)</th>\n",
       "      <th>ALB, albumin, (g/L)</th>\n",
       "      <th>Preoperative CEA（ng/ml）</th>\n",
       "      <th>Preoperative CA19-9（U/ml）</th>\n",
       "      <th>Grade</th>\n",
       "      <th>TNM Stage</th>\n",
       "      <th>Age</th>\n",
       "      <th>Gender</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S-II</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>45</td>\n",
       "      <td>4</td>\n",
       "      <td>223</td>\n",
       "      <td>G2</td>\n",
       "      <td>I</td>\n",
       "      <td>81</td>\n",
       "      <td>Female</td>\n",
       "      <td>CCAT104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S-I</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>11.6</td>\n",
       "      <td>9.9</td>\n",
       "      <td>47.9</td>\n",
       "      <td>1.1</td>\n",
       "      <td>19.3</td>\n",
       "      <td>G2</td>\n",
       "      <td>II</td>\n",
       "      <td>49</td>\n",
       "      <td>Female</td>\n",
       "      <td>CCAT58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S-I</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>20.1</td>\n",
       "      <td>3.6</td>\n",
       "      <td>42.8</td>\n",
       "      <td>1.9</td>\n",
       "      <td>36.9</td>\n",
       "      <td>G2</td>\n",
       "      <td>I</td>\n",
       "      <td>59</td>\n",
       "      <td>Female</td>\n",
       "      <td>CCAT46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S-I</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>10.7</td>\n",
       "      <td>19.3</td>\n",
       "      <td>44.1</td>\n",
       "      <td>1.4</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>G2</td>\n",
       "      <td>I</td>\n",
       "      <td>66</td>\n",
       "      <td>Female</td>\n",
       "      <td>CCAT22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S-II</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>27</td>\n",
       "      <td>9.1</td>\n",
       "      <td>48</td>\n",
       "      <td>55.2</td>\n",
       "      <td>7.7</td>\n",
       "      <td>G2</td>\n",
       "      <td>III</td>\n",
       "      <td>47</td>\n",
       "      <td>Female</td>\n",
       "      <td>CCAT123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S-III</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>43</td>\n",
       "      <td>308.5</td>\n",
       "      <td>34</td>\n",
       "      <td>165</td>\n",
       "      <td>1.7</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>I</td>\n",
       "      <td>76</td>\n",
       "      <td>Male</td>\n",
       "      <td>CCAT196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S-III</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>53</td>\n",
       "      <td>50.4</td>\n",
       "      <td>41</td>\n",
       "      <td>3.3</td>\n",
       "      <td>108.1</td>\n",
       "      <td>G2</td>\n",
       "      <td>II</td>\n",
       "      <td>53</td>\n",
       "      <td>Male</td>\n",
       "      <td>CCAT200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S-II</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>131</td>\n",
       "      <td>167.2</td>\n",
       "      <td>38</td>\n",
       "      <td>4.6</td>\n",
       "      <td>8637</td>\n",
       "      <td>G2</td>\n",
       "      <td>III</td>\n",
       "      <td>54</td>\n",
       "      <td>Male</td>\n",
       "      <td>CCAT207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S-III</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>86</td>\n",
       "      <td>266.3</td>\n",
       "      <td>44</td>\n",
       "      <td>4.9</td>\n",
       "      <td>643.5</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>III</td>\n",
       "      <td>58</td>\n",
       "      <td>Male</td>\n",
       "      <td>CCAT213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S-III</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>551</td>\n",
       "      <td>61</td>\n",
       "      <td>45</td>\n",
       "      <td>2</td>\n",
       "      <td>5.7</td>\n",
       "      <td>G2</td>\n",
       "      <td>III</td>\n",
       "      <td>52</td>\n",
       "      <td>Female</td>\n",
       "      <td>CCAT99</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>138 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Histology Proteome subtype Lymphovascular invasion (LVI)  \\\n",
       "0        iCCA             S-II                            No   \n",
       "1        iCCA              S-I                           Yes   \n",
       "2        iCCA              S-I                            No   \n",
       "3        iCCA              S-I                           Yes   \n",
       "4        iCCA             S-II                            No   \n",
       "..        ...              ...                           ...   \n",
       "133      eCCA            S-III                            No   \n",
       "134      eCCA            S-III                            No   \n",
       "135      eCCA             S-II                            No   \n",
       "136      eCCA            S-III                           Yes   \n",
       "137      eCCA            S-III                            No   \n",
       "\n",
       "    Perineural invasion (PNI) ALT, aminoleucine transferase (U/L)  \\\n",
       "0                          No                                  24   \n",
       "1                          No                                11.6   \n",
       "2                          No                                20.1   \n",
       "3                         Yes                                10.7   \n",
       "4                          No                                  27   \n",
       "..                        ...                                 ...   \n",
       "133                       Yes                                  43   \n",
       "134                       Yes                                  53   \n",
       "135                       Yes                                 131   \n",
       "136                       Yes                                  86   \n",
       "137                       Yes                                 551   \n",
       "\n",
       "    TB, total bilirubin (µmol/L) ALB, albumin, (g/L) Preoperative CEA（ng/ml）  \\\n",
       "0                              8                  45                       4   \n",
       "1                            9.9                47.9                     1.1   \n",
       "2                            3.6                42.8                     1.9   \n",
       "3                           19.3                44.1                     1.4   \n",
       "4                            9.1                  48                    55.2   \n",
       "..                           ...                 ...                     ...   \n",
       "133                        308.5                  34                     165   \n",
       "134                         50.4                  41                     3.3   \n",
       "135                        167.2                  38                     4.6   \n",
       "136                        266.3                  44                     4.9   \n",
       "137                           61                  45                       2   \n",
       "\n",
       "    Preoperative CA19-9（U/ml） Grade TNM Stage  Age  Gender gene_symbol  \n",
       "0                         223    G2         I   81  Female     CCAT104  \n",
       "1                        19.3    G2        II   49  Female      CCAT58  \n",
       "2                        36.9    G2         I   59  Female      CCAT46  \n",
       "3                     Unknown    G2         I   66  Female      CCAT22  \n",
       "4                         7.7    G2       III   47  Female     CCAT123  \n",
       "..                        ...   ...       ...  ...     ...         ...  \n",
       "133                       1.7  G2_3         I   76    Male     CCAT196  \n",
       "134                     108.1    G2        II   53    Male     CCAT200  \n",
       "135                      8637    G2       III   54    Male     CCAT207  \n",
       "136                     643.5  G2_3       III   58    Male     CCAT213  \n",
       "137                       5.7    G2       III   52  Female      CCAT99  \n",
       "\n",
       "[138 rows x 14 columns]"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fig1b_anatation1 = fig1b_anatation[fig1b_anatation_list]\n",
    "fig1b_anatation1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Histology</th>\n",
       "      <th>Proteome subtype</th>\n",
       "      <th>Lymphovascular invasion (LVI)</th>\n",
       "      <th>Perineural invasion (PNI)</th>\n",
       "      <th>ALT, aminoleucine transferase (U/L)</th>\n",
       "      <th>TB, total bilirubin (µmol/L)</th>\n",
       "      <th>ALB, albumin, (g/L)</th>\n",
       "      <th>Preoperative CEA（ng/ml）</th>\n",
       "      <th>Preoperative CA19-9（U/ml）</th>\n",
       "      <th>Grade</th>\n",
       "      <th>...</th>\n",
       "      <th>PT</th>\n",
       "      <th>HBsAg</th>\n",
       "      <th>HBcAb</th>\n",
       "      <th>HBV</th>\n",
       "      <th>GGT</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Tbil</th>\n",
       "      <th>CA199</th>\n",
       "      <th>Perineural_invasion</th>\n",
       "      <th>Lymphovascular_invasion</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S-II</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>45</td>\n",
       "      <td>4</td>\n",
       "      <td>223</td>\n",
       "      <td>G2</td>\n",
       "      <td>...</td>\n",
       "      <td>Normal</td>\n",
       "      <td>negative</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>NA_1</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S-I</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>11.6</td>\n",
       "      <td>9.9</td>\n",
       "      <td>47.9</td>\n",
       "      <td>1.1</td>\n",
       "      <td>19.3</td>\n",
       "      <td>G2</td>\n",
       "      <td>...</td>\n",
       "      <td>Normal</td>\n",
       "      <td>negative</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S-I</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>20.1</td>\n",
       "      <td>3.6</td>\n",
       "      <td>42.8</td>\n",
       "      <td>1.9</td>\n",
       "      <td>36.9</td>\n",
       "      <td>G2</td>\n",
       "      <td>...</td>\n",
       "      <td>Normal</td>\n",
       "      <td>positive</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S-I</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>10.7</td>\n",
       "      <td>19.3</td>\n",
       "      <td>44.1</td>\n",
       "      <td>1.4</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>G2</td>\n",
       "      <td>...</td>\n",
       "      <td>Normal</td>\n",
       "      <td>positive</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NA_1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>iCCA</td>\n",
       "      <td>S-II</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>27</td>\n",
       "      <td>9.1</td>\n",
       "      <td>48</td>\n",
       "      <td>55.2</td>\n",
       "      <td>7.7</td>\n",
       "      <td>G2</td>\n",
       "      <td>...</td>\n",
       "      <td>Normal</td>\n",
       "      <td>negative</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S-III</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>43</td>\n",
       "      <td>308.5</td>\n",
       "      <td>34</td>\n",
       "      <td>165</td>\n",
       "      <td>1.7</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>...</td>\n",
       "      <td>Normal</td>\n",
       "      <td>negative</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S-III</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>53</td>\n",
       "      <td>50.4</td>\n",
       "      <td>41</td>\n",
       "      <td>3.3</td>\n",
       "      <td>108.1</td>\n",
       "      <td>G2</td>\n",
       "      <td>...</td>\n",
       "      <td>Normal</td>\n",
       "      <td>negative</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S-II</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>131</td>\n",
       "      <td>167.2</td>\n",
       "      <td>38</td>\n",
       "      <td>4.6</td>\n",
       "      <td>8637</td>\n",
       "      <td>G2</td>\n",
       "      <td>...</td>\n",
       "      <td>Below</td>\n",
       "      <td>negative</td>\n",
       "      <td>positive</td>\n",
       "      <td>Ever</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S-III</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>86</td>\n",
       "      <td>266.3</td>\n",
       "      <td>44</td>\n",
       "      <td>4.9</td>\n",
       "      <td>643.5</td>\n",
       "      <td>G2_3</td>\n",
       "      <td>...</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>negative</td>\n",
       "      <td>positive</td>\n",
       "      <td>Never</td>\n",
       "      <td>NA_1</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>eCCA</td>\n",
       "      <td>S-III</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>551</td>\n",
       "      <td>61</td>\n",
       "      <td>45</td>\n",
       "      <td>2</td>\n",
       "      <td>5.7</td>\n",
       "      <td>G2</td>\n",
       "      <td>...</td>\n",
       "      <td>Normal</td>\n",
       "      <td>negative</td>\n",
       "      <td>negative</td>\n",
       "      <td>Never</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Elevated</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>138 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Histology Proteome subtype Lymphovascular invasion (LVI)  \\\n",
       "0        iCCA             S-II                            No   \n",
       "1        iCCA              S-I                           Yes   \n",
       "2        iCCA              S-I                            No   \n",
       "3        iCCA              S-I                           Yes   \n",
       "4        iCCA             S-II                            No   \n",
       "..        ...              ...                           ...   \n",
       "133      eCCA            S-III                            No   \n",
       "134      eCCA            S-III                            No   \n",
       "135      eCCA             S-II                            No   \n",
       "136      eCCA            S-III                           Yes   \n",
       "137      eCCA            S-III                            No   \n",
       "\n",
       "    Perineural invasion (PNI) ALT, aminoleucine transferase (U/L)  \\\n",
       "0                          No                                  24   \n",
       "1                          No                                11.6   \n",
       "2                          No                                20.1   \n",
       "3                         Yes                                10.7   \n",
       "4                          No                                  27   \n",
       "..                        ...                                 ...   \n",
       "133                       Yes                                  43   \n",
       "134                       Yes                                  53   \n",
       "135                       Yes                                 131   \n",
       "136                       Yes                                  86   \n",
       "137                       Yes                                 551   \n",
       "\n",
       "    TB, total bilirubin (µmol/L) ALB, albumin, (g/L) Preoperative CEA（ng/ml）  \\\n",
       "0                              8                  45                       4   \n",
       "1                            9.9                47.9                     1.1   \n",
       "2                            3.6                42.8                     1.9   \n",
       "3                           19.3                44.1                     1.4   \n",
       "4                            9.1                  48                    55.2   \n",
       "..                           ...                 ...                     ...   \n",
       "133                        308.5                  34                     165   \n",
       "134                         50.4                  41                     3.3   \n",
       "135                        167.2                  38                     4.6   \n",
       "136                        266.3                  44                     4.9   \n",
       "137                           61                  45                       2   \n",
       "\n",
       "    Preoperative CA19-9（U/ml） Grade  ...        PT     HBsAg     HBcAb    HBV  \\\n",
       "0                         223    G2  ...    Normal  negative  positive   Ever   \n",
       "1                        19.3    G2  ...    Normal  negative  positive   Ever   \n",
       "2                        36.9    G2  ...    Normal  positive  positive   Ever   \n",
       "3                     Unknown    G2  ...    Normal  positive  positive   Ever   \n",
       "4                         7.7    G2  ...    Normal  negative  positive   Ever   \n",
       "..                        ...   ...  ...       ...       ...       ...    ...   \n",
       "133                       1.7  G2_3  ...    Normal  negative  positive   Ever   \n",
       "134                     108.1    G2  ...    Normal  negative  positive   Ever   \n",
       "135                      8637    G2  ...     Below  negative  positive   Ever   \n",
       "136                     643.5  G2_3  ...  Elevated  negative  positive  Never   \n",
       "137                       5.7    G2  ...    Normal  negative  negative  Never   \n",
       "\n",
       "          GGT       ALT      Tbil     CA199 Perineural_invasion  \\\n",
       "0        NA_1    Normal    Normal  Elevated                  No   \n",
       "1      Normal    Normal    Normal    Normal                  No   \n",
       "2      Normal    Normal    Normal    Normal                  No   \n",
       "3      Normal    Normal    Normal      NA_1                 Yes   \n",
       "4      Normal    Normal    Normal    Normal                  No   \n",
       "..        ...       ...       ...       ...                 ...   \n",
       "133  Elevated  Elevated  Elevated    Normal                 Yes   \n",
       "134  Elevated  Elevated  Elevated  Elevated                 Yes   \n",
       "135  Elevated  Elevated  Elevated  Elevated                 Yes   \n",
       "136      NA_1  Elevated  Elevated  Elevated                 Yes   \n",
       "137  Elevated  Elevated  Elevated    Normal                 Yes   \n",
       "\n",
       "    Lymphovascular_invasion  \n",
       "0                        No  \n",
       "1                       Yes  \n",
       "2                        No  \n",
       "3                       Yes  \n",
       "4                        No  \n",
       "..                      ...  \n",
       "133                      No  \n",
       "134                      No  \n",
       "135                      No  \n",
       "136                     Yes  \n",
       "137                      No  \n",
       "\n",
       "[138 rows x 25 columns]"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fig1b_anatation1 = pd.merge(fig1b_anatation1, clinical_select, left_on = 'gene_symbol', right_on = 'gene_T', how = 'left')\n",
    "fig1b_anatation1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig1b_anatation1.to_excel(paste0(outTableDir,\"fig1b_anatation1.xlsx\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Comparison of mutation frequencies with other cohorts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alteration</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>Mutation_Type</th>\n",
       "      <th>Variant allele frequency (VAF)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>111</td>\n",
       "      <td>1</td>\n",
       "      <td>36772804</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SH3D21</td>\n",
       "      <td>NM_001162530.1</td>\n",
       "      <td>c.263G&gt;A</td>\n",
       "      <td>p.Cys88Tyr</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>0.120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>111</td>\n",
       "      <td>1</td>\n",
       "      <td>87041251</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>CLCA4</td>\n",
       "      <td>NM_012128.3</td>\n",
       "      <td>c.1920A&gt;C</td>\n",
       "      <td>p.Thr640=</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>0.083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>111</td>\n",
       "      <td>1</td>\n",
       "      <td>100715341</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DBT</td>\n",
       "      <td>NM_001918.3</td>\n",
       "      <td>c.36G&gt;A</td>\n",
       "      <td>p.Arg12=</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>0.080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>111</td>\n",
       "      <td>1</td>\n",
       "      <td>115130458</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DENND2C</td>\n",
       "      <td>NM_001256404.1</td>\n",
       "      <td>c.2547G&gt;A</td>\n",
       "      <td>p.Arg849=</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>0.138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>111</td>\n",
       "      <td>1</td>\n",
       "      <td>151131427</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TNFAIP8L2</td>\n",
       "      <td>NM_024575.4</td>\n",
       "      <td>c.254C&gt;T</td>\n",
       "      <td>p.Ala85Val</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>0.165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17000</th>\n",
       "      <td>517</td>\n",
       "      <td>19</td>\n",
       "      <td>13227005</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>TRMT1</td>\n",
       "      <td>NM_001136035.2</td>\n",
       "      <td>c.209A&gt;G</td>\n",
       "      <td>p.Asn70Ser</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>0.313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17001</th>\n",
       "      <td>517</td>\n",
       "      <td>19</td>\n",
       "      <td>13339512</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>NM_001127221.1</td>\n",
       "      <td>c.5628+1G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>0.465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17002</th>\n",
       "      <td>517</td>\n",
       "      <td>19</td>\n",
       "      <td>51135688</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SYT3</td>\n",
       "      <td>NM_001160328.1</td>\n",
       "      <td>c.529G&gt;A</td>\n",
       "      <td>p.Ala177Thr</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>0.450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17003</th>\n",
       "      <td>517</td>\n",
       "      <td>19</td>\n",
       "      <td>52729226</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PPP2R1A</td>\n",
       "      <td>NM_014225.5</td>\n",
       "      <td>c.1762C&gt;G</td>\n",
       "      <td>p.Leu588Val</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>0.237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17004</th>\n",
       "      <td>517</td>\n",
       "      <td>X</td>\n",
       "      <td>105855945</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>CXorf57</td>\n",
       "      <td>NM_018015.5</td>\n",
       "      <td>c.635A&gt;G</td>\n",
       "      <td>p.Asn212Ser</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>0.628</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17005 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Sample_ID Chromosome   Position Reference Alteration       Gene  \\\n",
       "0            111          1   36772804         G          A     SH3D21   \n",
       "1            111          1   87041251         A          C      CLCA4   \n",
       "2            111          1  100715341         C          T        DBT   \n",
       "3            111          1  115130458         C          T    DENND2C   \n",
       "4            111          1  151131427         C          T  TNFAIP8L2   \n",
       "...          ...        ...        ...       ...        ...        ...   \n",
       "17000        517         19   13227005         T          C      TRMT1   \n",
       "17001        517         19   13339512         C          T    CACNA1A   \n",
       "17002        517         19   51135688         C          T       SYT3   \n",
       "17003        517         19   52729226         C          G    PPP2R1A   \n",
       "17004        517          X  105855945         A          G    CXorf57   \n",
       "\n",
       "        Transcript_ID        HGVSc        HGVSp         Mutation_Type  \\\n",
       "0      NM_001162530.1     c.263G>A   p.Cys88Tyr      missense_variant   \n",
       "1         NM_012128.3    c.1920A>C    p.Thr640=    synonymous_variant   \n",
       "2         NM_001918.3      c.36G>A     p.Arg12=    synonymous_variant   \n",
       "3      NM_001256404.1    c.2547G>A    p.Arg849=    synonymous_variant   \n",
       "4         NM_024575.4     c.254C>T   p.Ala85Val      missense_variant   \n",
       "...               ...          ...          ...                   ...   \n",
       "17000  NM_001136035.2     c.209A>G   p.Asn70Ser      missense_variant   \n",
       "17001  NM_001127221.1  c.5628+1G>A          NaN  splice_donor_variant   \n",
       "17002  NM_001160328.1     c.529G>A  p.Ala177Thr      missense_variant   \n",
       "17003     NM_014225.5    c.1762C>G  p.Leu588Val      missense_variant   \n",
       "17004     NM_018015.5     c.635A>G  p.Asn212Ser      missense_variant   \n",
       "\n",
       "       Variant allele frequency (VAF)  \n",
       "0                               0.120  \n",
       "1                               0.083  \n",
       "2                               0.080  \n",
       "3                               0.138  \n",
       "4                               0.165  \n",
       "...                               ...  \n",
       "17000                           0.313  \n",
       "17001                           0.465  \n",
       "17002                           0.450  \n",
       "17003                           0.237  \n",
       "17004                           0.628  \n",
       "\n",
       "[17005 rows x 11 columns]"
      ]
     },
     "execution_count": 247,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# from utils import read_specific_sheet\n",
    "Dong_vaf_data = pd.read_excel(\"/Users/ranpeng/Desktop/CCA/Revised_data/iCCA_data/1-s2.0-S1535610821006590-mmc2.xlsx\", sheet_name = 'S1B. WES mutations', header = 1)\n",
    "Dong_vaf_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Mutation_Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>111</td>\n",
       "      <td>SH3D21</td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>111</td>\n",
       "      <td>CLCA4</td>\n",
       "      <td>synonymous_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>111</td>\n",
       "      <td>DBT</td>\n",
       "      <td>synonymous_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>111</td>\n",
       "      <td>DENND2C</td>\n",
       "      <td>synonymous_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>111</td>\n",
       "      <td>TNFAIP8L2</td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17000</th>\n",
       "      <td>517</td>\n",
       "      <td>TRMT1</td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17001</th>\n",
       "      <td>517</td>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17002</th>\n",
       "      <td>517</td>\n",
       "      <td>SYT3</td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17003</th>\n",
       "      <td>517</td>\n",
       "      <td>PPP2R1A</td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17004</th>\n",
       "      <td>517</td>\n",
       "      <td>CXorf57</td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17005 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Sample_ID       Gene         Mutation_Type\n",
       "0            111     SH3D21      missense_variant\n",
       "1            111      CLCA4    synonymous_variant\n",
       "2            111        DBT    synonymous_variant\n",
       "3            111    DENND2C    synonymous_variant\n",
       "4            111  TNFAIP8L2      missense_variant\n",
       "...          ...        ...                   ...\n",
       "17000        517      TRMT1      missense_variant\n",
       "17001        517    CACNA1A  splice_donor_variant\n",
       "17002        517       SYT3      missense_variant\n",
       "17003        517    PPP2R1A      missense_variant\n",
       "17004        517    CXorf57      missense_variant\n",
       "\n",
       "[17005 rows x 3 columns]"
      ]
     },
     "execution_count": 248,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutation_list = ['Sample_ID','Gene', 'Mutation_Type']\n",
    "Dong_mutation_data = Dong_vaf_data[mutation_list]\n",
    "Dong_mutation_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['missense_variant', 'synonymous_variant',\n",
       "       'splice_acceptor_variant', 'splice_donor_variant', 'stop_gained',\n",
       "       'splice_region_variant', 'sequence_feature', 'frameshift_variant',\n",
       "       'stop_lost', 'start_lost', 'disruptive_inframe_deletion',\n",
       "       'disruptive_inframe_insertion', 'conservative_inframe_deletion',\n",
       "       'conservative_inframe_insertion'], dtype=object)"
      ]
     },
     "execution_count": 249,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Dong_mutation_data['Mutation_Type'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-250-3b9ce3ac4e3f>:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Dong_mutation_data['Mutation_Type'] = Dong_mutation_data['Mutation_Type'].map(mutation_dic)\n"
     ]
    }
   ],
   "source": [
    "mutation_dic = {'missense_variant':1, \n",
    "                'synonymous_variant':2,\n",
    "                'splice_acceptor_variant':3, \n",
    "                'splice_donor_variant':4, \n",
    "                'stop_gained':5,\n",
    "                'splice_region_variant':6, \n",
    "                'sequence_feature':7, \n",
    "                'frameshift_variant':8,\n",
    "                'stop_lost':9, \n",
    "                'start_lost':10, \n",
    "                'disruptive_inframe_deletion':11,\n",
    "                'disruptive_inframe_insertion':12, \n",
    "                'conservative_inframe_deletion':13,\n",
    "                'conservative_inframe_insertion':14}\n",
    "\n",
    "\n",
    "Dong_mutation_data['Mutation_Type'] = Dong_mutation_data['Mutation_Type'].map(mutation_dic)\n",
    "Dong_mutation_data_wide = Dong_mutation_data.pivot_table(\n",
    "    index=[\"Gene\"], columns='Sample_ID', values='Mutation_Type')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {},
   "outputs": [],
   "source": [
    "mutation_dic = {1:'missense_variant', \n",
    "                2:'synonymous_variant',\n",
    "                3:'splice_acceptor_variant', \n",
    "                4:'splice_donor_variant', \n",
    "                5:'stop_gained',\n",
    "                6:'splice_region_variant', \n",
    "                7:'sequence_feature', \n",
    "                8:'frameshift_variant',\n",
    "                9:'stop_lost', \n",
    "                10:'start_lost', \n",
    "                11:'disruptive_inframe_deletion',\n",
    "                12:'disruptive_inframe_insertion', \n",
    "                13:'conservative_inframe_deletion',\n",
    "                14:'conservative_inframe_insertion'}\n",
    "\n",
    "\n",
    "Dong_mutation_data_wide = Dong_mutation_data_wide.replace(mutation_dic)\n",
    "# Dong_mutation_data_wide.to_excel(paste0(outTableDir,\"Dong_mutation_data_wide.xlsx\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene\n",
       "A2M        0.007905\n",
       "A2ML1      0.011858\n",
       "A3GALT2    0.003953\n",
       "A4GALT     0.007905\n",
       "AAAS       0.007905\n",
       "             ...   \n",
       "ZXDB       0.011858\n",
       "ZXDC       0.007905\n",
       "ZYG11A     0.007905\n",
       "ZZEF1      0.007905\n",
       "ZZZ3       0.007905\n",
       "Name: Mutation_frequency, Length: 8950, dtype: float64"
      ]
     },
     "execution_count": 253,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Dong_mutation_data_wide.replace('synonymous_variant',np.nan, inplace=True)\n",
    "Dong_mutation_data_wide['Mutation_frequency'] = Dong_mutation_data_wide.apply(lambda x: x.count()/253, axis=1)\n",
    "Dong_mutation_data_wide['Mutation_count'] = Dong_mutation_data_wide.apply(lambda x: x.count(), axis=1)\n",
    "Dong_mutation_data_wide['Mutation_frequency']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save mutation data to /Users/ranpeng/Desktop/Integrated_multiomics_analysis_CCA/output_tables/CCA_clinical_triliner_tables.xlsx\n",
    "from utils import save_multisheet_excel\n",
    "append_file = r\"/Users/ranpeng/Desktop/Integrated_multiomics_analysis_CCA/output_tables/CCA_clinical_triliner_tables.xlsx\"\n",
    "save_multisheet_excel(append_file, input_file = Dong_mutation_data_wide, sheet_name=\"Dong_mutation_data2\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## gene mutation frequency correlation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GeneSymbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CDKN2A</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDKN2B</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NF1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PBRM1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KRAS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATM</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SMAD4</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDH1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IDH2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APC</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BAP1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SYNE1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARID1A</th>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: [CDKN2A, CDKN2B, FGFR2, NF1, PBRM1, KRAS, ATM, SMAD4, IDH1, IDH2, APC, BAP1, TP53, SYNE1, ARID1A]"
      ]
     },
     "execution_count": 245,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cohort_mut = pd.read_clipboard(index_col = 0)\n",
    "cohort_mut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [],
   "source": [
    "Fudan_Dong_cohort_iCCA_China_253_2022 = pd.merge(cohort_mut, Dong_mutation_data_wide[['Mutation_frequency']], left_index=True, right_index=True, how='left')\n",
    "Fudan_Dong_cohort_iCCA_China_253_2022.to_clipboard()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### correlation cohort mutation frequency"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_mutation_matrix = pd.read_clipboard(index_col = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n  \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Created with matplotlib (https://matplotlib.org/) -->\n<svg height=\"712.689687pt\" version=\"1.1\" viewBox=\"0 0 709.835 712.689687\" width=\"709.835pt\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n <metadata>\n  <rdf:RDF xmlns:cc=\"http://creativecommons.org/ns#\" xmlns:dc=\"http://purl.org/dc/elements/1.1/\" xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\">\n   <cc:Work>\n    <dc:type rdf:resource=\"http://purl.org/dc/dcmitype/StillImage\"/>\n    <dc:date>2022-03-15T15:44:07.518875</dc:date>\n    <dc:format>image/svg+xml</dc:format>\n    <dc:creator>\n     <cc:Agent>\n      <dc:title>Matplotlib v3.3.4, https://matplotlib.org/</dc:title>\n     </cc:Agent>\n    </dc:creator>\n   </cc:Work>\n  </rdf:RDF>\n </metadata>\n <defs>\n  <style type=\"text/css\">*{stroke-linecap:butt;stroke-linejoin:round;}</style>\n </defs>\n <g id=\"figure_1\">\n  <g id=\"patch_1\">\n   <path d=\"M 0 712.689687 \nL 709.835 712.689687 \nL 709.835 0 \nL 0 0 \nz\n\" style=\"fill:none;\"/>\n  </g>\n  <g id=\"axes_1\">\n   <g id=\"LineCollection_1\">\n    <path clip-path=\"url(#p8274827798)\" d=\"M 96.143 175.474545 \nL 87.190712 175.474545 \nL 87.190712 226.299116 \nL 96.143 226.299116 \n\" style=\"fill:none;stroke:#333333;stroke-width:0.5;\"/>\n    <path clip-path=\"url(#p8274827798)\" d=\"M 96.143 378.77283 \nL 88.664317 378.77283 \nL 88.664317 429.597402 \nL 96.143 429.597402 \n\" style=\"fill:none;stroke:#333333;stroke-width:0.5;\"/>\n    <path clip-path=\"url(#p8274827798)\" d=\"M 96.143 327.948259 \nL 62.418815 327.948259 \nL 62.418815 404.185116 \nL 88.664317 404.185116 \n\" style=\"fill:none;stroke:#333333;stroke-width:0.5;\"/>\n    <path clip-path=\"url(#p8274827798)\" d=\"M 96.143 277.123688 \nL 59.821333 277.123688 \nL 59.821333 366.066688 \nL 62.418815 366.066688 \n\" style=\"fill:none;stroke:#333333;stroke-width:0.5;\"/>\n    <path clip-path=\"url(#p8274827798)\" d=\"M 87.190712 200.88683 \nL 40.666473 200.88683 \nL 40.666473 321.595188 \nL 59.821333 321.595188 \n\" style=\"fill:none;stroke:#333333;stroke-width:0.5;\"/>\n    <path clip-path=\"url(#p8274827798)\" d=\"M 96.143 124.649973 \nL 11.435381 124.649973 \nL 11.435381 261.241009 \nL 40.666473 261.241009 \n\" style=\"fill:none;stroke:#333333;stroke-width:0.5;\"/>\n   </g>\n  </g>\n  <g id=\"axes_2\">\n   <g id=\"LineCollection_2\">\n    <path clip-path=\"url(#p7e472281ad)\" d=\"M 172.619857 98.997688 \nL 172.619857 90.0454 \nL 223.444429 90.0454 \nL 223.444429 98.997688 \n\" style=\"fill:none;stroke:#333333;stroke-width:0.5;\"/>\n    <path clip-path=\"url(#p7e472281ad)\" d=\"M 375.918143 98.997688 \nL 375.918143 91.519005 \nL 426.742714 91.519005 \nL 426.742714 98.997688 \n\" style=\"fill:none;stroke:#333333;stroke-width:0.5;\"/>\n    <path clip-path=\"url(#p7e472281ad)\" d=\"M 325.093571 98.997688 \nL 325.093571 65.273503 \nL 401.330429 65.273503 \nL 401.330429 91.519005 \n\" style=\"fill:none;stroke:#333333;stroke-width:0.5;\"/>\n    <path clip-path=\"url(#p7e472281ad)\" d=\"M 274.269 98.997688 \nL 274.269 62.676021 \nL 363.212 62.676021 \nL 363.212 65.273503 \n\" style=\"fill:none;stroke:#333333;stroke-width:0.5;\"/>\n    <path clip-path=\"url(#p7e472281ad)\" d=\"M 198.032143 90.0454 \nL 198.032143 43.521161 \nL 318.7405 43.521161 \nL 318.7405 62.676021 \n\" style=\"fill:none;stroke:#333333;stroke-width:0.5;\"/>\n    <path clip-path=\"url(#p7e472281ad)\" d=\"M 121.795286 98.997688 \nL 121.795286 14.290068 \nL 258.386321 14.290068 \nL 258.386321 43.521161 \n\" style=\"fill:none;stroke:#333333;stroke-width:0.5;\"/>\n   </g>\n  </g>\n  <g id=\"axes_3\">\n   <g id=\"patch_2\">\n    <path d=\"M 96.383 455.009687 \nL 452.155 455.009687 \nL 452.155 99.237688 \nL 96.383 99.237688 \nz\n\" style=\"fill:#ffffff;\"/>\n   </g>\n   <g id=\"QuadMesh_1\">\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 96.383 99.237688 \nL 147.207571 99.237688 \nL 147.207571 150.062259 \nL 96.383 150.062259 \nL 96.383 99.237688 \n\" style=\"fill:#fc0000;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 147.207571 99.237688 \nL 198.032143 99.237688 \nL 198.032143 150.062259 \nL 147.207571 150.062259 \nL 147.207571 99.237688 \n\" style=\"fill:#ffffff;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 198.032143 99.237688 \nL 248.856714 99.237688 \nL 248.856714 150.062259 \nL 198.032143 150.062259 \nL 198.032143 99.237688 \n\" style=\"fill:#ffffff;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 248.856714 99.237688 \nL 299.681286 99.237688 \nL 299.681286 150.062259 \nL 248.856714 150.062259 \nL 248.856714 99.237688 \n\" style=\"fill:#fe9292;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 299.681286 99.237688 \nL 350.505857 99.237688 \nL 350.505857 150.062259 \nL 299.681286 150.062259 \nL 299.681286 99.237688 \n\" style=\"fill:#fd4949;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 350.505857 99.237688 \nL 401.330429 99.237688 \nL 401.330429 150.062259 \nL 350.505857 150.062259 \nL 350.505857 99.237688 \n\" style=\"fill:#fe9292;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 401.330429 99.237688 \nL 452.155 99.237688 \nL 452.155 150.062259 \nL 401.330429 150.062259 \nL 401.330429 99.237688 \n\" style=\"fill:#fe9292;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 96.383 150.062259 \nL 147.207571 150.062259 \nL 147.207571 200.88683 \nL 96.383 200.88683 \nL 96.383 150.062259 \n\" style=\"fill:#ffffff;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 147.207571 150.062259 \nL 198.032143 150.062259 \nL 198.032143 200.88683 \nL 147.207571 200.88683 \nL 147.207571 150.062259 \n\" style=\"fill:#fc0000;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 198.032143 150.062259 \nL 248.856714 150.062259 \nL 248.856714 200.88683 \nL 198.032143 200.88683 \nL 198.032143 150.062259 \n\" style=\"fill:#fc0000;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 248.856714 150.062259 \nL 299.681286 150.062259 \nL 299.681286 200.88683 \nL 248.856714 200.88683 \nL 248.856714 150.062259 \n\" style=\"fill:#fd5b5b;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 299.681286 150.062259 \nL 350.505857 150.062259 \nL 350.505857 200.88683 \nL 299.681286 200.88683 \nL 299.681286 150.062259 \n\" style=\"fill:#fea4a4;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 350.505857 150.062259 \nL 401.330429 150.062259 \nL 401.330429 200.88683 \nL 350.505857 200.88683 \nL 350.505857 150.062259 \n\" style=\"fill:#fd4949;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 401.330429 150.062259 \nL 452.155 150.062259 \nL 452.155 200.88683 \nL 401.330429 200.88683 \nL 401.330429 150.062259 \n\" style=\"fill:#fd3737;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 96.383 200.88683 \nL 147.207571 200.88683 \nL 147.207571 251.711402 \nL 96.383 251.711402 \nL 96.383 200.88683 \n\" style=\"fill:#ffffff;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 147.207571 200.88683 \nL 198.032143 200.88683 \nL 198.032143 251.711402 \nL 147.207571 251.711402 \nL 147.207571 200.88683 \n\" style=\"fill:#fc0000;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 198.032143 200.88683 \nL 248.856714 200.88683 \nL 248.856714 251.711402 \nL 198.032143 251.711402 \nL 198.032143 200.88683 \n\" style=\"fill:#fc0000;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 248.856714 200.88683 \nL 299.681286 200.88683 \nL 299.681286 251.711402 \nL 248.856714 251.711402 \nL 248.856714 200.88683 \n\" style=\"fill:#fd4949;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 299.681286 200.88683 \nL 350.505857 200.88683 \nL 350.505857 251.711402 \nL 299.681286 251.711402 \nL 299.681286 200.88683 \n\" style=\"fill:#fe9292;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 350.505857 200.88683 \nL 401.330429 200.88683 \nL 401.330429 251.711402 \nL 350.505857 251.711402 \nL 350.505857 200.88683 \n\" style=\"fill:#fd4949;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 401.330429 200.88683 \nL 452.155 200.88683 \nL 452.155 251.711402 \nL 401.330429 251.711402 \nL 401.330429 200.88683 \n\" style=\"fill:#fd5b5b;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 96.383 251.711402 \nL 147.207571 251.711402 \nL 147.207571 302.535973 \nL 96.383 302.535973 \nL 96.383 251.711402 \n\" style=\"fill:#fe9292;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 147.207571 251.711402 \nL 198.032143 251.711402 \nL 198.032143 302.535973 \nL 147.207571 302.535973 \nL 147.207571 251.711402 \n\" style=\"fill:#fd5b5b;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 198.032143 251.711402 \nL 248.856714 251.711402 \nL 248.856714 302.535973 \nL 198.032143 302.535973 \nL 198.032143 251.711402 \n\" style=\"fill:#fd4949;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 248.856714 251.711402 \nL 299.681286 251.711402 \nL 299.681286 302.535973 \nL 248.856714 302.535973 \nL 248.856714 251.711402 \n\" style=\"fill:#fc0000;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 299.681286 251.711402 \nL 350.505857 251.711402 \nL 350.505857 302.535973 \nL 299.681286 302.535973 \nL 299.681286 251.711402 \n\" style=\"fill:#fd4949;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 350.505857 251.711402 \nL 401.330429 251.711402 \nL 401.330429 302.535973 \nL 350.505857 302.535973 \nL 350.505857 251.711402 \n\" style=\"fill:#fd5b5b;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 401.330429 251.711402 \nL 452.155 251.711402 \nL 452.155 302.535973 \nL 401.330429 302.535973 \nL 401.330429 251.711402 \n\" style=\"fill:#fd4949;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 96.383 302.535973 \nL 147.207571 302.535973 \nL 147.207571 353.360545 \nL 96.383 353.360545 \nL 96.383 302.535973 \n\" style=\"fill:#fd4949;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 147.207571 302.535973 \nL 198.032143 302.535973 \nL 198.032143 353.360545 \nL 147.207571 353.360545 \nL 147.207571 302.535973 \n\" style=\"fill:#fea4a4;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 198.032143 302.535973 \nL 248.856714 302.535973 \nL 248.856714 353.360545 \nL 198.032143 353.360545 \nL 198.032143 302.535973 \n\" style=\"fill:#fe9292;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 248.856714 302.535973 \nL 299.681286 302.535973 \nL 299.681286 353.360545 \nL 248.856714 353.360545 \nL 248.856714 302.535973 \n\" style=\"fill:#fd4949;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 299.681286 302.535973 \nL 350.505857 302.535973 \nL 350.505857 353.360545 \nL 299.681286 353.360545 \nL 299.681286 302.535973 \n\" style=\"fill:#fc0000;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 350.505857 302.535973 \nL 401.330429 302.535973 \nL 401.330429 353.360545 \nL 350.505857 353.360545 \nL 350.505857 302.535973 \n\" style=\"fill:#fc2424;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 401.330429 302.535973 \nL 452.155 302.535973 \nL 452.155 353.360545 \nL 401.330429 353.360545 \nL 401.330429 302.535973 \n\" style=\"fill:#fc2424;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 96.383 353.360545 \nL 147.207571 353.360545 \nL 147.207571 404.185116 \nL 96.383 404.185116 \nL 96.383 353.360545 \n\" style=\"fill:#fe9292;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 147.207571 353.360545 \nL 198.032143 353.360545 \nL 198.032143 404.185116 \nL 147.207571 404.185116 \nL 147.207571 353.360545 \n\" style=\"fill:#fd4949;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 198.032143 353.360545 \nL 248.856714 353.360545 \nL 248.856714 404.185116 \nL 198.032143 404.185116 \nL 198.032143 353.360545 \n\" style=\"fill:#fd4949;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 248.856714 353.360545 \nL 299.681286 353.360545 \nL 299.681286 404.185116 \nL 248.856714 404.185116 \nL 248.856714 353.360545 \n\" style=\"fill:#fd5b5b;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 299.681286 353.360545 \nL 350.505857 353.360545 \nL 350.505857 404.185116 \nL 299.681286 404.185116 \nL 299.681286 353.360545 \n\" style=\"fill:#fc2424;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 350.505857 353.360545 \nL 401.330429 353.360545 \nL 401.330429 404.185116 \nL 350.505857 404.185116 \nL 350.505857 353.360545 \n\" style=\"fill:#fc0000;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 401.330429 353.360545 \nL 452.155 353.360545 \nL 452.155 404.185116 \nL 401.330429 404.185116 \nL 401.330429 353.360545 \n\" style=\"fill:#fc0000;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 96.383 404.185116 \nL 147.207571 404.185116 \nL 147.207571 455.009687 \nL 96.383 455.009687 \nL 96.383 404.185116 \n\" style=\"fill:#fe9292;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 147.207571 404.185116 \nL 198.032143 404.185116 \nL 198.032143 455.009687 \nL 147.207571 455.009687 \nL 147.207571 404.185116 \n\" style=\"fill:#fd3737;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 198.032143 404.185116 \nL 248.856714 404.185116 \nL 248.856714 455.009687 \nL 198.032143 455.009687 \nL 198.032143 404.185116 \n\" style=\"fill:#fd5b5b;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 248.856714 404.185116 \nL 299.681286 404.185116 \nL 299.681286 455.009687 \nL 248.856714 455.009687 \nL 248.856714 404.185116 \n\" style=\"fill:#fd4949;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 299.681286 404.185116 \nL 350.505857 404.185116 \nL 350.505857 455.009687 \nL 299.681286 455.009687 \nL 299.681286 404.185116 \n\" style=\"fill:#fc2424;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 350.505857 404.185116 \nL 401.330429 404.185116 \nL 401.330429 455.009687 \nL 350.505857 455.009687 \nL 350.505857 404.185116 \n\" style=\"fill:#fc0000;stroke:#ffffff;stroke-width:1.5;\"/>\n    <path clip-path=\"url(#peec049f61d)\" d=\"M 401.330429 404.185116 \nL 452.155 404.185116 \nL 452.155 455.009687 \nL 401.330429 455.009687 \nL 401.330429 404.185116 \n\" style=\"fill:#fc0000;stroke:#ffffff;stroke-width:1.5;\"/>\n   </g>\n   <g id=\"matplotlib.axis_1\">\n    <g id=\"xtick_1\">\n     <g id=\"line2d_1\">\n      <defs>\n       <path d=\"M 0 0 \nL 0 3.5 \n\" id=\"maea3499d55\" style=\"stroke:#000000;stroke-width:0.8;\"/>\n      </defs>\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"121.795286\" xlink:href=\"#maea3499d55\" y=\"455.009687\"/>\n      </g>\n     </g>\n     <g id=\"text_1\">\n      <!-- TCGA(CCA_US_Can_2017) -->\n      <g transform=\"translate(124.900286 614.066562)rotate(-90)scale(0.12 -0.12)\">\n       <defs>\n        <path d=\"M 25.921875 0 \nL 25.921875 63.140625 \nL 2.34375 63.140625 \nL 2.34375 71.578125 \nL 59.078125 71.578125 \nL 59.078125 63.140625 \nL 35.40625 63.140625 \nL 35.40625 0 \nz\n\" id=\"ArialMT-84\"/>\n        <path d=\"M 58.796875 25.09375 \nL 68.265625 22.703125 \nQ 65.28125 11.03125 57.546875 4.90625 \nQ 49.8125 -1.21875 38.625 -1.21875 \nQ 27.046875 -1.21875 19.796875 3.484375 \nQ 12.546875 8.203125 8.765625 17.140625 \nQ 4.984375 26.078125 4.984375 36.328125 \nQ 4.984375 47.515625 9.25 55.828125 \nQ 13.53125 64.15625 21.40625 68.46875 \nQ 29.296875 72.796875 38.765625 72.796875 \nQ 49.515625 72.796875 56.828125 67.328125 \nQ 64.15625 61.859375 67.046875 51.953125 \nL 57.71875 49.75 \nQ 55.21875 57.5625 50.484375 61.125 \nQ 45.75 64.703125 38.578125 64.703125 \nQ 30.328125 64.703125 24.78125 60.734375 \nQ 19.234375 56.78125 16.984375 50.109375 \nQ 14.75 43.453125 14.75 36.375 \nQ 14.75 27.25 17.40625 20.4375 \nQ 20.0625 13.625 25.671875 10.25 \nQ 31.296875 6.890625 37.84375 6.890625 \nQ 45.796875 6.890625 51.3125 11.46875 \nQ 56.84375 16.0625 58.796875 25.09375 \nz\n\" id=\"ArialMT-67\"/>\n        <path d=\"M 41.21875 28.078125 \nL 41.21875 36.46875 \nL 71.53125 36.53125 \nL 71.53125 9.96875 \nQ 64.546875 4.390625 57.125 1.578125 \nQ 49.703125 -1.21875 41.890625 -1.21875 \nQ 31.34375 -1.21875 22.71875 3.296875 \nQ 14.109375 7.8125 9.71875 16.359375 \nQ 5.328125 24.90625 5.328125 35.453125 \nQ 5.328125 45.90625 9.6875 54.953125 \nQ 14.0625 64.015625 22.265625 68.40625 \nQ 30.46875 72.796875 41.15625 72.796875 \nQ 48.921875 72.796875 55.1875 70.28125 \nQ 61.46875 67.78125 65.03125 63.28125 \nQ 68.609375 58.796875 70.453125 51.5625 \nL 61.921875 49.21875 \nQ 60.296875 54.6875 57.90625 57.8125 \nQ 55.515625 60.9375 51.0625 62.8125 \nQ 46.625 64.703125 41.21875 64.703125 \nQ 34.71875 64.703125 29.984375 62.71875 \nQ 25.25 60.75 22.34375 57.515625 \nQ 19.4375 54.296875 17.828125 50.4375 \nQ 15.09375 43.796875 15.09375 36.03125 \nQ 15.09375 26.46875 18.390625 20.015625 \nQ 21.6875 13.578125 27.984375 10.453125 \nQ 34.28125 7.328125 41.359375 7.328125 \nQ 47.515625 7.328125 53.375 9.6875 \nQ 59.234375 12.0625 62.25 14.75 \nL 62.25 28.078125 \nz\n\" id=\"ArialMT-71\"/>\n        <path d=\"M -0.140625 0 \nL 27.34375 71.578125 \nL 37.546875 71.578125 \nL 66.84375 0 \nL 56.0625 0 \nL 47.703125 21.6875 \nL 17.78125 21.6875 \nL 9.90625 0 \nz\nM 20.515625 29.390625 \nL 44.78125 29.390625 \nL 37.3125 49.21875 \nQ 33.890625 58.25 32.234375 64.0625 \nQ 30.859375 57.171875 28.375 50.390625 \nz\n\" id=\"ArialMT-65\"/>\n        <path d=\"M 23.390625 -21.046875 \nQ 16.109375 -11.859375 11.078125 0.4375 \nQ 6.0625 12.75 6.0625 25.921875 \nQ 6.0625 37.546875 9.8125 48.1875 \nQ 14.203125 60.546875 23.390625 72.796875 \nL 29.6875 72.796875 \nQ 23.78125 62.640625 21.875 58.296875 \nQ 18.890625 51.5625 17.1875 44.234375 \nQ 15.09375 35.109375 15.09375 25.875 \nQ 15.09375 2.390625 29.6875 -21.046875 \nz\n\" id=\"ArialMT-40\"/>\n        <path d=\"M -1.515625 -19.875 \nL -1.515625 -13.53125 \nL 56.734375 -13.53125 \nL 56.734375 -19.875 \nz\n\" id=\"ArialMT-95\"/>\n        <path d=\"M 54.6875 71.578125 \nL 64.15625 71.578125 \nL 64.15625 30.21875 \nQ 64.15625 19.4375 61.71875 13.078125 \nQ 59.28125 6.734375 52.90625 2.75 \nQ 46.53125 -1.21875 36.1875 -1.21875 \nQ 26.125 -1.21875 19.71875 2.25 \nQ 13.328125 5.71875 10.59375 12.28125 \nQ 7.859375 18.84375 7.859375 30.21875 \nL 7.859375 71.578125 \nL 17.328125 71.578125 \nL 17.328125 30.28125 \nQ 17.328125 20.953125 19.0625 16.53125 \nQ 20.796875 12.109375 25.015625 9.71875 \nQ 29.25 7.328125 35.359375 7.328125 \nQ 45.796875 7.328125 50.234375 12.0625 \nQ 54.6875 16.796875 54.6875 30.28125 \nz\n\" id=\"ArialMT-85\"/>\n        <path d=\"M 4.5 23 \nL 13.421875 23.78125 \nQ 14.0625 18.40625 16.375 14.96875 \nQ 18.703125 11.53125 23.578125 9.40625 \nQ 28.46875 7.28125 34.578125 7.28125 \nQ 39.984375 7.28125 44.140625 8.890625 \nQ 48.296875 10.5 50.3125 13.296875 \nQ 52.34375 16.109375 52.34375 19.4375 \nQ 52.34375 22.796875 50.390625 25.3125 \nQ 48.4375 27.828125 43.953125 29.546875 \nQ 41.0625 30.671875 31.203125 33.03125 \nQ 21.34375 35.40625 17.390625 37.5 \nQ 12.25 40.1875 9.734375 44.15625 \nQ 7.234375 48.140625 7.234375 53.078125 \nQ 7.234375 58.5 10.296875 63.203125 \nQ 13.375 67.921875 19.28125 70.359375 \nQ 25.203125 72.796875 32.421875 72.796875 \nQ 40.375 72.796875 46.453125 70.234375 \nQ 52.546875 67.671875 55.8125 62.6875 \nQ 59.078125 57.71875 59.328125 51.421875 \nL 50.25 50.734375 \nQ 49.515625 57.515625 45.28125 60.984375 \nQ 41.0625 64.453125 32.8125 64.453125 \nQ 24.21875 64.453125 20.28125 61.296875 \nQ 16.359375 58.15625 16.359375 53.71875 \nQ 16.359375 49.859375 19.140625 47.359375 \nQ 21.875 44.875 33.421875 42.265625 \nQ 44.96875 39.65625 49.265625 37.703125 \nQ 55.515625 34.8125 58.484375 30.390625 \nQ 61.46875 25.984375 61.46875 20.21875 \nQ 61.46875 14.5 58.203125 9.4375 \nQ 54.9375 4.390625 48.796875 1.578125 \nQ 42.671875 -1.21875 35.015625 -1.21875 \nQ 25.296875 -1.21875 18.71875 1.609375 \nQ 12.15625 4.4375 8.421875 10.125 \nQ 4.6875 15.828125 4.5 23 \nz\n\" id=\"ArialMT-83\"/>\n        <path d=\"M 40.4375 6.390625 \nQ 35.546875 2.25 31.03125 0.53125 \nQ 26.515625 -1.171875 21.34375 -1.171875 \nQ 12.796875 -1.171875 8.203125 3 \nQ 3.609375 7.171875 3.609375 13.671875 \nQ 3.609375 17.484375 5.34375 20.625 \nQ 7.078125 23.78125 9.890625 25.6875 \nQ 12.703125 27.59375 16.21875 28.5625 \nQ 18.796875 29.25 24.03125 29.890625 \nQ 34.671875 31.15625 39.703125 32.90625 \nQ 39.75 34.71875 39.75 35.203125 \nQ 39.75 40.578125 37.25 42.78125 \nQ 33.890625 45.75 27.25 45.75 \nQ 21.046875 45.75 18.09375 43.578125 \nQ 15.140625 41.40625 13.71875 35.890625 \nL 5.125 37.0625 \nQ 6.296875 42.578125 8.984375 45.96875 \nQ 11.671875 49.359375 16.75 51.1875 \nQ 21.828125 53.03125 28.515625 53.03125 \nQ 35.15625 53.03125 39.296875 51.46875 \nQ 43.453125 49.90625 45.40625 47.53125 \nQ 47.359375 45.171875 48.140625 41.546875 \nQ 48.578125 39.3125 48.578125 33.453125 \nL 48.578125 21.734375 \nQ 48.578125 9.46875 49.140625 6.21875 \nQ 49.703125 2.984375 51.375 0 \nL 42.1875 0 \nQ 40.828125 2.734375 40.4375 6.390625 \nz\nM 39.703125 26.03125 \nQ 34.90625 24.078125 25.34375 22.703125 \nQ 19.921875 21.921875 17.671875 20.9375 \nQ 15.4375 19.96875 14.203125 18.09375 \nQ 12.984375 16.21875 12.984375 13.921875 \nQ 12.984375 10.40625 15.640625 8.0625 \nQ 18.3125 5.71875 23.4375 5.71875 \nQ 28.515625 5.71875 32.46875 7.9375 \nQ 36.421875 10.15625 38.28125 14.015625 \nQ 39.703125 17 39.703125 22.796875 \nz\n\" id=\"ArialMT-97\"/>\n        <path d=\"M 6.59375 0 \nL 6.59375 51.859375 \nL 14.5 51.859375 \nL 14.5 44.484375 \nQ 20.21875 53.03125 31 53.03125 \nQ 35.6875 53.03125 39.625 51.34375 \nQ 43.5625 49.65625 45.515625 46.921875 \nQ 47.46875 44.1875 48.25 40.4375 \nQ 48.734375 37.984375 48.734375 31.890625 \nL 48.734375 0 \nL 39.9375 0 \nL 39.9375 31.546875 \nQ 39.9375 36.921875 38.90625 39.578125 \nQ 37.890625 42.234375 35.28125 43.8125 \nQ 32.671875 45.40625 29.15625 45.40625 \nQ 23.53125 45.40625 19.453125 41.84375 \nQ 15.375 38.28125 15.375 28.328125 \nL 15.375 0 \nz\n\" id=\"ArialMT-110\"/>\n        <path d=\"M 50.34375 8.453125 \nL 50.34375 0 \nL 3.03125 0 \nQ 2.9375 3.171875 4.046875 6.109375 \nQ 5.859375 10.9375 9.828125 15.625 \nQ 13.8125 20.3125 21.34375 26.46875 \nQ 33.015625 36.03125 37.109375 41.625 \nQ 41.21875 47.21875 41.21875 52.203125 \nQ 41.21875 57.421875 37.46875 61 \nQ 33.734375 64.59375 27.734375 64.59375 \nQ 21.390625 64.59375 17.578125 60.78125 \nQ 13.765625 56.984375 13.71875 50.25 \nL 4.6875 51.171875 \nQ 5.609375 61.28125 11.65625 66.578125 \nQ 17.71875 71.875 27.9375 71.875 \nQ 38.234375 71.875 44.234375 66.15625 \nQ 50.25 60.453125 50.25 52 \nQ 50.25 47.703125 48.484375 43.546875 \nQ 46.734375 39.40625 42.65625 34.8125 \nQ 38.578125 30.21875 29.109375 22.21875 \nQ 21.1875 15.578125 18.9375 13.203125 \nQ 16.703125 10.84375 15.234375 8.453125 \nz\n\" id=\"ArialMT-50\"/>\n        <path d=\"M 4.15625 35.296875 \nQ 4.15625 48 6.765625 55.734375 \nQ 9.375 63.484375 14.515625 67.671875 \nQ 19.671875 71.875 27.484375 71.875 \nQ 33.25 71.875 37.59375 69.546875 \nQ 41.9375 67.234375 44.765625 62.859375 \nQ 47.609375 58.5 49.21875 52.21875 \nQ 50.828125 45.953125 50.828125 35.296875 \nQ 50.828125 22.703125 48.234375 14.96875 \nQ 45.65625 7.234375 40.5 3 \nQ 35.359375 -1.21875 27.484375 -1.21875 \nQ 17.140625 -1.21875 11.234375 6.203125 \nQ 4.15625 15.140625 4.15625 35.296875 \nz\nM 13.1875 35.296875 \nQ 13.1875 17.671875 17.3125 11.828125 \nQ 21.4375 6 27.484375 6 \nQ 33.546875 6 37.671875 11.859375 \nQ 41.796875 17.71875 41.796875 35.296875 \nQ 41.796875 52.984375 37.671875 58.78125 \nQ 33.546875 64.59375 27.390625 64.59375 \nQ 21.34375 64.59375 17.71875 59.46875 \nQ 13.1875 52.9375 13.1875 35.296875 \nz\n\" id=\"ArialMT-48\"/>\n        <path d=\"M 37.25 0 \nL 28.46875 0 \nL 28.46875 56 \nQ 25.296875 52.984375 20.140625 49.953125 \nQ 14.984375 46.921875 10.890625 45.40625 \nL 10.890625 53.90625 \nQ 18.265625 57.375 23.78125 62.296875 \nQ 29.296875 67.234375 31.59375 71.875 \nL 37.25 71.875 \nz\n\" id=\"ArialMT-49\"/>\n        <path d=\"M 4.734375 62.203125 \nL 4.734375 70.65625 \nL 51.078125 70.65625 \nL 51.078125 63.8125 \nQ 44.234375 56.546875 37.515625 44.484375 \nQ 30.8125 32.421875 27.15625 19.671875 \nQ 24.515625 10.6875 23.78125 0 \nL 14.75 0 \nQ 14.890625 8.453125 18.0625 20.40625 \nQ 21.234375 32.375 27.171875 43.484375 \nQ 33.109375 54.59375 39.796875 62.203125 \nz\n\" id=\"ArialMT-55\"/>\n        <path d=\"M 12.359375 -21.046875 \nL 6.0625 -21.046875 \nQ 20.65625 2.390625 20.65625 25.875 \nQ 20.65625 35.0625 18.5625 44.09375 \nQ 16.890625 51.421875 13.921875 58.15625 \nQ 12.015625 62.546875 6.0625 72.796875 \nL 12.359375 72.796875 \nQ 21.53125 60.546875 25.921875 48.1875 \nQ 29.6875 37.546875 29.6875 25.921875 \nQ 29.6875 12.75 24.625 0.4375 \nQ 19.578125 -11.859375 12.359375 -21.046875 \nz\n\" id=\"ArialMT-41\"/>\n       </defs>\n       <use xlink:href=\"#ArialMT-84\"/>\n       <use x=\"61.083984\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"133.300781\" xlink:href=\"#ArialMT-71\"/>\n       <use x=\"211.083984\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"277.783203\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"311.083984\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"383.300781\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"455.517578\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"522.216797\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"577.832031\" xlink:href=\"#ArialMT-85\"/>\n       <use x=\"650.048828\" xlink:href=\"#ArialMT-83\"/>\n       <use x=\"716.748047\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"772.363281\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"844.580078\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"900.195312\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"955.810547\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1011.425781\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1067.041016\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1122.65625\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"1178.271484\" xlink:href=\"#ArialMT-55\"/>\n       <use x=\"1233.886719\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"xtick_2\">\n     <g id=\"line2d_2\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"172.619857\" xlink:href=\"#maea3499d55\" y=\"455.009687\"/>\n      </g>\n     </g>\n     <g id=\"text_2\">\n      <!-- ICGC(BTC_Japan_260_2015) -->\n      <g transform=\"translate(175.724857 623.424688)rotate(-90)scale(0.12 -0.12)\">\n       <defs>\n        <path d=\"M 9.328125 0 \nL 9.328125 71.578125 \nL 18.796875 71.578125 \nL 18.796875 0 \nz\n\" id=\"ArialMT-73\"/>\n        <path d=\"M 7.328125 0 \nL 7.328125 71.578125 \nL 34.1875 71.578125 \nQ 42.390625 71.578125 47.34375 69.40625 \nQ 52.296875 67.234375 55.09375 62.71875 \nQ 57.90625 58.203125 57.90625 53.265625 \nQ 57.90625 48.6875 55.421875 44.625 \nQ 52.9375 40.578125 47.90625 38.09375 \nQ 54.390625 36.1875 57.875 31.59375 \nQ 61.375 27 61.375 20.75 \nQ 61.375 15.71875 59.25 11.390625 \nQ 57.125 7.078125 54 4.734375 \nQ 50.875 2.390625 46.15625 1.1875 \nQ 41.453125 0 34.625 0 \nz\nM 16.796875 41.5 \nL 32.28125 41.5 \nQ 38.578125 41.5 41.3125 42.328125 \nQ 44.921875 43.40625 46.75 45.890625 \nQ 48.578125 48.390625 48.578125 52.15625 \nQ 48.578125 55.71875 46.875 58.421875 \nQ 45.171875 61.140625 41.984375 62.140625 \nQ 38.8125 63.140625 31.109375 63.140625 \nL 16.796875 63.140625 \nz\nM 16.796875 8.453125 \nL 34.625 8.453125 \nQ 39.203125 8.453125 41.0625 8.796875 \nQ 44.34375 9.375 46.53125 10.734375 \nQ 48.734375 12.109375 50.140625 14.71875 \nQ 51.5625 17.328125 51.5625 20.75 \nQ 51.5625 24.75 49.515625 27.703125 \nQ 47.46875 30.671875 43.828125 31.859375 \nQ 40.1875 33.0625 33.34375 33.0625 \nL 16.796875 33.0625 \nz\n\" id=\"ArialMT-66\"/>\n        <path d=\"M 2.875 20.3125 \nL 11.421875 21.484375 \nQ 11.765625 13.28125 14.5 10.25 \nQ 17.234375 7.234375 22.078125 7.234375 \nQ 25.640625 7.234375 28.21875 8.859375 \nQ 30.8125 10.5 31.78125 13.296875 \nQ 32.765625 16.109375 32.765625 22.265625 \nL 32.765625 71.578125 \nL 42.234375 71.578125 \nL 42.234375 22.796875 \nQ 42.234375 13.8125 40.0625 8.875 \nQ 37.890625 3.953125 33.171875 1.359375 \nQ 28.46875 -1.21875 22.125 -1.21875 \nQ 12.703125 -1.21875 7.6875 4.203125 \nQ 2.6875 9.625 2.875 20.3125 \nz\n\" id=\"ArialMT-74\"/>\n        <path d=\"M 6.59375 -19.875 \nL 6.59375 51.859375 \nL 14.59375 51.859375 \nL 14.59375 45.125 \nQ 17.4375 49.078125 21 51.046875 \nQ 24.5625 53.03125 29.640625 53.03125 \nQ 36.28125 53.03125 41.359375 49.609375 \nQ 46.4375 46.1875 49.015625 39.953125 \nQ 51.609375 33.734375 51.609375 26.3125 \nQ 51.609375 18.359375 48.75 11.984375 \nQ 45.90625 5.609375 40.453125 2.21875 \nQ 35.015625 -1.171875 29 -1.171875 \nQ 24.609375 -1.171875 21.109375 0.6875 \nQ 17.625 2.546875 15.375 5.375 \nL 15.375 -19.875 \nz\nM 14.546875 25.640625 \nQ 14.546875 15.625 18.59375 10.84375 \nQ 22.65625 6.0625 28.421875 6.0625 \nQ 34.28125 6.0625 38.453125 11.015625 \nQ 42.625 15.96875 42.625 26.375 \nQ 42.625 36.28125 38.546875 41.203125 \nQ 34.46875 46.140625 28.8125 46.140625 \nQ 23.1875 46.140625 18.859375 40.890625 \nQ 14.546875 35.640625 14.546875 25.640625 \nz\n\" id=\"ArialMT-112\"/>\n        <path d=\"M 49.75 54.046875 \nL 41.015625 53.375 \nQ 39.84375 58.546875 37.703125 60.890625 \nQ 34.125 64.65625 28.90625 64.65625 \nQ 24.703125 64.65625 21.53125 62.3125 \nQ 17.390625 59.28125 14.984375 53.46875 \nQ 12.59375 47.65625 12.5 36.921875 \nQ 15.671875 41.75 20.265625 44.09375 \nQ 24.859375 46.4375 29.890625 46.4375 \nQ 38.671875 46.4375 44.84375 39.96875 \nQ 51.03125 33.5 51.03125 23.25 \nQ 51.03125 16.5 48.125 10.71875 \nQ 45.21875 4.9375 40.140625 1.859375 \nQ 35.0625 -1.21875 28.609375 -1.21875 \nQ 17.625 -1.21875 10.6875 6.859375 \nQ 3.765625 14.9375 3.765625 33.5 \nQ 3.765625 54.25 11.421875 63.671875 \nQ 18.109375 71.875 29.4375 71.875 \nQ 37.890625 71.875 43.28125 67.140625 \nQ 48.6875 62.40625 49.75 54.046875 \nz\nM 13.875 23.1875 \nQ 13.875 18.65625 15.796875 14.5 \nQ 17.71875 10.359375 21.1875 8.171875 \nQ 24.65625 6 28.46875 6 \nQ 34.03125 6 38.03125 10.484375 \nQ 42.046875 14.984375 42.046875 22.703125 \nQ 42.046875 30.125 38.078125 34.390625 \nQ 34.125 38.671875 28.125 38.671875 \nQ 22.171875 38.671875 18.015625 34.390625 \nQ 13.875 30.125 13.875 23.1875 \nz\n\" id=\"ArialMT-54\"/>\n        <path d=\"M 4.15625 18.75 \nL 13.375 19.53125 \nQ 14.40625 12.796875 18.140625 9.390625 \nQ 21.875 6 27.15625 6 \nQ 33.5 6 37.890625 10.78125 \nQ 42.28125 15.578125 42.28125 23.484375 \nQ 42.28125 31 38.0625 35.34375 \nQ 33.84375 39.703125 27 39.703125 \nQ 22.75 39.703125 19.328125 37.765625 \nQ 15.921875 35.84375 13.96875 32.765625 \nL 5.71875 33.84375 \nL 12.640625 70.609375 \nL 48.25 70.609375 \nL 48.25 62.203125 \nL 19.671875 62.203125 \nL 15.828125 42.96875 \nQ 22.265625 47.46875 29.34375 47.46875 \nQ 38.71875 47.46875 45.15625 40.96875 \nQ 51.609375 34.46875 51.609375 24.265625 \nQ 51.609375 14.546875 45.953125 7.46875 \nQ 39.0625 -1.21875 27.15625 -1.21875 \nQ 17.390625 -1.21875 11.203125 4.25 \nQ 5.03125 9.71875 4.15625 18.75 \nz\n\" id=\"ArialMT-53\"/>\n       </defs>\n       <use xlink:href=\"#ArialMT-73\"/>\n       <use x=\"27.783203\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"100\" xlink:href=\"#ArialMT-71\"/>\n       <use x=\"177.783203\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"250\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"283.300781\" xlink:href=\"#ArialMT-66\"/>\n       <use x=\"350\" xlink:href=\"#ArialMT-84\"/>\n       <use x=\"411.083984\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"483.300781\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"538.916016\" xlink:href=\"#ArialMT-74\"/>\n       <use x=\"588.916016\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"644.53125\" xlink:href=\"#ArialMT-112\"/>\n       <use x=\"700.146484\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"755.761719\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"811.376953\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"866.992188\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"922.607422\" xlink:href=\"#ArialMT-54\"/>\n       <use x=\"978.222656\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1033.837891\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1089.453125\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1145.068359\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1200.683594\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"1256.298828\" xlink:href=\"#ArialMT-53\"/>\n       <use x=\"1311.914062\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"xtick_3\">\n     <g id=\"line2d_3\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"223.444429\" xlink:href=\"#maea3499d55\" y=\"455.009687\"/>\n      </g>\n     </g>\n     <g id=\"text_3\">\n      <!-- Jusakul et al.(CCA_10countries_489_2017) -->\n      <g transform=\"translate(226.549429 694.136562)rotate(-90)scale(0.12 -0.12)\">\n       <defs>\n        <path d=\"M 40.578125 0 \nL 40.578125 7.625 \nQ 34.515625 -1.171875 24.125 -1.171875 \nQ 19.53125 -1.171875 15.546875 0.578125 \nQ 11.578125 2.34375 9.640625 5 \nQ 7.71875 7.671875 6.9375 11.53125 \nQ 6.390625 14.109375 6.390625 19.734375 \nL 6.390625 51.859375 \nL 15.1875 51.859375 \nL 15.1875 23.09375 \nQ 15.1875 16.21875 15.71875 13.8125 \nQ 16.546875 10.359375 19.234375 8.375 \nQ 21.921875 6.390625 25.875 6.390625 \nQ 29.828125 6.390625 33.296875 8.421875 \nQ 36.765625 10.453125 38.203125 13.9375 \nQ 39.65625 17.4375 39.65625 24.078125 \nL 39.65625 51.859375 \nL 48.4375 51.859375 \nL 48.4375 0 \nz\n\" id=\"ArialMT-117\"/>\n        <path d=\"M 3.078125 15.484375 \nL 11.765625 16.84375 \nQ 12.5 11.625 15.84375 8.84375 \nQ 19.1875 6.0625 25.203125 6.0625 \nQ 31.25 6.0625 34.171875 8.515625 \nQ 37.109375 10.984375 37.109375 14.3125 \nQ 37.109375 17.28125 34.515625 19 \nQ 32.71875 20.171875 25.53125 21.96875 \nQ 15.875 24.421875 12.140625 26.203125 \nQ 8.40625 27.984375 6.46875 31.125 \nQ 4.546875 34.28125 4.546875 38.09375 \nQ 4.546875 41.546875 6.125 44.5 \nQ 7.71875 47.46875 10.453125 49.421875 \nQ 12.5 50.921875 16.03125 51.96875 \nQ 19.578125 53.03125 23.640625 53.03125 \nQ 29.734375 53.03125 34.34375 51.265625 \nQ 38.96875 49.515625 41.15625 46.5 \nQ 43.359375 43.5 44.1875 38.484375 \nL 35.59375 37.3125 \nQ 35.015625 41.3125 32.203125 43.546875 \nQ 29.390625 45.796875 24.265625 45.796875 \nQ 18.21875 45.796875 15.625 43.796875 \nQ 13.03125 41.796875 13.03125 39.109375 \nQ 13.03125 37.40625 14.109375 36.03125 \nQ 15.1875 34.625 17.484375 33.6875 \nQ 18.796875 33.203125 25.25 31.453125 \nQ 34.578125 28.953125 38.25 27.359375 \nQ 41.9375 25.78125 44.03125 22.75 \nQ 46.140625 19.734375 46.140625 15.234375 \nQ 46.140625 10.84375 43.578125 6.953125 \nQ 41.015625 3.078125 36.171875 0.953125 \nQ 31.34375 -1.171875 25.25 -1.171875 \nQ 15.140625 -1.171875 9.84375 3.03125 \nQ 4.546875 7.234375 3.078125 15.484375 \nz\n\" id=\"ArialMT-115\"/>\n        <path d=\"M 6.640625 0 \nL 6.640625 71.578125 \nL 15.4375 71.578125 \nL 15.4375 30.765625 \nL 36.234375 51.859375 \nL 47.609375 51.859375 \nL 27.78125 32.625 \nL 49.609375 0 \nL 38.765625 0 \nL 21.625 26.515625 \nL 15.4375 20.5625 \nL 15.4375 0 \nz\n\" id=\"ArialMT-107\"/>\n        <path d=\"M 6.390625 0 \nL 6.390625 71.578125 \nL 15.1875 71.578125 \nL 15.1875 0 \nz\n\" id=\"ArialMT-108\"/>\n        <path id=\"ArialMT-32\"/>\n        <path d=\"M 42.09375 16.703125 \nL 51.171875 15.578125 \nQ 49.03125 7.625 43.21875 3.21875 \nQ 37.40625 -1.171875 28.375 -1.171875 \nQ 17 -1.171875 10.328125 5.828125 \nQ 3.65625 12.84375 3.65625 25.484375 \nQ 3.65625 38.578125 10.390625 45.796875 \nQ 17.140625 53.03125 27.875 53.03125 \nQ 38.28125 53.03125 44.875 45.953125 \nQ 51.46875 38.875 51.46875 26.03125 \nQ 51.46875 25.25 51.421875 23.6875 \nL 12.75 23.6875 \nQ 13.234375 15.140625 17.578125 10.59375 \nQ 21.921875 6.0625 28.421875 6.0625 \nQ 33.25 6.0625 36.671875 8.59375 \nQ 40.09375 11.140625 42.09375 16.703125 \nz\nM 13.234375 30.90625 \nL 42.1875 30.90625 \nQ 41.609375 37.453125 38.875 40.71875 \nQ 34.671875 45.796875 27.984375 45.796875 \nQ 21.921875 45.796875 17.796875 41.75 \nQ 13.671875 37.703125 13.234375 30.90625 \nz\n\" id=\"ArialMT-101\"/>\n        <path d=\"M 25.78125 7.859375 \nL 27.046875 0.09375 \nQ 23.34375 -0.6875 20.40625 -0.6875 \nQ 15.625 -0.6875 12.984375 0.828125 \nQ 10.359375 2.34375 9.28125 4.8125 \nQ 8.203125 7.28125 8.203125 15.1875 \nL 8.203125 45.015625 \nL 1.765625 45.015625 \nL 1.765625 51.859375 \nL 8.203125 51.859375 \nL 8.203125 64.703125 \nL 16.9375 69.96875 \nL 16.9375 51.859375 \nL 25.78125 51.859375 \nL 25.78125 45.015625 \nL 16.9375 45.015625 \nL 16.9375 14.703125 \nQ 16.9375 10.9375 17.40625 9.859375 \nQ 17.875 8.796875 18.921875 8.15625 \nQ 19.96875 7.515625 21.921875 7.515625 \nQ 23.390625 7.515625 25.78125 7.859375 \nz\n\" id=\"ArialMT-116\"/>\n        <path d=\"M 9.078125 0 \nL 9.078125 10.015625 \nL 19.09375 10.015625 \nL 19.09375 0 \nz\n\" id=\"ArialMT-46\"/>\n        <path d=\"M 40.4375 19 \nL 49.078125 17.875 \nQ 47.65625 8.9375 41.8125 3.875 \nQ 35.984375 -1.171875 27.484375 -1.171875 \nQ 16.84375 -1.171875 10.375 5.78125 \nQ 3.90625 12.75 3.90625 25.734375 \nQ 3.90625 34.125 6.6875 40.421875 \nQ 9.46875 46.734375 15.15625 49.875 \nQ 20.84375 53.03125 27.546875 53.03125 \nQ 35.984375 53.03125 41.359375 48.75 \nQ 46.734375 44.484375 48.25 36.625 \nL 39.703125 35.296875 \nQ 38.484375 40.53125 35.375 43.15625 \nQ 32.28125 45.796875 27.875 45.796875 \nQ 21.234375 45.796875 17.078125 41.03125 \nQ 12.9375 36.28125 12.9375 25.984375 \nQ 12.9375 15.53125 16.9375 10.796875 \nQ 20.953125 6.0625 27.390625 6.0625 \nQ 32.5625 6.0625 36.03125 9.234375 \nQ 39.5 12.40625 40.4375 19 \nz\n\" id=\"ArialMT-99\"/>\n        <path d=\"M 3.328125 25.921875 \nQ 3.328125 40.328125 11.328125 47.265625 \nQ 18.015625 53.03125 27.640625 53.03125 \nQ 38.328125 53.03125 45.109375 46.015625 \nQ 51.90625 39.015625 51.90625 26.65625 \nQ 51.90625 16.65625 48.90625 10.90625 \nQ 45.90625 5.171875 40.15625 2 \nQ 34.421875 -1.171875 27.640625 -1.171875 \nQ 16.75 -1.171875 10.03125 5.8125 \nQ 3.328125 12.796875 3.328125 25.921875 \nz\nM 12.359375 25.921875 \nQ 12.359375 15.96875 16.703125 11.015625 \nQ 21.046875 6.0625 27.640625 6.0625 \nQ 34.1875 6.0625 38.53125 11.03125 \nQ 42.875 16.015625 42.875 26.21875 \nQ 42.875 35.84375 38.5 40.796875 \nQ 34.125 45.75 27.640625 45.75 \nQ 21.046875 45.75 16.703125 40.8125 \nQ 12.359375 35.890625 12.359375 25.921875 \nz\n\" id=\"ArialMT-111\"/>\n        <path d=\"M 6.5 0 \nL 6.5 51.859375 \nL 14.40625 51.859375 \nL 14.40625 44 \nQ 17.4375 49.515625 20 51.265625 \nQ 22.5625 53.03125 25.640625 53.03125 \nQ 30.078125 53.03125 34.671875 50.203125 \nL 31.640625 42.046875 \nQ 28.421875 43.953125 25.203125 43.953125 \nQ 22.3125 43.953125 20.015625 42.21875 \nQ 17.71875 40.484375 16.75 37.40625 \nQ 15.28125 32.71875 15.28125 27.15625 \nL 15.28125 0 \nz\n\" id=\"ArialMT-114\"/>\n        <path d=\"M 6.640625 61.46875 \nL 6.640625 71.578125 \nL 15.4375 71.578125 \nL 15.4375 61.46875 \nz\nM 6.640625 0 \nL 6.640625 51.859375 \nL 15.4375 51.859375 \nL 15.4375 0 \nz\n\" id=\"ArialMT-105\"/>\n        <path d=\"M 32.328125 0 \nL 32.328125 17.140625 \nL 1.265625 17.140625 \nL 1.265625 25.203125 \nL 33.9375 71.578125 \nL 41.109375 71.578125 \nL 41.109375 25.203125 \nL 50.78125 25.203125 \nL 50.78125 17.140625 \nL 41.109375 17.140625 \nL 41.109375 0 \nz\nM 32.328125 25.203125 \nL 32.328125 57.46875 \nL 9.90625 25.203125 \nz\n\" id=\"ArialMT-52\"/>\n        <path d=\"M 17.671875 38.8125 \nQ 12.203125 40.828125 9.5625 44.53125 \nQ 6.9375 48.25 6.9375 53.421875 \nQ 6.9375 61.234375 12.546875 66.546875 \nQ 18.171875 71.875 27.484375 71.875 \nQ 36.859375 71.875 42.578125 66.421875 \nQ 48.296875 60.984375 48.296875 53.171875 \nQ 48.296875 48.1875 45.671875 44.5 \nQ 43.0625 40.828125 37.75 38.8125 \nQ 44.34375 36.671875 47.78125 31.875 \nQ 51.21875 27.09375 51.21875 20.453125 \nQ 51.21875 11.28125 44.71875 5.03125 \nQ 38.234375 -1.21875 27.640625 -1.21875 \nQ 17.046875 -1.21875 10.546875 5.046875 \nQ 4.046875 11.328125 4.046875 20.703125 \nQ 4.046875 27.6875 7.59375 32.390625 \nQ 11.140625 37.109375 17.671875 38.8125 \nz\nM 15.921875 53.71875 \nQ 15.921875 48.640625 19.1875 45.40625 \nQ 22.46875 42.1875 27.6875 42.1875 \nQ 32.765625 42.1875 36.015625 45.375 \nQ 39.265625 48.578125 39.265625 53.21875 \nQ 39.265625 58.0625 35.90625 61.359375 \nQ 32.5625 64.65625 27.59375 64.65625 \nQ 22.5625 64.65625 19.234375 61.421875 \nQ 15.921875 58.203125 15.921875 53.71875 \nz\nM 13.09375 20.65625 \nQ 13.09375 16.890625 14.875 13.375 \nQ 16.65625 9.859375 20.171875 7.921875 \nQ 23.6875 6 27.734375 6 \nQ 34.03125 6 38.125 10.046875 \nQ 42.234375 14.109375 42.234375 20.359375 \nQ 42.234375 26.703125 38.015625 30.859375 \nQ 33.796875 35.015625 27.4375 35.015625 \nQ 21.234375 35.015625 17.15625 30.90625 \nQ 13.09375 26.8125 13.09375 20.65625 \nz\n\" id=\"ArialMT-56\"/>\n        <path d=\"M 5.46875 16.546875 \nL 13.921875 17.328125 \nQ 14.984375 11.375 18.015625 8.6875 \nQ 21.046875 6 25.78125 6 \nQ 29.828125 6 32.875 7.859375 \nQ 35.9375 9.71875 37.890625 12.8125 \nQ 39.84375 15.921875 41.15625 21.1875 \nQ 42.484375 26.46875 42.484375 31.9375 \nQ 42.484375 32.515625 42.4375 33.6875 \nQ 39.796875 29.5 35.234375 26.875 \nQ 30.671875 24.265625 25.34375 24.265625 \nQ 16.453125 24.265625 10.296875 30.703125 \nQ 4.15625 37.15625 4.15625 47.703125 \nQ 4.15625 58.59375 10.578125 65.234375 \nQ 17 71.875 26.65625 71.875 \nQ 33.640625 71.875 39.421875 68.109375 \nQ 45.21875 64.359375 48.21875 57.390625 \nQ 51.21875 50.4375 51.21875 37.25 \nQ 51.21875 23.53125 48.234375 15.40625 \nQ 45.265625 7.28125 39.375 3.03125 \nQ 33.5 -1.21875 25.59375 -1.21875 \nQ 17.1875 -1.21875 11.859375 3.4375 \nQ 6.546875 8.109375 5.46875 16.546875 \nz\nM 41.453125 48.140625 \nQ 41.453125 55.71875 37.421875 60.15625 \nQ 33.40625 64.59375 27.734375 64.59375 \nQ 21.875 64.59375 17.53125 59.8125 \nQ 13.1875 55.03125 13.1875 47.40625 \nQ 13.1875 40.578125 17.3125 36.296875 \nQ 21.4375 32.03125 27.484375 32.03125 \nQ 33.59375 32.03125 37.515625 36.296875 \nQ 41.453125 40.578125 41.453125 48.140625 \nz\n\" id=\"ArialMT-57\"/>\n       </defs>\n       <use xlink:href=\"#ArialMT-74\"/>\n       <use x=\"50\" xlink:href=\"#ArialMT-117\"/>\n       <use x=\"105.615234\" xlink:href=\"#ArialMT-115\"/>\n       <use x=\"155.615234\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"211.230469\" xlink:href=\"#ArialMT-107\"/>\n       <use x=\"261.230469\" xlink:href=\"#ArialMT-117\"/>\n       <use x=\"316.845703\" xlink:href=\"#ArialMT-108\"/>\n       <use x=\"339.0625\" xlink:href=\"#ArialMT-32\"/>\n       <use x=\"366.845703\" xlink:href=\"#ArialMT-101\"/>\n       <use x=\"422.460938\" xlink:href=\"#ArialMT-116\"/>\n       <use x=\"450.244141\" xlink:href=\"#ArialMT-32\"/>\n       <use x=\"478.027344\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"533.642578\" xlink:href=\"#ArialMT-108\"/>\n       <use x=\"555.859375\" xlink:href=\"#ArialMT-46\"/>\n       <use x=\"583.642578\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"616.943359\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"689.160156\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"761.376953\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"828.076172\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"883.691406\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"939.306641\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"994.921875\" xlink:href=\"#ArialMT-99\"/>\n       <use x=\"1044.921875\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"1100.537109\" xlink:href=\"#ArialMT-117\"/>\n       <use x=\"1156.152344\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"1211.767578\" xlink:href=\"#ArialMT-116\"/>\n       <use x=\"1239.550781\" xlink:href=\"#ArialMT-114\"/>\n       <use x=\"1272.851562\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"1295.068359\" xlink:href=\"#ArialMT-101\"/>\n       <use x=\"1350.683594\" xlink:href=\"#ArialMT-115\"/>\n       <use x=\"1400.683594\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1456.298828\" xlink:href=\"#ArialMT-52\"/>\n       <use x=\"1511.914062\" xlink:href=\"#ArialMT-56\"/>\n       <use x=\"1567.529297\" xlink:href=\"#ArialMT-57\"/>\n       <use x=\"1623.144531\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1678.759766\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1734.375\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1789.990234\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"1845.605469\" xlink:href=\"#ArialMT-55\"/>\n       <use x=\"1901.220703\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"xtick_4\">\n     <g id=\"line2d_4\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"274.269\" xlink:href=\"#maea3499d55\" y=\"455.009687\"/>\n      </g>\n     </g>\n     <g id=\"text_4\">\n      <!-- Fudan_cohort(CCA_China_217_2022) -->\n      <g transform=\"translate(277.374 667.464687)rotate(-90)scale(0.12 -0.12)\">\n       <defs>\n        <path d=\"M 8.203125 0 \nL 8.203125 71.578125 \nL 56.5 71.578125 \nL 56.5 63.140625 \nL 17.671875 63.140625 \nL 17.671875 40.96875 \nL 51.265625 40.96875 \nL 51.265625 32.515625 \nL 17.671875 32.515625 \nL 17.671875 0 \nz\n\" id=\"ArialMT-70\"/>\n        <path d=\"M 40.234375 0 \nL 40.234375 6.546875 \nQ 35.296875 -1.171875 25.734375 -1.171875 \nQ 19.53125 -1.171875 14.328125 2.25 \nQ 9.125 5.671875 6.265625 11.796875 \nQ 3.421875 17.921875 3.421875 25.875 \nQ 3.421875 33.640625 6 39.96875 \nQ 8.59375 46.296875 13.765625 49.65625 \nQ 18.953125 53.03125 25.34375 53.03125 \nQ 30.03125 53.03125 33.6875 51.046875 \nQ 37.359375 49.078125 39.65625 45.90625 \nL 39.65625 71.578125 \nL 48.390625 71.578125 \nL 48.390625 0 \nz\nM 12.453125 25.875 \nQ 12.453125 15.921875 16.640625 10.984375 \nQ 20.84375 6.0625 26.5625 6.0625 \nQ 32.328125 6.0625 36.34375 10.765625 \nQ 40.375 15.484375 40.375 25.140625 \nQ 40.375 35.796875 36.265625 40.765625 \nQ 32.171875 45.75 26.171875 45.75 \nQ 20.3125 45.75 16.375 40.96875 \nQ 12.453125 36.1875 12.453125 25.875 \nz\n\" id=\"ArialMT-100\"/>\n        <path d=\"M 6.59375 0 \nL 6.59375 71.578125 \nL 15.375 71.578125 \nL 15.375 45.90625 \nQ 21.53125 53.03125 30.90625 53.03125 \nQ 36.671875 53.03125 40.921875 50.75 \nQ 45.171875 48.484375 47 44.484375 \nQ 48.828125 40.484375 48.828125 32.859375 \nL 48.828125 0 \nL 40.046875 0 \nL 40.046875 32.859375 \nQ 40.046875 39.453125 37.1875 42.453125 \nQ 34.328125 45.453125 29.109375 45.453125 \nQ 25.203125 45.453125 21.75 43.421875 \nQ 18.3125 41.40625 16.84375 37.9375 \nQ 15.375 34.46875 15.375 28.375 \nL 15.375 0 \nz\n\" id=\"ArialMT-104\"/>\n       </defs>\n       <use xlink:href=\"#ArialMT-70\"/>\n       <use x=\"61.083984\" xlink:href=\"#ArialMT-117\"/>\n       <use x=\"116.699219\" xlink:href=\"#ArialMT-100\"/>\n       <use x=\"172.314453\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"227.929688\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"283.544922\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"339.160156\" xlink:href=\"#ArialMT-99\"/>\n       <use x=\"389.160156\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"444.775391\" xlink:href=\"#ArialMT-104\"/>\n       <use x=\"500.390625\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"556.005859\" xlink:href=\"#ArialMT-114\"/>\n       <use x=\"589.306641\" xlink:href=\"#ArialMT-116\"/>\n       <use x=\"617.089844\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"650.390625\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"722.607422\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"794.824219\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"861.523438\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"917.138672\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"989.355469\" xlink:href=\"#ArialMT-104\"/>\n       <use x=\"1044.970703\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"1067.1875\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"1122.802734\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"1178.417969\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1234.033203\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1289.648438\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"1345.263672\" xlink:href=\"#ArialMT-55\"/>\n       <use x=\"1400.878906\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1456.494141\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1512.109375\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1567.724609\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1623.339844\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1678.955078\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"xtick_5\">\n     <g id=\"line2d_5\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"325.093571\" xlink:href=\"#maea3499d55\" y=\"455.009687\"/>\n      </g>\n     </g>\n     <g id=\"text_5\">\n      <!-- MSKCC_cohort(CCA_US_198_2018) -->\n      <g transform=\"translate(328.198571 662.098438)rotate(-90)scale(0.12 -0.12)\">\n       <defs>\n        <path d=\"M 7.421875 0 \nL 7.421875 71.578125 \nL 21.6875 71.578125 \nL 38.625 20.90625 \nQ 40.96875 13.8125 42.046875 10.296875 \nQ 43.265625 14.203125 45.84375 21.78125 \nL 62.984375 71.578125 \nL 75.734375 71.578125 \nL 75.734375 0 \nL 66.609375 0 \nL 66.609375 59.90625 \nL 45.796875 0 \nL 37.25 0 \nL 16.546875 60.9375 \nL 16.546875 0 \nz\n\" id=\"ArialMT-77\"/>\n        <path d=\"M 7.328125 0 \nL 7.328125 71.578125 \nL 16.796875 71.578125 \nL 16.796875 36.078125 \nL 52.34375 71.578125 \nL 65.1875 71.578125 \nL 35.15625 42.578125 \nL 66.5 0 \nL 54 0 \nL 28.515625 36.234375 \nL 16.796875 24.8125 \nL 16.796875 0 \nz\n\" id=\"ArialMT-75\"/>\n       </defs>\n       <use xlink:href=\"#ArialMT-77\"/>\n       <use x=\"83.300781\" xlink:href=\"#ArialMT-83\"/>\n       <use x=\"150\" xlink:href=\"#ArialMT-75\"/>\n       <use x=\"216.699219\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"288.916016\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"361.132812\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"416.748047\" xlink:href=\"#ArialMT-99\"/>\n       <use x=\"466.748047\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"522.363281\" xlink:href=\"#ArialMT-104\"/>\n       <use x=\"577.978516\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"633.59375\" xlink:href=\"#ArialMT-114\"/>\n       <use x=\"666.894531\" xlink:href=\"#ArialMT-116\"/>\n       <use x=\"694.677734\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"727.978516\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"800.195312\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"872.412109\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"939.111328\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"994.726562\" xlink:href=\"#ArialMT-85\"/>\n       <use x=\"1066.943359\" xlink:href=\"#ArialMT-83\"/>\n       <use x=\"1133.642578\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1189.257812\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"1244.873047\" xlink:href=\"#ArialMT-57\"/>\n       <use x=\"1300.488281\" xlink:href=\"#ArialMT-56\"/>\n       <use x=\"1356.103516\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1411.71875\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1467.333984\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1522.949219\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"1578.564453\" xlink:href=\"#ArialMT-56\"/>\n       <use x=\"1634.179688\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"xtick_6\">\n     <g id=\"line2d_6\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"375.918143\" xlink:href=\"#maea3499d55\" y=\"455.009687\"/>\n      </g>\n     </g>\n     <g id=\"text_6\">\n      <!-- Xiang-Yu et al.(iCCA_China_225_2022) -->\n      <g transform=\"translate(379.023143 674.137812)rotate(-90)scale(0.12 -0.12)\">\n       <defs>\n        <path d=\"M 0.4375 0 \nL 28.125 37.3125 \nL 3.71875 71.578125 \nL 14.984375 71.578125 \nL 27.984375 53.21875 \nQ 32.03125 47.515625 33.734375 44.4375 \nQ 36.140625 48.34375 39.40625 52.59375 \nL 53.8125 71.578125 \nL 64.109375 71.578125 \nL 38.96875 37.84375 \nL 66.0625 0 \nL 54.34375 0 \nL 36.328125 25.53125 \nQ 34.8125 27.734375 33.203125 30.328125 \nQ 30.8125 26.421875 29.78125 24.953125 \nL 11.8125 0 \nz\n\" id=\"ArialMT-88\"/>\n        <path d=\"M 4.984375 -4.296875 \nL 13.53125 -5.5625 \nQ 14.0625 -9.515625 16.5 -11.328125 \nQ 19.78125 -13.765625 25.4375 -13.765625 \nQ 31.546875 -13.765625 34.859375 -11.328125 \nQ 38.1875 -8.890625 39.359375 -4.5 \nQ 40.046875 -1.8125 39.984375 6.78125 \nQ 34.234375 0 25.640625 0 \nQ 14.9375 0 9.078125 7.71875 \nQ 3.21875 15.4375 3.21875 26.21875 \nQ 3.21875 33.640625 5.90625 39.90625 \nQ 8.59375 46.1875 13.6875 49.609375 \nQ 18.796875 53.03125 25.6875 53.03125 \nQ 34.859375 53.03125 40.828125 45.609375 \nL 40.828125 51.859375 \nL 48.921875 51.859375 \nL 48.921875 7.03125 \nQ 48.921875 -5.078125 46.453125 -10.125 \nQ 44 -15.1875 38.640625 -18.109375 \nQ 33.296875 -21.046875 25.484375 -21.046875 \nQ 16.21875 -21.046875 10.5 -16.875 \nQ 4.78125 -12.703125 4.984375 -4.296875 \nz\nM 12.25 26.859375 \nQ 12.25 16.65625 16.296875 11.96875 \nQ 20.359375 7.28125 26.46875 7.28125 \nQ 32.515625 7.28125 36.609375 11.9375 \nQ 40.71875 16.609375 40.71875 26.5625 \nQ 40.71875 36.078125 36.5 40.90625 \nQ 32.28125 45.75 26.3125 45.75 \nQ 20.453125 45.75 16.34375 40.984375 \nQ 12.25 36.234375 12.25 26.859375 \nz\n\" id=\"ArialMT-103\"/>\n        <path d=\"M 3.171875 21.484375 \nL 3.171875 30.328125 \nL 30.171875 30.328125 \nL 30.171875 21.484375 \nz\n\" id=\"ArialMT-45\"/>\n        <path d=\"M 27.875 0 \nL 27.875 30.328125 \nL 0.296875 71.578125 \nL 11.8125 71.578125 \nL 25.921875 50 \nQ 29.828125 43.953125 33.203125 37.890625 \nQ 36.421875 43.5 41.015625 50.53125 \nL 54.890625 71.578125 \nL 65.921875 71.578125 \nL 37.359375 30.328125 \nL 37.359375 0 \nz\n\" id=\"ArialMT-89\"/>\n       </defs>\n       <use xlink:href=\"#ArialMT-88\"/>\n       <use x=\"66.699219\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"88.916016\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"144.53125\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"200.146484\" xlink:href=\"#ArialMT-103\"/>\n       <use x=\"255.761719\" xlink:href=\"#ArialMT-45\"/>\n       <use x=\"289.0625\" xlink:href=\"#ArialMT-89\"/>\n       <use x=\"350.261719\" xlink:href=\"#ArialMT-117\"/>\n       <use x=\"405.876953\" xlink:href=\"#ArialMT-32\"/>\n       <use x=\"433.660156\" xlink:href=\"#ArialMT-101\"/>\n       <use x=\"489.275391\" xlink:href=\"#ArialMT-116\"/>\n       <use x=\"517.058594\" xlink:href=\"#ArialMT-32\"/>\n       <use x=\"544.841797\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"600.457031\" xlink:href=\"#ArialMT-108\"/>\n       <use x=\"622.673828\" xlink:href=\"#ArialMT-46\"/>\n       <use x=\"650.457031\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"683.757812\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"705.974609\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"778.191406\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"850.408203\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"917.107422\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"972.722656\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"1044.939453\" xlink:href=\"#ArialMT-104\"/>\n       <use x=\"1100.554688\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"1122.771484\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"1178.386719\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"1234.001953\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1289.617188\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1345.232422\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1400.847656\" xlink:href=\"#ArialMT-53\"/>\n       <use x=\"1456.462891\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1512.078125\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1567.693359\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1623.308594\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1678.923828\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1734.539062\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"xtick_7\">\n     <g id=\"line2d_7\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"426.742714\" xlink:href=\"#maea3499d55\" y=\"455.009687\"/>\n      </g>\n     </g>\n     <g id=\"text_7\">\n      <!-- Fudan_Dong_cohort(iCCA_China_253_2022) -->\n      <g transform=\"translate(429.847714 705.489687)rotate(-90)scale(0.12 -0.12)\">\n       <defs>\n        <path d=\"M 7.71875 0 \nL 7.71875 71.578125 \nL 32.375 71.578125 \nQ 40.71875 71.578125 45.125 70.5625 \nQ 51.265625 69.140625 55.609375 65.4375 \nQ 61.28125 60.640625 64.078125 53.1875 \nQ 66.890625 45.75 66.890625 36.1875 \nQ 66.890625 28.03125 64.984375 21.734375 \nQ 63.09375 15.4375 60.109375 11.296875 \nQ 57.125 7.171875 53.578125 4.796875 \nQ 50.046875 2.4375 45.046875 1.21875 \nQ 40.046875 0 33.546875 0 \nz\nM 17.1875 8.453125 \nL 32.46875 8.453125 \nQ 39.546875 8.453125 43.578125 9.765625 \nQ 47.609375 11.078125 50 13.484375 \nQ 53.375 16.84375 55.25 22.53125 \nQ 57.125 28.21875 57.125 36.328125 \nQ 57.125 47.5625 53.4375 53.59375 \nQ 49.75 59.625 44.484375 61.671875 \nQ 40.671875 63.140625 32.234375 63.140625 \nL 17.1875 63.140625 \nz\n\" id=\"ArialMT-68\"/>\n        <path d=\"M 4.203125 18.890625 \nL 12.984375 20.0625 \nQ 14.5 12.59375 18.140625 9.296875 \nQ 21.78125 6 27 6 \nQ 33.203125 6 37.46875 10.296875 \nQ 41.75 14.59375 41.75 20.953125 \nQ 41.75 27 37.796875 30.921875 \nQ 33.84375 34.859375 27.734375 34.859375 \nQ 25.25 34.859375 21.53125 33.890625 \nL 22.515625 41.609375 \nQ 23.390625 41.5 23.921875 41.5 \nQ 29.546875 41.5 34.03125 44.421875 \nQ 38.53125 47.359375 38.53125 53.46875 \nQ 38.53125 58.296875 35.25 61.46875 \nQ 31.984375 64.65625 26.8125 64.65625 \nQ 21.6875 64.65625 18.265625 61.421875 \nQ 14.84375 58.203125 13.875 51.765625 \nL 5.078125 53.328125 \nQ 6.6875 62.15625 12.390625 67.015625 \nQ 18.109375 71.875 26.609375 71.875 \nQ 32.46875 71.875 37.390625 69.359375 \nQ 42.328125 66.84375 44.9375 62.5 \nQ 47.5625 58.15625 47.5625 53.265625 \nQ 47.5625 48.640625 45.0625 44.828125 \nQ 42.578125 41.015625 37.703125 38.765625 \nQ 44.046875 37.3125 47.5625 32.6875 \nQ 51.078125 28.078125 51.078125 21.140625 \nQ 51.078125 11.765625 44.234375 5.25 \nQ 37.40625 -1.265625 26.953125 -1.265625 \nQ 17.53125 -1.265625 11.296875 4.34375 \nQ 5.078125 9.96875 4.203125 18.890625 \nz\n\" id=\"ArialMT-51\"/>\n       </defs>\n       <use xlink:href=\"#ArialMT-70\"/>\n       <use x=\"61.083984\" xlink:href=\"#ArialMT-117\"/>\n       <use x=\"116.699219\" xlink:href=\"#ArialMT-100\"/>\n       <use x=\"172.314453\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"227.929688\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"283.544922\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"339.160156\" xlink:href=\"#ArialMT-68\"/>\n       <use x=\"411.376953\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"466.992188\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"522.607422\" xlink:href=\"#ArialMT-103\"/>\n       <use x=\"578.222656\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"633.837891\" xlink:href=\"#ArialMT-99\"/>\n       <use x=\"683.837891\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"739.453125\" xlink:href=\"#ArialMT-104\"/>\n       <use x=\"795.068359\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"850.683594\" xlink:href=\"#ArialMT-114\"/>\n       <use x=\"883.984375\" xlink:href=\"#ArialMT-116\"/>\n       <use x=\"911.767578\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"945.068359\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"967.285156\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"1039.501953\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"1111.71875\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"1178.417969\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1234.033203\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"1306.25\" xlink:href=\"#ArialMT-104\"/>\n       <use x=\"1361.865234\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"1384.082031\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"1439.697266\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"1495.3125\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1550.927734\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1606.542969\" xlink:href=\"#ArialMT-53\"/>\n       <use x=\"1662.158203\" xlink:href=\"#ArialMT-51\"/>\n       <use x=\"1717.773438\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1773.388672\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1829.003906\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1884.619141\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1940.234375\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1995.849609\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n   </g>\n   <g id=\"matplotlib.axis_2\">\n    <g id=\"ytick_1\">\n     <g id=\"line2d_8\">\n      <defs>\n       <path d=\"M 0 0 \nL 3.5 0 \n\" id=\"m43883b6bfa\" style=\"stroke:#000000;stroke-width:0.8;\"/>\n      </defs>\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"452.155\" xlink:href=\"#m43883b6bfa\" y=\"124.649973\"/>\n      </g>\n     </g>\n     <g id=\"text_8\">\n      <!-- TCGA(CCA_US_Can_2017) -->\n      <g transform=\"translate(459.155 129.017786)scale(0.12 -0.12)\">\n       <use xlink:href=\"#ArialMT-84\"/>\n       <use x=\"61.083984\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"133.300781\" xlink:href=\"#ArialMT-71\"/>\n       <use x=\"211.083984\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"277.783203\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"311.083984\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"383.300781\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"455.517578\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"522.216797\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"577.832031\" xlink:href=\"#ArialMT-85\"/>\n       <use x=\"650.048828\" xlink:href=\"#ArialMT-83\"/>\n       <use x=\"716.748047\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"772.363281\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"844.580078\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"900.195312\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"955.810547\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1011.425781\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1067.041016\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1122.65625\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"1178.271484\" xlink:href=\"#ArialMT-55\"/>\n       <use x=\"1233.886719\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"ytick_2\">\n     <g id=\"line2d_9\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"452.155\" xlink:href=\"#m43883b6bfa\" y=\"175.474545\"/>\n      </g>\n     </g>\n     <g id=\"text_9\">\n      <!-- ICGC(BTC_Japan_260_2015) -->\n      <g transform=\"translate(459.155 179.842357)scale(0.12 -0.12)\">\n       <use xlink:href=\"#ArialMT-73\"/>\n       <use x=\"27.783203\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"100\" xlink:href=\"#ArialMT-71\"/>\n       <use x=\"177.783203\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"250\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"283.300781\" xlink:href=\"#ArialMT-66\"/>\n       <use x=\"350\" xlink:href=\"#ArialMT-84\"/>\n       <use x=\"411.083984\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"483.300781\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"538.916016\" xlink:href=\"#ArialMT-74\"/>\n       <use x=\"588.916016\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"644.53125\" xlink:href=\"#ArialMT-112\"/>\n       <use x=\"700.146484\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"755.761719\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"811.376953\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"866.992188\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"922.607422\" xlink:href=\"#ArialMT-54\"/>\n       <use x=\"978.222656\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1033.837891\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1089.453125\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1145.068359\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1200.683594\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"1256.298828\" xlink:href=\"#ArialMT-53\"/>\n       <use x=\"1311.914062\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"ytick_3\">\n     <g id=\"line2d_10\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"452.155\" xlink:href=\"#m43883b6bfa\" y=\"226.299116\"/>\n      </g>\n     </g>\n     <g id=\"text_10\">\n      <!-- Jusakul et al.(CCA_10countries_489_2017) -->\n      <g transform=\"translate(459.155 230.666929)scale(0.12 -0.12)\">\n       <use xlink:href=\"#ArialMT-74\"/>\n       <use x=\"50\" xlink:href=\"#ArialMT-117\"/>\n       <use x=\"105.615234\" xlink:href=\"#ArialMT-115\"/>\n       <use x=\"155.615234\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"211.230469\" xlink:href=\"#ArialMT-107\"/>\n       <use x=\"261.230469\" xlink:href=\"#ArialMT-117\"/>\n       <use x=\"316.845703\" xlink:href=\"#ArialMT-108\"/>\n       <use x=\"339.0625\" xlink:href=\"#ArialMT-32\"/>\n       <use x=\"366.845703\" xlink:href=\"#ArialMT-101\"/>\n       <use x=\"422.460938\" xlink:href=\"#ArialMT-116\"/>\n       <use x=\"450.244141\" xlink:href=\"#ArialMT-32\"/>\n       <use x=\"478.027344\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"533.642578\" xlink:href=\"#ArialMT-108\"/>\n       <use x=\"555.859375\" xlink:href=\"#ArialMT-46\"/>\n       <use x=\"583.642578\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"616.943359\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"689.160156\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"761.376953\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"828.076172\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"883.691406\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"939.306641\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"994.921875\" xlink:href=\"#ArialMT-99\"/>\n       <use x=\"1044.921875\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"1100.537109\" xlink:href=\"#ArialMT-117\"/>\n       <use x=\"1156.152344\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"1211.767578\" xlink:href=\"#ArialMT-116\"/>\n       <use x=\"1239.550781\" xlink:href=\"#ArialMT-114\"/>\n       <use x=\"1272.851562\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"1295.068359\" xlink:href=\"#ArialMT-101\"/>\n       <use x=\"1350.683594\" xlink:href=\"#ArialMT-115\"/>\n       <use x=\"1400.683594\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1456.298828\" xlink:href=\"#ArialMT-52\"/>\n       <use x=\"1511.914062\" xlink:href=\"#ArialMT-56\"/>\n       <use x=\"1567.529297\" xlink:href=\"#ArialMT-57\"/>\n       <use x=\"1623.144531\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1678.759766\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1734.375\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1789.990234\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"1845.605469\" xlink:href=\"#ArialMT-55\"/>\n       <use x=\"1901.220703\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"ytick_4\">\n     <g id=\"line2d_11\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"452.155\" xlink:href=\"#m43883b6bfa\" y=\"277.123688\"/>\n      </g>\n     </g>\n     <g id=\"text_11\">\n      <!-- Fudan_cohort(CCA_China_217_2022) -->\n      <g transform=\"translate(459.155 281.4915)scale(0.12 -0.12)\">\n       <use xlink:href=\"#ArialMT-70\"/>\n       <use x=\"61.083984\" xlink:href=\"#ArialMT-117\"/>\n       <use x=\"116.699219\" xlink:href=\"#ArialMT-100\"/>\n       <use x=\"172.314453\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"227.929688\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"283.544922\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"339.160156\" xlink:href=\"#ArialMT-99\"/>\n       <use x=\"389.160156\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"444.775391\" xlink:href=\"#ArialMT-104\"/>\n       <use x=\"500.390625\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"556.005859\" xlink:href=\"#ArialMT-114\"/>\n       <use x=\"589.306641\" xlink:href=\"#ArialMT-116\"/>\n       <use x=\"617.089844\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"650.390625\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"722.607422\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"794.824219\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"861.523438\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"917.138672\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"989.355469\" xlink:href=\"#ArialMT-104\"/>\n       <use x=\"1044.970703\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"1067.1875\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"1122.802734\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"1178.417969\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1234.033203\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1289.648438\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"1345.263672\" xlink:href=\"#ArialMT-55\"/>\n       <use x=\"1400.878906\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1456.494141\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1512.109375\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1567.724609\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1623.339844\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1678.955078\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"ytick_5\">\n     <g id=\"line2d_12\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"452.155\" xlink:href=\"#m43883b6bfa\" y=\"327.948259\"/>\n      </g>\n     </g>\n     <g id=\"text_12\">\n      <!-- MSKCC_cohort(CCA_US_198_2018) -->\n      <g transform=\"translate(459.155 332.316071)scale(0.12 -0.12)\">\n       <use xlink:href=\"#ArialMT-77\"/>\n       <use x=\"83.300781\" xlink:href=\"#ArialMT-83\"/>\n       <use x=\"150\" xlink:href=\"#ArialMT-75\"/>\n       <use x=\"216.699219\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"288.916016\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"361.132812\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"416.748047\" xlink:href=\"#ArialMT-99\"/>\n       <use x=\"466.748047\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"522.363281\" xlink:href=\"#ArialMT-104\"/>\n       <use x=\"577.978516\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"633.59375\" xlink:href=\"#ArialMT-114\"/>\n       <use x=\"666.894531\" xlink:href=\"#ArialMT-116\"/>\n       <use x=\"694.677734\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"727.978516\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"800.195312\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"872.412109\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"939.111328\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"994.726562\" xlink:href=\"#ArialMT-85\"/>\n       <use x=\"1066.943359\" xlink:href=\"#ArialMT-83\"/>\n       <use x=\"1133.642578\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1189.257812\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"1244.873047\" xlink:href=\"#ArialMT-57\"/>\n       <use x=\"1300.488281\" xlink:href=\"#ArialMT-56\"/>\n       <use x=\"1356.103516\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1411.71875\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1467.333984\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1522.949219\" xlink:href=\"#ArialMT-49\"/>\n       <use x=\"1578.564453\" xlink:href=\"#ArialMT-56\"/>\n       <use x=\"1634.179688\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"ytick_6\">\n     <g id=\"line2d_13\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"452.155\" xlink:href=\"#m43883b6bfa\" y=\"378.77283\"/>\n      </g>\n     </g>\n     <g id=\"text_13\">\n      <!-- Xiang-Yu et al.(iCCA_China_225_2022) -->\n      <g transform=\"translate(459.155 383.140643)scale(0.12 -0.12)\">\n       <use xlink:href=\"#ArialMT-88\"/>\n       <use x=\"66.699219\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"88.916016\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"144.53125\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"200.146484\" xlink:href=\"#ArialMT-103\"/>\n       <use x=\"255.761719\" xlink:href=\"#ArialMT-45\"/>\n       <use x=\"289.0625\" xlink:href=\"#ArialMT-89\"/>\n       <use x=\"350.261719\" xlink:href=\"#ArialMT-117\"/>\n       <use x=\"405.876953\" xlink:href=\"#ArialMT-32\"/>\n       <use x=\"433.660156\" xlink:href=\"#ArialMT-101\"/>\n       <use x=\"489.275391\" xlink:href=\"#ArialMT-116\"/>\n       <use x=\"517.058594\" xlink:href=\"#ArialMT-32\"/>\n       <use x=\"544.841797\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"600.457031\" xlink:href=\"#ArialMT-108\"/>\n       <use x=\"622.673828\" xlink:href=\"#ArialMT-46\"/>\n       <use x=\"650.457031\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"683.757812\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"705.974609\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"778.191406\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"850.408203\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"917.107422\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"972.722656\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"1044.939453\" xlink:href=\"#ArialMT-104\"/>\n       <use x=\"1100.554688\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"1122.771484\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"1178.386719\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"1234.001953\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1289.617188\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1345.232422\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1400.847656\" xlink:href=\"#ArialMT-53\"/>\n       <use x=\"1456.462891\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1512.078125\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1567.693359\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1623.308594\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1678.923828\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1734.539062\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"ytick_7\">\n     <g id=\"line2d_14\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"452.155\" xlink:href=\"#m43883b6bfa\" y=\"429.597402\"/>\n      </g>\n     </g>\n     <g id=\"text_14\">\n      <!-- Fudan_Dong_cohort(iCCA_China_253_2022) -->\n      <g transform=\"translate(459.155 433.965214)scale(0.12 -0.12)\">\n       <use xlink:href=\"#ArialMT-70\"/>\n       <use x=\"61.083984\" xlink:href=\"#ArialMT-117\"/>\n       <use x=\"116.699219\" xlink:href=\"#ArialMT-100\"/>\n       <use x=\"172.314453\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"227.929688\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"283.544922\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"339.160156\" xlink:href=\"#ArialMT-68\"/>\n       <use x=\"411.376953\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"466.992188\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"522.607422\" xlink:href=\"#ArialMT-103\"/>\n       <use x=\"578.222656\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"633.837891\" xlink:href=\"#ArialMT-99\"/>\n       <use x=\"683.837891\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"739.453125\" xlink:href=\"#ArialMT-104\"/>\n       <use x=\"795.068359\" xlink:href=\"#ArialMT-111\"/>\n       <use x=\"850.683594\" xlink:href=\"#ArialMT-114\"/>\n       <use x=\"883.984375\" xlink:href=\"#ArialMT-116\"/>\n       <use x=\"911.767578\" xlink:href=\"#ArialMT-40\"/>\n       <use x=\"945.068359\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"967.285156\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"1039.501953\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"1111.71875\" xlink:href=\"#ArialMT-65\"/>\n       <use x=\"1178.417969\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1234.033203\" xlink:href=\"#ArialMT-67\"/>\n       <use x=\"1306.25\" xlink:href=\"#ArialMT-104\"/>\n       <use x=\"1361.865234\" xlink:href=\"#ArialMT-105\"/>\n       <use x=\"1384.082031\" xlink:href=\"#ArialMT-110\"/>\n       <use x=\"1439.697266\" xlink:href=\"#ArialMT-97\"/>\n       <use x=\"1495.3125\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1550.927734\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1606.542969\" xlink:href=\"#ArialMT-53\"/>\n       <use x=\"1662.158203\" xlink:href=\"#ArialMT-51\"/>\n       <use x=\"1717.773438\" xlink:href=\"#ArialMT-95\"/>\n       <use x=\"1773.388672\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1829.003906\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"1884.619141\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1940.234375\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"1995.849609\" xlink:href=\"#ArialMT-41\"/>\n      </g>\n     </g>\n    </g>\n   </g>\n   <g id=\"text_15\">\n    <!-- 1 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(118.458723 127.752161)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-49\"/>\n    </g>\n   </g>\n   <g id=\"text_16\">\n    <!-- -0.063 -->\n    <g style=\"fill:#262626;\" transform=\"translate(155.60892 127.752161)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-45\"/>\n     <use x=\"33.300781\" xlink:href=\"#ArialMT-48\"/>\n     <use x=\"88.916016\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"116.699219\" xlink:href=\"#ArialMT-48\"/>\n     <use x=\"172.314453\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"227.929688\" xlink:href=\"#ArialMT-51\"/>\n    </g>\n   </g>\n   <g id=\"text_17\">\n    <!-- -0.05 -->\n    <g style=\"fill:#262626;\" transform=\"translate(209.770054 127.752161)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-45\"/>\n     <use x=\"33.300781\" xlink:href=\"#ArialMT-48\"/>\n     <use x=\"88.916016\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"116.699219\" xlink:href=\"#ArialMT-48\"/>\n     <use x=\"172.314453\" xlink:href=\"#ArialMT-53\"/>\n    </g>\n   </g>\n   <g id=\"text_18\">\n    <!-- 0.42 -->\n    <g style=\"fill:#262626;\" transform=\"translate(262.592438 127.752161)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-52\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-50\"/>\n    </g>\n   </g>\n   <g id=\"text_19\">\n    <!-- 0.71 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(313.417009 127.752161)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-55\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-49\"/>\n    </g>\n   </g>\n   <g id=\"text_20\">\n    <!-- 0.43 -->\n    <g style=\"fill:#262626;\" transform=\"translate(364.24158 127.752161)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-52\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-51\"/>\n    </g>\n   </g>\n   <g id=\"text_21\">\n    <!-- 0.4 -->\n    <g style=\"fill:#262626;\" transform=\"translate(418.402714 127.752161)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-52\"/>\n    </g>\n   </g>\n   <g id=\"text_22\">\n    <!-- -0.063 -->\n    <g style=\"fill:#262626;\" transform=\"translate(104.784348 178.576732)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-45\"/>\n     <use x=\"33.300781\" xlink:href=\"#ArialMT-48\"/>\n     <use x=\"88.916016\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"116.699219\" xlink:href=\"#ArialMT-48\"/>\n     <use x=\"172.314453\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"227.929688\" xlink:href=\"#ArialMT-51\"/>\n    </g>\n   </g>\n   <g id=\"text_23\">\n    <!-- 1 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(169.283295 178.576732)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-49\"/>\n    </g>\n   </g>\n   <g id=\"text_24\">\n    <!-- 0.94 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(211.767866 178.576732)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-57\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-52\"/>\n    </g>\n   </g>\n   <g id=\"text_25\">\n    <!-- 0.6 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(265.929 178.576732)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n    </g>\n   </g>\n   <g id=\"text_26\">\n    <!-- 0.36 -->\n    <g style=\"fill:#262626;\" transform=\"translate(313.417009 178.576732)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-51\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-54\"/>\n    </g>\n   </g>\n   <g id=\"text_27\">\n    <!-- 0.69 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(364.24158 178.576732)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-57\"/>\n    </g>\n   </g>\n   <g id=\"text_28\">\n    <!-- 0.72 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(415.066152 178.576732)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-55\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-50\"/>\n    </g>\n   </g>\n   <g id=\"text_29\">\n    <!-- -0.05 -->\n    <g style=\"fill:#262626;\" transform=\"translate(108.120911 229.401304)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-45\"/>\n     <use x=\"33.300781\" xlink:href=\"#ArialMT-48\"/>\n     <use x=\"88.916016\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"116.699219\" xlink:href=\"#ArialMT-48\"/>\n     <use x=\"172.314453\" xlink:href=\"#ArialMT-53\"/>\n    </g>\n   </g>\n   <g id=\"text_30\">\n    <!-- 0.94 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(160.943295 229.401304)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-57\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-52\"/>\n    </g>\n   </g>\n   <g id=\"text_31\">\n    <!-- 1 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(220.107866 229.401304)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-49\"/>\n    </g>\n   </g>\n   <g id=\"text_32\">\n    <!-- 0.68 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(262.592438 229.401304)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-56\"/>\n    </g>\n   </g>\n   <g id=\"text_33\">\n    <!-- 0.39 -->\n    <g style=\"fill:#262626;\" transform=\"translate(313.417009 229.401304)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-51\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-57\"/>\n    </g>\n   </g>\n   <g id=\"text_34\">\n    <!-- 0.66 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(364.24158 229.401304)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-54\"/>\n    </g>\n   </g>\n   <g id=\"text_35\">\n    <!-- 0.62 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(415.066152 229.401304)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-50\"/>\n    </g>\n   </g>\n   <g id=\"text_36\">\n    <!-- 0.42 -->\n    <g style=\"fill:#262626;\" transform=\"translate(110.118723 280.225875)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-52\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-50\"/>\n    </g>\n   </g>\n   <g id=\"text_37\">\n    <!-- 0.6 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(164.279857 280.225875)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n    </g>\n   </g>\n   <g id=\"text_38\">\n    <!-- 0.68 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(211.767866 280.225875)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-56\"/>\n    </g>\n   </g>\n   <g id=\"text_39\">\n    <!-- 1 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(270.932437 280.225875)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-49\"/>\n    </g>\n   </g>\n   <g id=\"text_40\">\n    <!-- 0.68 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(313.417009 280.225875)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-56\"/>\n    </g>\n   </g>\n   <g id=\"text_41\">\n    <!-- 0.62 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(364.24158 280.225875)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-50\"/>\n    </g>\n   </g>\n   <g id=\"text_42\">\n    <!-- 0.67 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(415.066152 280.225875)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-55\"/>\n    </g>\n   </g>\n   <g id=\"text_43\">\n    <!-- 0.71 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(110.118723 331.050446)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-55\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-49\"/>\n    </g>\n   </g>\n   <g id=\"text_44\">\n    <!-- 0.36 -->\n    <g style=\"fill:#262626;\" transform=\"translate(160.943295 331.050446)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-51\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-54\"/>\n    </g>\n   </g>\n   <g id=\"text_45\">\n    <!-- 0.39 -->\n    <g style=\"fill:#262626;\" transform=\"translate(211.767866 331.050446)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-51\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-57\"/>\n    </g>\n   </g>\n   <g id=\"text_46\">\n    <!-- 0.68 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(262.592438 331.050446)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-56\"/>\n    </g>\n   </g>\n   <g id=\"text_47\">\n    <!-- 1 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(321.757009 331.050446)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-49\"/>\n    </g>\n   </g>\n   <g id=\"text_48\">\n    <!-- 0.84 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(364.24158 331.050446)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-56\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-52\"/>\n    </g>\n   </g>\n   <g id=\"text_49\">\n    <!-- 0.8 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(418.402714 331.050446)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-56\"/>\n    </g>\n   </g>\n   <g id=\"text_50\">\n    <!-- 0.43 -->\n    <g style=\"fill:#262626;\" transform=\"translate(110.118723 381.875018)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-52\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-51\"/>\n    </g>\n   </g>\n   <g id=\"text_51\">\n    <!-- 0.69 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(160.943295 381.875018)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-57\"/>\n    </g>\n   </g>\n   <g id=\"text_52\">\n    <!-- 0.66 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(211.767866 381.875018)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-54\"/>\n    </g>\n   </g>\n   <g id=\"text_53\">\n    <!-- 0.62 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(262.592438 381.875018)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-50\"/>\n    </g>\n   </g>\n   <g id=\"text_54\">\n    <!-- 0.84 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(313.417009 381.875018)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-56\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-52\"/>\n    </g>\n   </g>\n   <g id=\"text_55\">\n    <!-- 1 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(372.58158 381.875018)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-49\"/>\n    </g>\n   </g>\n   <g id=\"text_56\">\n    <!-- 0.93 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(415.066152 381.875018)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-57\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-51\"/>\n    </g>\n   </g>\n   <g id=\"text_57\">\n    <!-- 0.4 -->\n    <g style=\"fill:#262626;\" transform=\"translate(113.455286 432.699589)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-52\"/>\n    </g>\n   </g>\n   <g id=\"text_58\">\n    <!-- 0.72 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(160.943295 432.699589)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-55\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-50\"/>\n    </g>\n   </g>\n   <g id=\"text_59\">\n    <!-- 0.62 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(211.767866 432.699589)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-50\"/>\n    </g>\n   </g>\n   <g id=\"text_60\">\n    <!-- 0.67 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(262.592438 432.699589)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-54\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-55\"/>\n    </g>\n   </g>\n   <g id=\"text_61\">\n    <!-- 0.8 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(316.753571 432.699589)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-56\"/>\n    </g>\n   </g>\n   <g id=\"text_62\">\n    <!-- 0.93 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(364.24158 432.699589)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-48\"/>\n     <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n     <use x=\"83.398438\" xlink:href=\"#ArialMT-57\"/>\n     <use x=\"139.013672\" xlink:href=\"#ArialMT-51\"/>\n    </g>\n   </g>\n   <g id=\"text_63\">\n    <!-- 1 -->\n    <g style=\"fill:#ffffff;\" transform=\"translate(423.406152 432.699589)scale(0.12 -0.12)\">\n     <use xlink:href=\"#ArialMT-49\"/>\n    </g>\n   </g>\n  </g>\n  <g id=\"axes_4\">\n   <g id=\"patch_3\">\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 141.094688 \nL 8.64 132.454688 \nL 8.64 20.134688 \nL 8.64 11.494688 \nL 44.64 11.494688 \nL 44.64 20.134688 \nL 44.64 132.454688 \nL 44.64 141.094688 \nz\n\" style=\"fill:#ffffff;stroke:#ffffff;stroke-linejoin:miter;stroke-width:0.01;\"/>\n   </g>\n   <g id=\"QuadMesh_2\">\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 141.094688 \nL 44.64 141.094688 \nL 44.64 132.454688 \nL 8.64 132.454688 \nL 8.64 141.094688 \n\" style=\"fill:#ffffff;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 132.454688 \nL 44.64 132.454688 \nL 44.64 123.814688 \nL 8.64 123.814688 \nL 8.64 132.454688 \n\" style=\"fill:#ffeded;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 123.814688 \nL 44.64 123.814688 \nL 44.64 115.174688 \nL 8.64 115.174688 \nL 8.64 123.814688 \n\" style=\"fill:#ffdbdb;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 115.174688 \nL 44.64 115.174688 \nL 44.64 106.534688 \nL 8.64 106.534688 \nL 8.64 115.174688 \n\" style=\"fill:#fec8c8;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 106.534688 \nL 44.64 106.534688 \nL 44.64 97.894688 \nL 8.64 97.894688 \nL 8.64 106.534688 \n\" style=\"fill:#feb6b6;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 97.894688 \nL 44.64 97.894688 \nL 44.64 89.254688 \nL 8.64 89.254688 \nL 8.64 97.894688 \n\" style=\"fill:#fea4a4;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 89.254688 \nL 44.64 89.254688 \nL 44.64 80.614688 \nL 8.64 80.614688 \nL 8.64 89.254688 \n\" style=\"fill:#fe9292;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 80.614688 \nL 44.64 80.614688 \nL 44.64 71.974688 \nL 8.64 71.974688 \nL 8.64 80.614688 \n\" style=\"fill:#fe8080;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 71.974688 \nL 44.64 71.974688 \nL 44.64 63.334688 \nL 8.64 63.334688 \nL 8.64 71.974688 \n\" style=\"fill:#fd6d6d;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 63.334688 \nL 44.64 63.334688 \nL 44.64 54.694688 \nL 8.64 54.694688 \nL 8.64 63.334688 \n\" style=\"fill:#fd5b5b;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 54.694688 \nL 44.64 54.694688 \nL 44.64 46.054688 \nL 8.64 46.054688 \nL 8.64 54.694688 \n\" style=\"fill:#fd4949;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 46.054688 \nL 44.64 46.054688 \nL 44.64 37.414688 \nL 8.64 37.414688 \nL 8.64 46.054688 \n\" style=\"fill:#fd3737;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 37.414688 \nL 44.64 37.414688 \nL 44.64 28.774688 \nL 8.64 28.774688 \nL 8.64 37.414688 \n\" style=\"fill:#fc2424;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 28.774688 \nL 44.64 28.774688 \nL 44.64 20.134688 \nL 8.64 20.134688 \nL 8.64 28.774688 \n\" style=\"fill:#fc1212;\"/>\n    <path clip-path=\"url(#p8803b15104)\" d=\"M 8.64 20.134688 \nL 44.64 20.134688 \nL 44.64 11.494688 \nL 8.64 11.494688 \nL 8.64 20.134688 \n\" style=\"fill:#fc0000;\"/>\n   </g>\n   <g id=\"matplotlib.axis_3\"/>\n   <g id=\"matplotlib.axis_4\">\n    <g id=\"ytick_8\">\n     <g id=\"line2d_15\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"44.64\" xlink:href=\"#m43883b6bfa\" y=\"133.393268\"/>\n      </g>\n     </g>\n     <g id=\"text_64\">\n      <!-- 0.00 -->\n      <g transform=\"translate(51.64 137.687956)scale(0.12 -0.12)\">\n       <use xlink:href=\"#ArialMT-48\"/>\n       <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n       <use x=\"83.398438\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"139.013672\" xlink:href=\"#ArialMT-48\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"ytick_9\">\n     <g id=\"line2d_16\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"44.64\" xlink:href=\"#m43883b6bfa\" y=\"102.918623\"/>\n      </g>\n     </g>\n     <g id=\"text_65\">\n      <!-- 0.25 -->\n      <g transform=\"translate(51.64 107.213311)scale(0.12 -0.12)\">\n       <use xlink:href=\"#ArialMT-48\"/>\n       <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n       <use x=\"83.398438\" xlink:href=\"#ArialMT-50\"/>\n       <use x=\"139.013672\" xlink:href=\"#ArialMT-53\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"ytick_10\">\n     <g id=\"line2d_17\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"44.64\" xlink:href=\"#m43883b6bfa\" y=\"72.443978\"/>\n      </g>\n     </g>\n     <g id=\"text_66\">\n      <!-- 0.50 -->\n      <g transform=\"translate(51.64 76.738665)scale(0.12 -0.12)\">\n       <use xlink:href=\"#ArialMT-48\"/>\n       <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n       <use x=\"83.398438\" xlink:href=\"#ArialMT-53\"/>\n       <use x=\"139.013672\" xlink:href=\"#ArialMT-48\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"ytick_11\">\n     <g id=\"line2d_18\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"44.64\" xlink:href=\"#m43883b6bfa\" y=\"41.969333\"/>\n      </g>\n     </g>\n     <g id=\"text_67\">\n      <!-- 0.75 -->\n      <g transform=\"translate(51.64 46.26402)scale(0.12 -0.12)\">\n       <use xlink:href=\"#ArialMT-48\"/>\n       <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n       <use x=\"83.398438\" xlink:href=\"#ArialMT-55\"/>\n       <use x=\"139.013672\" xlink:href=\"#ArialMT-53\"/>\n      </g>\n     </g>\n    </g>\n    <g id=\"ytick_12\">\n     <g id=\"line2d_19\">\n      <g>\n       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"44.64\" xlink:href=\"#m43883b6bfa\" y=\"11.494688\"/>\n      </g>\n     </g>\n     <g id=\"text_68\">\n      <!-- 1.00 -->\n      <g transform=\"translate(51.64 15.789375)scale(0.12 -0.12)\">\n       <use xlink:href=\"#ArialMT-49\"/>\n       <use x=\"55.615234\" xlink:href=\"#ArialMT-46\"/>\n       <use x=\"83.398438\" xlink:href=\"#ArialMT-48\"/>\n       <use x=\"139.013672\" xlink:href=\"#ArialMT-48\"/>\n      </g>\n     </g>\n    </g>\n   </g>\n   <g id=\"patch_4\">\n    <path d=\"M 8.64 141.094688 \nL 8.64 132.454688 \nL 8.64 20.134688 \nL 8.64 11.494688 \nL 44.64 11.494688 \nL 44.64 20.134688 \nL 44.64 132.454688 \nL 44.64 141.094688 \nz\n\" style=\"fill:none;\"/>\n   </g>\n  </g>\n </g>\n <defs>\n  <clipPath id=\"p8274827798\">\n   <rect height=\"355.772\" width=\"88.943\" x=\"7.2\" y=\"99.237688\"/>\n  </clipPath>\n  <clipPath id=\"p7e472281ad\">\n   <rect height=\"88.943\" width=\"355.772\" x=\"96.383\" y=\"10.054688\"/>\n  </clipPath>\n  <clipPath id=\"peec049f61d\">\n   <rect height=\"355.772\" width=\"355.772\" x=\"96.383\" y=\"99.237688\"/>\n  </clipPath>\n  <clipPath id=\"p8803b15104\">\n   <rect height=\"129.6\" width=\"36\" x=\"8.64\" y=\"11.494688\"/>\n  </clipPath>\n </defs>\n</svg>\n",
      "text/plain": [
       "<Figure size 720x720 with 4 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Xiang-Yu et al.(iCCA_China_225_2022)</th>\n",
       "      <th>TCGA(CCA_US_Can_2017)</th>\n",
       "      <th>MSKCC_cohort(CCA_US_198_2018)</th>\n",
       "      <th>ICGC(BTC_Japan_260_2015)</th>\n",
       "      <th>Jusakul et al.(CCA_10countries_489_2017)</th>\n",
       "      <th>Fudan_cohort(CCA_China_217_2022)</th>\n",
       "      <th>Fudan_Dong_cohort(iCCA_China_253_2022)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Xiang-Yu et al.(iCCA_China_225_2022)</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.427250</td>\n",
       "      <td>0.843495</td>\n",
       "      <td>0.694969</td>\n",
       "      <td>0.663790</td>\n",
       "      <td>0.615864</td>\n",
       "      <td>0.931905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA(CCA_US_Can_2017)</th>\n",
       "      <td>0.427250</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.707827</td>\n",
       "      <td>-0.063179</td>\n",
       "      <td>-0.049530</td>\n",
       "      <td>0.418849</td>\n",
       "      <td>0.402565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSKCC_cohort(CCA_US_198_2018)</th>\n",
       "      <td>0.843495</td>\n",
       "      <td>0.707827</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.355339</td>\n",
       "      <td>0.394845</td>\n",
       "      <td>0.683723</td>\n",
       "      <td>0.799958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ICGC(BTC_Japan_260_2015)</th>\n",
       "      <td>0.694969</td>\n",
       "      <td>-0.063179</td>\n",
       "      <td>0.355339</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.944585</td>\n",
       "      <td>0.599535</td>\n",
       "      <td>0.722350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Jusakul et al.(CCA_10countries_489_2017)</th>\n",
       "      <td>0.663790</td>\n",
       "      <td>-0.049530</td>\n",
       "      <td>0.394845</td>\n",
       "      <td>0.944585</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.679980</td>\n",
       "      <td>0.624711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Fudan_cohort(CCA_China_217_2022)</th>\n",
       "      <td>0.615864</td>\n",
       "      <td>0.418849</td>\n",
       "      <td>0.683723</td>\n",
       "      <td>0.599535</td>\n",
       "      <td>0.679980</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.673290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Fudan_Dong_cohort(iCCA_China_253_2022)</th>\n",
       "      <td>0.931905</td>\n",
       "      <td>0.402565</td>\n",
       "      <td>0.799958</td>\n",
       "      <td>0.722350</td>\n",
       "      <td>0.624711</td>\n",
       "      <td>0.673290</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Xiang-Yu et al.(iCCA_China_225_2022)  \\\n",
       "Xiang-Yu et al.(iCCA_China_225_2022)                                  1.000000   \n",
       "TCGA(CCA_US_Can_2017)                                                 0.427250   \n",
       "MSKCC_cohort(CCA_US_198_2018)                                         0.843495   \n",
       "ICGC(BTC_Japan_260_2015)                                              0.694969   \n",
       "Jusakul et al.(CCA_10countries_489_2017)                              0.663790   \n",
       "Fudan_cohort(CCA_China_217_2022)                                      0.615864   \n",
       "Fudan_Dong_cohort(iCCA_China_253_2022)                                0.931905   \n",
       "\n",
       "                                          TCGA(CCA_US_Can_2017)  \\\n",
       "Xiang-Yu et al.(iCCA_China_225_2022)                   0.427250   \n",
       "TCGA(CCA_US_Can_2017)                                  1.000000   \n",
       "MSKCC_cohort(CCA_US_198_2018)                          0.707827   \n",
       "ICGC(BTC_Japan_260_2015)                              -0.063179   \n",
       "Jusakul et al.(CCA_10countries_489_2017)              -0.049530   \n",
       "Fudan_cohort(CCA_China_217_2022)                       0.418849   \n",
       "Fudan_Dong_cohort(iCCA_China_253_2022)                 0.402565   \n",
       "\n",
       "                                          MSKCC_cohort(CCA_US_198_2018)  \\\n",
       "Xiang-Yu et al.(iCCA_China_225_2022)                           0.843495   \n",
       "TCGA(CCA_US_Can_2017)                                          0.707827   \n",
       "MSKCC_cohort(CCA_US_198_2018)                                  1.000000   \n",
       "ICGC(BTC_Japan_260_2015)                                       0.355339   \n",
       "Jusakul et al.(CCA_10countries_489_2017)                       0.394845   \n",
       "Fudan_cohort(CCA_China_217_2022)                               0.683723   \n",
       "Fudan_Dong_cohort(iCCA_China_253_2022)                         0.799958   \n",
       "\n",
       "                                          ICGC(BTC_Japan_260_2015)  \\\n",
       "Xiang-Yu et al.(iCCA_China_225_2022)                      0.694969   \n",
       "TCGA(CCA_US_Can_2017)                                    -0.063179   \n",
       "MSKCC_cohort(CCA_US_198_2018)                             0.355339   \n",
       "ICGC(BTC_Japan_260_2015)                                  1.000000   \n",
       "Jusakul et al.(CCA_10countries_489_2017)                  0.944585   \n",
       "Fudan_cohort(CCA_China_217_2022)                          0.599535   \n",
       "Fudan_Dong_cohort(iCCA_China_253_2022)                    0.722350   \n",
       "\n",
       "                                          Jusakul et al.(CCA_10countries_489_2017)  \\\n",
       "Xiang-Yu et al.(iCCA_China_225_2022)                                      0.663790   \n",
       "TCGA(CCA_US_Can_2017)                                                    -0.049530   \n",
       "MSKCC_cohort(CCA_US_198_2018)                                             0.394845   \n",
       "ICGC(BTC_Japan_260_2015)                                                  0.944585   \n",
       "Jusakul et al.(CCA_10countries_489_2017)                                  1.000000   \n",
       "Fudan_cohort(CCA_China_217_2022)                                          0.679980   \n",
       "Fudan_Dong_cohort(iCCA_China_253_2022)                                    0.624711   \n",
       "\n",
       "                                          Fudan_cohort(CCA_China_217_2022)  \\\n",
       "Xiang-Yu et al.(iCCA_China_225_2022)                              0.615864   \n",
       "TCGA(CCA_US_Can_2017)                                             0.418849   \n",
       "MSKCC_cohort(CCA_US_198_2018)                                     0.683723   \n",
       "ICGC(BTC_Japan_260_2015)                                          0.599535   \n",
       "Jusakul et al.(CCA_10countries_489_2017)                          0.679980   \n",
       "Fudan_cohort(CCA_China_217_2022)                                  1.000000   \n",
       "Fudan_Dong_cohort(iCCA_China_253_2022)                            0.673290   \n",
       "\n",
       "                                          Fudan_Dong_cohort(iCCA_China_253_2022)  \n",
       "Xiang-Yu et al.(iCCA_China_225_2022)                                    0.931905  \n",
       "TCGA(CCA_US_Can_2017)                                                   0.402565  \n",
       "MSKCC_cohort(CCA_US_198_2018)                                           0.799958  \n",
       "ICGC(BTC_Japan_260_2015)                                                0.722350  \n",
       "Jusakul et al.(CCA_10countries_489_2017)                                0.624711  \n",
       "Fudan_cohort(CCA_China_217_2022)                                        0.673290  \n",
       "Fudan_Dong_cohort(iCCA_China_253_2022)                                  1.000000  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import seaborn as sn\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.colors as mcolors\n",
    "# define the colormap ['#1c3e9f', 'white', '#fc0000']EE1D24 2369AD fc0000\n",
    "cmap = mcolors.LinearSegmentedColormap.from_list(\n",
    "    \"red_blue_gradient\", ['white', '#fc0000'], N=15)\n",
    "\n",
    "def correlation_heatmap(data, method = 'pearson', log_transformed = False, outpath = None):\n",
    "    \"\"\" Correaltion heatmap and correlation matrix \n",
    "    Args:\n",
    "    --------------\n",
    "    data -> Dataframe: dataframe of raw data\n",
    "    method -> str: pearson or spearman\n",
    "    log_transformed -> bool: whether to use log10 transformed data\n",
    "    \n",
    "    Return:\n",
    "    -----------\n",
    "    Dataframe: Correaltion of experiments data\n",
    "    \n",
    "    \"\"\"\n",
    "    if log_transformed:\n",
    "        data = np.log10(data)\n",
    "    corr = data.corr(method = method)\n",
    "    ## Platelet selection refernce: https://learnku.com/articles/39890\n",
    "    sn.clustermap(corr, annot= True, cmap=cmap,linewidths=1.5, color='black')  ## optional cmap: RdYlGn; RdYlGn_r; YlGn; rocket_r; YlGnBu; YlGnBu_r; YlOrBr; YlOrBr_r; YlOrRd; YlOrRd_r; RdBu_r\n",
    "    # sn.heatmap(corr, annot=True, cmap='vlag')  ## optional cmap: RdYlGn; RdYlGn_r; YlGn; rocket_r; YlGnBu; YlGnBu_r; YlOrBr; YlOrBr_r; YlOrRd; YlOrRd_r;\n",
    "    if outpath:\n",
    "        plt.savefig(outpath + \"/correlation_heatmap4.pdf\")\n",
    "        corr.to_csv(outpath + \"/correlation_matrix.csv\")\n",
    "    plt.show()\n",
    "    return corr\n",
    "\n",
    "\n",
    "# corr = QC1.corr(method ='pearson')\n",
    "# sn.heatmap(corr, cmap=\"rocket_r\", annot=True)\n",
    "# QC_data_293T = pd.read_csv(\"/Users/ranpeng/Desktop/Desktop/项目文件/对外服务/罗俊一/results/QC_data.csv\", index_col = 0)\n",
    "outpath = \"/Users/ranpeng/Desktop/Integrated_multiomics_analysis_CCA/figure_subpanels/Fig1\"\n",
    "correlation_heatmap(cohort_mutation_matrix, method = 'pearson', outpath = outpath)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "b280c0ad830b7655c5b9fcdd6e5c3b5ca1f2792faa48bd8ab0a65566022419f1"
  },
  "kernelspec": {
   "display_name": "Python 3.8.12 64-bit ('base': conda)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
